---
title: "N_deficiency (minusN) responsive_genes 3"
output:
  html_document:
    df_print: paged
editor_options:
  chunk_output_type: console
---

Purpose:

|our purpose|Study|plant|samples|transcriptome platform|available data|available raw data|
|----|----|--|------|--|--------|-|

#### under construction #####
|Root response to NO3 or NH4 addtition after having been N depleted for 26h|[Patterson et al. 2011](https://onlinelibrary.wiley.com/doi/full/10.1111/j.1365-3040.2010.02158.x)|
Arabidopsis thaliana|Col plants raised hydroponically on full nutrient supply for ~20d, then raised in N free for 26hrs (depleting N), then at 0, 0.5, 1.5, 8h after either NO3 or NH4 added, root samples taken for microarray. |Microarray (ATH1 chip)|Table S2, S3, & S4 provide FDR adjusted DGE to both N sources, NO3 specific, or NH4 specific over time.  Tables list FC & FDR pvalue.  https://onlinelibrary.wiley.com/action/downloadSupplement?doi=10.1111%2Fj.1365-3040.2010.02158.x&file=PCE_2158_sm_TableS1-8.xls|[GSE29589](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE29589)|

To Do
* This is not ATH1. Stop working on it.
* Expression pattern graph
* Knit to finalize analysis.

# installation
```{r eval=FALSE, include=FALSE}
install.packages("tidyverse")
install.packages("broom")
install.packages("broom.mixed")
install.packages("lmerTest")
install.packages("rmarkdown")
# https://bioconductor.org/packages/devel/workflows/vignettes/maEndToEnd/inst/doc/MA-Workflow.html
install.packages("devtools")
library(devtools)
devtools::install_github("r-lib/remotes")
#library(remotes)
#packageVersion("remotes") # has to be 1.1.1.9000 or later
remotes::install_github("b-klaus/maEndToEnd", ref="master")
```

# prep
```{r setup, include=FALSE}
library(lmerTest);library(broom);library(broom.mixed);library(tidyverse);library(lme4);library(affy)
# https://bioconductor.org/packages/devel/workflows/vignettes/maEndToEnd/inst/doc/MA-Workflow.html
source("Expression_pattern_graph.R")
```
# Read cel files and normalize data (copied from "RNAseqVSmicroarray_publication_final.R")
```{r eval=TRUE}
cel.filename<-list.files(pattern=".CEL$") # no directory info
cel.filename2<-list.files(pattern=".CEL",path=file.path("","Volumes","data_work","Data8","Microarray","Patterson_2011_N","GSE29589_RAW"),full.names=TRUE) # "" is needed for file.path()!!!  with directory info
Data1<-ReadAffy(filenames =cel.filename2) 
head(Data1)
eset <- rma(Data1) # normalize chips (old way)
# using expresso (see affy vignett)
 eset.expresso<-expresso(Data1,normalize.method="qspline",
               bgcorrect.method="rma",pmcorrect.method="pmonly",
               summary.method="liwong")
 class(eset.expresso) # ExpressionSet object. See Biobase vignette
 head(eset.expresso)
 exprs(eset.expresso) %>% head()# expression data
 experimentData(eset.expresso)
```
# read probe ID and AGI
```{r eval=TRUE}
affyData <- read_csv(file.path("..","Annotation","input","affy_ATH1_array_elements-2010-12-20.csv")) #read TAIR annotation data from file (TAIR10)
affyData2<-affyData[,c("array_element_name","is_control","locus","description","chromosome","start","stop")]
names(affyData2) <- c("element","is.control","AGI","description","chromosome","start","stop")#get rid of underlines
head(affyData2)
# only unique element per AGI (is this OK? ASk Julin)
affyData2.elementPerAGI.table<- affyData2 %>% group_by(AGI) %>% summarize(n=n())
table(affyData2.elementPerAGI.table$n)
affyData2unique<-affyData2.elementPerAGI.table %>% filter(n<2) %>% left_join(affyData2,by="AGI")
dim(affyData2unique) #[1] 21789     8 use this instead of affyData2
dim(affyData2) #[1] 22810     7
```


# Make expression value tibble
```{r eval=TRUE}
featureNames(eset.expresso) %>% head() # only probe name
exp.values<-exprs(eset.expresso) %>% as_tibble(rownames="element") 
```

# experimental design
```{r eval=TRUE}
# extract experiment data info
#cel.filename<-list.files(pattern=".CEL",path=file.path("","Volumes","data_work","Data8","Microarray","CN_GutieÌrrez_2007","E-MEXP-828.raw.1"),full.names=FALSE) # "" is needed for file.path()!!!  with directory info
# needs exp design description info

# design (under construction)
# GSM732600	immediately prior to nitrogen addtion, biological rep1
# GSM732601	immediately prior to nitrogen addtion, biological rep2
# GSM732602	immediately prior to nitrogen addtion, biological rep3
# GSM732603	1.5 hr after nitrate addition, biological rep1
# GSM732604	1.5 hr after nitrate addition, biological rep2
# GSM732605	1.5 hr after nitrate addition, biological rep3
# GSM732606	1.5 hr after ammonium addition, biological rep1
# GSM732607	1.5 hr after ammonium addition, biological rep2
# GSM732608	1.5 hr after ammonium addition, biological rep3
# GSM732609	8 hr after nitrate addition, biological rep1
# GSM732610	8 hr after nitrate addition, biological rep2
# GSM732611	8 hr after nitrate addition, biological rep3
# GSM732612	8 hr after ammonium addition, biological rep1
# GSM732613	8 hr after ammonium addition, biological rep2
# GSM732614	8 hr after ammonium addition, biological rep3

design <- tibble(cel_file=c("GSM732600","GSM732601","GSM732602","GSM732603","GSM732604","GSM732605","GSM732606","GSM732607","GSM732608","GSM732609","GSM732610","GSM732611","GSM732612","GSM732613","GSM732614"),
                            trt1=rep(c("0hr","1.5hr","8hr"),c(3,6,6)),trt2=c("none","none","none",rep(rep(c("nitrate","ammonium"),each=3),2)),
                 rep=rep(c("rep1","rep2","rep3"),5),
                 tissue="root"
                            )


# design <- tibble(cel_file=cel.filename) %>% mutate(cel_file2=str_replace(cel_file,"Wild-type","WT")) %>% separate(cel_file2,into=c("NRID","Wang","num","condition","rep","extra"),extra="drop",sep="_") %>% mutate(condition=str_replace(condition,"K-nitrate","KNO3"),condition=str_replace(condition,"KCL","KCl")) %>% 
#   separate(condition,into=c("gt1","gt2","conc","unit","trt","time","unit2","tissue")) %>% mutate(rep=str_replace(rep,"Rep",""),tissue=str_replace(tissue,"s$","")) %>% dplyr::select(cel_file,gt1,trt,tissue,rep)
# #
# design
# # root or shoot
# design.root <- design %>% filter(tissue=="root") 
# design.shoot <- design %>% filter(tissue=="shoot") 
```

# combine
```{r eval=TRUE}
exp.values.combined <- exp.values %>% gather("cel_file","value",-element) %>% mutate(cel_file=str_remove(cel_file,"\\.CEL.gz")) %>% left_join(design,by="cel_file") %>% 
  left_join(affyData2, by="element") 
# check 10 genes
exp.values.combined %>% arrange(AGI) %>% head(10)

#### wait! this is not what I want #######
# select only 0 mM sucrose
exp.values.combined.nosucrose <- exp.values.combined %>% dplyr::filter(trt2==" 0 mM sucrose")
# factor (N)
exp.values.combined.nosucrose <- exp.values.combined.nosucrose %>% mutate(trt1=factor(trt1, levels=c("0 mM nitrate","5 mM nitrate","10 mM nitrate","15 mM nitrate"))) #%>% str() # N deficiency
# write in csv file
write_csv(exp.values.combined.nosucrose,path=file.path("intermediate","minusN_Gutierrez_2007","exp.values.combined.nosucrose.csv.gz"))
```

# lmer function
```{r eval=TRUE}
dofit.lmer.ATH1 <- function(genedata) { # gives a lot of singularity warnings
  lmer(value ~ trt + (1|rep),
       data=genedata)
}
```

# root
# organize the data to make it easy to fit a model to each element (not gene)
```{r eval=TRUE}
exp.values.combined <- read_csv(file.path("intermediate","minusN_Gutierrez_2007","exp.values.combined.nosucrose.csv.gz"))
#exp.values.combined %>% View()
exp.values.combined.root.nest <- exp.values.combined %>%
  mutate(trt=factor(trt1, levels=c("5 mM nitrate","0 mM nitrate","10 mM nitrate","15 mM nitrate"))) %>% dplyr::select(value,trt,element,id) %>% dplyr::rename(rep=id) %>%
  group_by(element) %>%
  nest()
head(exp.values.combined.root.nest)
exp.values.combined.root.nest %>% summarize(num=n()) %>% arrange(num) %>% View() #OK
exp.values.combined.root.nest$data[[1]] %>% str() # factor...
```

# do a fit for each element
```{r eval=TRUE, warning=FALSE}
root.fits <- exp.values.combined.root.nest %>% #head(100) %>% 
  #magrittr::extract(1:10,) %>% # comment out this line to do full set; uncomment for testing on small set.
  mutate(fit.lmer.ATH1 = map(data, dofit.lmer.ATH1),
         lmersingular = map_lgl(fit.lmer.ATH1, isSingular))
head(root.fits)
```

# add model summaries (5-10 minutes) using broom package (read "introduction to broom" vegnet)
```{r eval=TRUE}
root.fits.summary <- root.fits %>%
  mutate(glancelmer = map(fit.lmer.ATH1, broom::glance),
         tidylmer = map(fit.lmer.ATH1, broom::tidy)
  ) # see tidy.lm
```

# unnesting by unnest() 
```{r}
root.fits.summary %>% unnest(tidylmer) %>% head(100) %>% View()
```

# make a 1 row summary of the tidy results (modified from copy from J's script "SingleGeneLME_DGE.Rmd")

```{r eval=TRUE}
root.fits.summary.table.all <- root.fits.summary %>% unnest(tidylmer, .drop = TRUE) %>%
  select(element, term, estimate, std.error, p.value) %>%
  na.omit() %>%
  mutate(p.value = ifelse(term=="(Intercept)", NA, p.value), # we don't care about intercept p-value and don't want it to mess up FDR
         FDR = p.adjust(p.value, "fdr")) %>%
  gather(key="key", value="value", estimate, std.error, p.value, FDR) %>%
  mutate(key=str_c(key, "_", term)) %>%
  select(-term) %>%
  spread(key=key, value=value) #%>% 
  #  arrange(FDR_trtKCl) #%>% 
  #head(100) %>% View()
# how many DEGs?
root.fits.summary.table.all %>% filter(`FDR_trt0 mM nitrate` < 0.05) %>% dim() # 0 mM nitrate vs 5 mM
write_csv(root.fits.summary.table.all,path=file.path("custom_categories","minusN_Gutierrez2007.root.fits.summary.table.all.csv.gz"))

```

# How to calculate logFC?
# read root.fits.summary.table.all 
```{r}
root.fits.summary.table.all<-read_csv(file.path("custom_categories","minusN_Gutierrez2007.root.fits.summary.table.all.csv.gz"))
root.fits.summary.table.all
# only this works
fits.summary.unnest<-root.fits.summary %>% 
  #head(10) %>% 
  mutate(predicted.values=map(fit.lmer.ATH1,predict)) %>% unnest(predicted.values) 
# adding samplename column
samplename <- design %>% filter(trt2==" 0 mM sucrose") %>% unite(samplename,trt1,trt2,id,sep=".") %>% dplyr::select(samplename) %>% as_vector()
# calculate predicted expression values (used also for expression graph)
fits.element.exp.value<-fits.summary.unnest %>% ungroup() %>% inner_join(affyData2unique, by="element") %>% group_by(element) %>% mutate(samplename=samplename)  %>% 
  separate(samplename,c("trt1","trt2","id"),sep="\\.") %>% rename(value=predicted.values) %>% dplyr::select(element,value,AGI,description,trt1,trt2,id)
# save fits.element.exp.value
write_csv(fits.element.exp.value,file.path("custom_categories","minusN_Gutierrez_2007.root.fits.element.exp.value.csv.gz"))
# calculate mean value per element, treatment
fits.element.exp.value.mean <- fits.element.exp.value %>%
  group_by(element,trt1) %>% 
  summarise(mean.element=mean(value)) %>% 
  spread(trt1,mean.element,-1) %>%
  inner_join(affyData2unique, by="element") # only one element per one locus
#  check if "one" per one element
 fits.element.exp.value.mean %>% group_by(element) %>% summarize(num.element=n()) %>% arrange(num.element) %>% View()
 # calculate log fold changes per each locus (one locus = one element)
 fits.summary.AGI.FC <- fits.element.exp.value.mean %>% 
   group_by(AGI) %>% 
   mutate(
     mean.FC.0per5mM.AGI=log(`0 mM nitrate`/`5 mM nitrate`),
     mean.FC.10per5mM.AGI=log(`10 mM nitrate`/`5 mM nitrate`),
     mean.FC.15per5mM.AGI=log(`15 mM nitrate`/`5 mM nitrate`)) %>%
  # add FDR_trt5 mM nitrate, 
  inner_join(root.fits.summary.table.all 
            %>% dplyr::select(element,`FDR_trt0 mM nitrate`,`FDR_trt10 mM nitrate`,`FDR_trt15 mM nitrate`), by="element") 
 fits.summary.AGI.FC %>% View()
 # check AGI number
 fits.summary.AGI.FC %>% group_by(AGI) %>% summarize(num.AGI=n()) %>% arrange(num.AGI) %>% View() # there is no  "no_match"
# write
write_csv(fits.summary.AGI.FC,path=file.path("custom_categories","minusN_Gutierrez_2007.root.fits.summary.AGI.FC.csv.gz"))
```


# expression pattern (root)
```{r eval=TRUE}
fits.summary.AGI.FC <- read_csv(file.path("custom_categories","minusN_Gutierrez_2007.root.fits.summary.AGI.FC.csv.gz"))
fits.element.exp.value <- read_csv(file.path("custom_categories","minusN_Gutierrez_2007.root.fits.element.exp.value.csv.gz"))
# minusN_Gutierrez_2007 N deficienty
gene.of.interest.FDR.up <- fits.summary.AGI.FC %>% ungroup() %>% rename(FDR=`FDR_trtKCl`) %>% filter(FDR<0.05,mean.FC.AGI>0) %>% dplyr::select(AGI,FDR) %>% arrange(FDR) 
print("logFC>0")
p.up<-expression.pattern.AT.graph.ATH1(target.genes.FDR=gene.of.interest.FDR.up[1:5,],
                              data=fits.element.exp.value,
                              sample.description=design,
                              title="minusN_Gutierrez_2007 root -N up") # good
p.up
ggsave(p.up,filename=file.path("custom_categories","minusN_Gutierrez_2007_root.up.png"))
expression.down.plot <- "minusN_Gutierrez_2007_root.down.png"

# minusN_Gutierrez_2007 N deficienty down
gene.of.interest.FDR.down <- fits.summary.AGI.FC %>% ungroup() %>% rename(FDR=`FDR_trtKCl`) %>% filter(FDR<0.05,mean.FC.AGI<0) %>% dplyr::select(AGI,FDR) %>% arrange(FDR) 
print("logFC<0")
p.down<-expression.pattern.AT.graph.ATH1(target.genes.FDR=gene.of.interest.FDR.down[1:5,],
                              data=fits.element.exp.value,
                              sample.description=design,
                              title="minusN_Gutierrez_2007 root -N down") # good
p.down
ggsave(p.down,filename=file.path("custom_categories","minusN_Gutierrez_2007_root.down.png"))

```


# Session info
```{r}
sessionInfo()
```
