---
title: "10_WGCNA_including_R500_signedhybrid_moduleColors"
author: "Kazu"
date: "7/19/2021"
output: html_document
  #flexdashboard::flex_dashboard:
  #  css: libs/my-theme.css
  #  orientation: columns
  #  vertical_layout: scroll
    #social: menu
  #  source_code: embed
    #theme: default
editor_options: 
  chunk_output_type: inline
---

purpose: WGCNA analysis to look correlation of plant data and expression data
background: 

* Plus R500
* GSEA with leaf length correlated modules only (110920)
* bug fixed in making.gmx() which does not affect results in this, but affected in "10_WGCNA_including_R500_FDR_modules.Rmd" (111020)
* serious bug fixed in WGCNA.module.trait.cor() (120220)
* cleaning up for Br_Microbe_Paper_2021 repository (030521)
* Switched to ""signed hybrid" in ,networkType (041621), which will results all WGCNA module related analysis.
* see blog posts for [signed vs unsigned1](https://peterlangfelder.com/2018/11/25/signed-or-unsigned-which-network-type-is-preferable/]), [signed vs unsigned2](https://peterlangfelder.com/2018/11/25/__trashed/) (referred from Julin)
* replace dynamicColors into moduleColors to match modules in MEs (that affect ORA and others?, 071921)
* Change output format from flex_dashboard to html (Nov 15, 2021)
* Julin found the format of "WGCNA.signedhybrid.moduleColors.voom5.csv" was disrupted (March 29, 2023). OMG, needs to fix it.

```{r setup}
rm(list=ls())
knitr::opts_chunk$set(error = TRUE,warning=FALSE)
library(tidyverse);library(readr);library(readxl);library(openxlsx)
# The following setting is important for WGCNA, do not omit.
 options(stringsAsFactors = FALSE)
# install.packages(c("dynamicTreeCut", "cluster", "flashClust", "Hmisc", "reshape", "foreach", "doParallel") ) 
library(WGCNA);library(flashClust) # for WGCNA in Whitney
enableWGCNAThreads(4) # for Whitney
# for my computer
disableWGCNAThreads()
```


load normalized log2 expression data (Brassica genome v3.0 annotation) 
```{r}
# plus R500 in residual
root.voom5 <- readr::read_tsv(file.path("..","output","voom_expression.e1and3.resid.exp_gt.root.plusR500.txt.gz"))
leaf.voom5 <- readr::read_tsv(file.path("..","output","voom_expression.e1and3.resid.exp_gt.leaf.plusR500.txt.gz"))
```

### check if voom is log2 transformed
```{r}
root.voom5[1:1000,-1] %>% gather() %>% ggplot(aes(x=value))+geom_histogram() # yes. log2 transformed
```

make sample description data frame

```{r}
sample.description.e1 <- read_csv(file.path("..","..","plant","output","Br.mbio.e1.sample.description.csv"))
sample.description.e3 <- read_csv(file.path("..","..","plant","output","Br.mbio.e3.sample.description.csv"))

sample.description.e1 %>% summarize(n_distinct(group))
sample.description.e1 %>% group_by(trt) %>% summarize(n())
##
root.voom5.e1 <- tibble(root.voom5 = colnames(root.voom5)[-1]) %>% inner_join(sample.description.e1[,c("sample","genotype","tissue","trt")],by=c("root.voom5"="sample"))
root.voom5.e3 <- tibble(root.voom5 = colnames(root.voom5)[-1]) %>% mutate(genotype="FPsc") %>% inner_join(sample.description.e3[,c("sample","tissue","trt")],by=c("root.voom5"="sample"))  
root.voom5.sample <- bind_rows(root.voom5.e1,root.voom5.e3) %>% mutate(trt=str_remove(trt,"5E_"))
# for Whitney
# root.voom5.sample <- bind_rows(root.voom5.e1,root.voom5.e3)
# root.voom5.sample$trt <- gsub("5E_","",root.voom5.sample$trt)
root.voom5.sample %>% group_by(genotype) %>% summarise(num=n())


# leaf.voom5.sample
leaf.voom5.e1 <- tibble(leaf.voom5 = colnames(leaf.voom5)[-1]) %>% inner_join(sample.description.e1[,c("sample","genotype","tissue","trt")],by=c("leaf.voom5"="sample"))
leaf.voom5.e3 <- tibble(leaf.voom5 = colnames(leaf.voom5)[-1]) %>% mutate(genotype="FPsc") %>% inner_join(sample.description.e3[,c("sample","tissue","trt")],by=c("leaf.voom5"="sample"))  
leaf.voom5.sample <- bind_rows(leaf.voom5.e1,leaf.voom5.e3) %>% mutate(trt=str_remove(trt,"5E_")) # str_remove did not work in Whitney (041721)
# for Whitney
# leaf.voom5.sample <- bind_rows(leaf.voom5.e1,leaf.voom5.e3)
# leaf.voom5.sample$trt <- gsub("5E_","",root.voom5.sample$trt)
leaf.voom5.sample %>% group_by(genotype) %>% summarise(num=n())

```

# creating Phenotype Data format for GSEA. (not necesary anymore? March 29, 2023)
# CLS: Categorical (e.g tumor vs normal) class file format (*.cls) (http://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats#CLS:_Categorical_.28e.g_tumor_vs_normal.29_class_file_format_.28.2A.cls.29)
## root.voom5
```{r}
# for plusR500 (root.voom5)
Total.root.plusR500 <- dim(root.voom5.sample)[1]
description <- "# LIVE DEAD"
Live.Dead <- ifelse(root.voom5.sample$trt=="live",0,1) # 0 is live
# writing cls file
cat(paste(Total.root.plusR500,"2","1","\n"),file=file.path("plant","output","root.voom5.cls"))
cat(paste(description,"\n"),file=file.path("plant","output","root.voom5.cls"),append=TRUE)
cat(Live.Dead,file=file.path("plant","output","root.voom5.cls"),append=TRUE)
```

## leaf.voom5 phenotype data
```{r}
# for plusR500 (shoot.voom5)
Total.leaf.plusR500 <- dim(leaf.voom5.sample)[1]
description <- "# LIVE DEAD"
Live.Dead <- ifelse(leaf.voom5.sample$trt=="live",0,1)
# writing cls file
cat(paste(Total.leaf.plusR500,"2","1","\n"),file=file.path("plant","output","leaf.voom5.cls"))
cat(paste(description,"\n"),file=file.path("plant","output","leaf.voom5.cls"),append=TRUE)
cat(Live.Dead,file=file.path("plant","output","leaf.voom5.cls"),append=TRUE)
```


Checking data for excessive missing values and identification of outlier samples 
(revised, Nov 15, 2021)
```{r eval=FALSE, include=FALSE}
# root
datExpr.root <- t(root.voom5[,-1]) 
gsg<-goodSamplesGenes(datExpr.root,verbose=3)
gsg$allOK # TRUE. If the this is "TRUE", all genes have passed the cuts. If not, we remove the offending genes and samples from the data.

# leaf
datExpr.leaf <- t(leaf.voom5[,-1]) 
dim(datExpr.leaf) # [1]    28 25682 
# no more datExpr.root (Nov 15, 2021)
gsg<-goodSamplesGenes(datExpr.leaf,verbose=3)
gsg$allOK # TRUE. If the this is "TRUE", all genes have passed the cuts. If not, we remove the offending genes and samples from the data.

# if (!gsg$allOK)
# {
# # Optionally, print the gene and sample names that were removed:
# if (sum(!gsg$goodGenes)>0)
# printFlush(paste("Removing genes:", paste(names(datExpr)[!gsg$goodGenes], collapse = ", ")));
# if (sum(!gsg$goodSamples)>0)
# printFlush(paste("Removing samples:", paste(rownames(datExpr)[!gsg$goodSamples], collapse = ", ")));
# # Remove the offending genes and samples from the data:
# datExpr = datExpr[gsg$goodSamples, gsg$goodGenes]
# }
dim(datExpr.root) 
```

sample outliers? 
```{r eval=FALSE, include=FALSE}
#sampleTree.root = hclust(dist(datExpr.root), method = "average")
sampleTree.root = hclust(dist(datExpr.root), method = "average")
sampleTree.leaf = hclust(dist(datExpr.leaf), method = "average")

# Plot the sample tree: Open a graphic output window of size 12 by 9 inches
# The user should change the dimensions if the window is too large or too small.
#sizeGrWindow(12,9)
#pdf(file = "Plots/sampleClustering.pdf", width = 12, height = 9);
#par(cex = 0.6);
#par(mar = c(0,4,2,0))
```

```{r eval=FALSE, include=FALSE}
plot(sampleTree.root, main = "Sample clustering to detect outliers (root)", sub="", xlab="", cex.lab = 1.5,
cex.axis = 1.5, cex.main = 2)
plot(sampleTree.leaf, main = "Sample clustering to detect outliers (leaf)", sub="", xlab="", cex.lab = 1.5,
cex.axis = 1.5, cex.main = 2)
# no need to eliminate samples 
```


Network construction and module detection by WGCNA 
```{r eval=FALSE, include=FALSE}
voom.WGCNA.signedhybrid <- function(voom.data="",datname="") {
  
  datExpr<-t(voom.data[,-1]) 
colnames(datExpr)<-voom.data$genes
rownames(datExpr)<-colnames(voom.data)[-1]
# Choose a set of soft-thresholding powers
powers = c(c(1:10), seq(from = 12, to=20, by=2))
# Call the network topology analysis function
sft = pickSoftThreshold(datExpr, powerVector = powers, verbose = 5,networkType="signed hybrid") # add networkType="signed hybrid" (041621)
# Plot the results:
sizeGrWindow(9, 5)
par(mfrow = c(1,2));
cex1 = 0.9;
# Scale-free topology fit index as a function of the soft-thresholding power
plot(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2],
     xlab="Soft Threshold (power)",ylab="Scale Free Topology Model Fit,signed R^2",type="n",
    main = paste("Scale independence"));
text(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2],
    labels=powers,cex=cex1,col="red");
# this line corresponds to using an R^2 cut-off of h
abline(h=0.90,col="red")
# Mean connectivity as a function of the soft-thresholding power
plot(sft$fitIndices[,1], sft$fitIndices[,5],
    xlab="Soft Threshold (power)",ylab="Mean Connectivity", type="n",
    main = paste("Mean connectivity"))
text(sft$fitIndices[,1], sft$fitIndices[,5], labels=powers, cex=cex1,col="red")
# softPower is set as 4 (040119). how about 10?
softPower = 10;
adjacency = adjacency(datExpr, power = softPower,type = "signed hybrid");
#type = "signed hybrid" (041621)
# Turn adjacency into topological overlap
TOM = TOMsimilarity(adjacency);
dissTOM = 1-TOM
#
# Call the hierarchical clustering function
geneTree = flashClust(as.dist(dissTOM), method = "average");
# flashClust() not found! (080720)
# Plot the resulting clustering tree (dendrogram)
sizeGrWindow(12,9)
plot(geneTree, xlab="", sub="", main = "Gene clustering on TOM-based dissimilarity",
    labels = FALSE, hang = 0.04);
#
# We like large modules, so we set the minimum module size relatively high:
minModuleSize = 30;
# Module identification using dynamic tree cut:
dynamicMods = cutreeDynamic(dendro = geneTree, distM = dissTOM,
              deepSplit = 2, pamRespectsDendro = FALSE,
              minClusterSize = minModuleSize);
table(dynamicMods)
#
# Convert numeric lables into colors
dynamicColors = labels2colors(dynamicMods)
table(dynamicColors)
# Plot the dendrogram and colors underneath
sizeGrWindow(8,6)
plotDendroAndColors(geneTree, dynamicColors, "Dynamic Tree Cut",
                  dendroLabels = FALSE, hang = 0.03,
                  addGuide = TRUE, guideHang = 0.05,
                  main = "Gene dendrogram and module colors")
#
# Calculate eigengenes
MEList = moduleEigengenes(datExpr, colors = dynamicColors)
MEs = MEList$eigengenes
# Calculate dissimilarity of module eigengenes
MEDiss = 1-cor(MEs);
# Cluster module eigengenes
METree = flashClust(as.dist(MEDiss), method = "average");
# Plot the result
sizeGrWindow(7, 6)
plot(METree, main = "Clustering of module eigengenes",
xlab = "", sub = "")
#
MEDissThres = 0.25
# Plot the cut line into the dendrogram
abline(h=MEDissThres, col = "red")
# Call an automatic merging function
merge = mergeCloseModules(datExpr, dynamicColors, cutHeight = MEDissThres, verbose = 3)
# The merged module colors
mergedColors = merge$colors;
# Eigengenes of the new merged modules:
mergedMEs = merge$newMEs;
#
sizeGrWindow(12, 9)
#pdf(file = "Plots/geneDendro-3.pdf", wi = 9, he = 6)
plotDendroAndColors(geneTree, cbind(dynamicColors, mergedColors),
c("Dynamic Tree Cut", "Merged dynamic"),
dendroLabels = FALSE, hang = 0.03,
addGuide = TRUE, guideHang = 0.05)
#dev.off()
#
# Rename to moduleColors
moduleColors = mergedColors
# Construct numerical labels corresponding to the colors
colorOrder = c("grey", standardColors(50));
moduleLabels = match(moduleColors, colorOrder)-1;
MEs = mergedMEs
# Save module colors and labels for use in subsequent parts
save(dynamicColors,MEs, moduleLabels, moduleColors, geneTree, adjacency, TOM, file =paste(datname,".WGCNA.softthresh10.signedhybrid.RData",sep=""))
}
#  
voom.WGCNA.signedhybrid(voom.data=root.voom5,datname="voom_expression.e1and3.resid.exp_gt.root.plusR500.signedhybrid.txt.gz")
voom.WGCNA.signedhybrid(voom.data=leaf.voom5,datname="voom_expression.e1and3.resid.exp_gt.leaf.plusR500.signedhybrid.txt.gz")

```

# gene annotation
```{r}
# annotation file for v3.0annotation
#Br.v3.0.At.BLAST <- read_csv(file.path("annotation","output","Brapa_v3.0_annotated.csv")) 
Br.v3.0.At.BLAST <- read_csv(file.path("annotation","output","Brapa_V3.0_annotated.noAt_Currator_summary.csv")) # April 3, 2023
# This annotation is redundant with name (Br grene). Eg 
Br.v3.0.At.BLAST %>% filter(name=="BraA01g040570.3C")
# reduce the redundancy (112418)
Br.v3.0anno.At.BLAST.highscore <- Br.v3.0.At.BLAST %>% group_by(name) %>% arrange(desc(score)) %>% slice(1)
# reduce the redundancy for Br (051120) (do not use)
Br.v3.0anno.At.BLAST.highscore.Br2 <- Br.v3.0anno.At.BLAST.highscore %>% group_by(AGI) %>% arrange(desc(score)) %>% dplyr::slice(1) #%>% View()
```

# Using moduleColors (after merging): leaf_avg_std (replace dynamicColors into moduleColors to match with modules in MEs used for ML analysis with leaf length) (071921)
```{r}
# load dynamicColors
load(file.path("RNA","output","voom_expression.e1and3.resid.exp_gt.root.plusR500.signedhybrid.txt.gz.WGCNA.softthresh10.signedhybrid.RData")) # 041721
length(dynamicColors);dim(root.voom5);length(moduleColors);dim(MEs) # 40 25
tibble(moduleColors) %>% count(moduleColors) %>% dim() #25, which matches to MEs
tibble(dynamicColors) %>% count(dynamicColors) %>% dim() #49, do not use it.

root.voom5.mod3 <- root.voom5 %>% mutate(root.voom5.module.signedhybrid=moduleColors) %>% dplyr::select(genes, root.voom5.module.signedhybrid)
# size of 25 modules (root) 
root.voom5.mod3 %>% dplyr::count(root.voom5.module.signedhybrid)
# 
load(file.path("RNA","output","voom_expression.e1and3.resid.exp_gt.leaf.plusR500.signedhybrid.txt.gz.WGCNA.softthresh10.signedhybrid.RData"))
length(dynamicColors);dim(leaf.voom5)
leaf.voom5.mod3 <- leaf.voom5 %>% mutate(leaf.voom5.module.signedhybrid=moduleColors) %>% dplyr::select(genes, leaf.voom5.module.signedhybrid)
# size of 31 modules (leaf) 
leaf.voom5.mod3 %>% dplyr::count(leaf.voom5.module.signedhybrid)

```


# add modules (WGCNA.signedhybrid.modules.voom5.csv.gz) -> (WGCNA.signedhybrid.moduleColors.voom5.csv.gz) (071921). Problem found in the CSV file (March 30, 2023). Use Brapa_v3.0_annotated.csv without "At_Curator_summary" -> "Brapa_V3.0_annotated.noAt_Currator_summary.csv"
```{r}
?matchLabels()
Br.v3.0anno.At.BLAST.highscore.moduleColors.signedhybrid <- Br.v3.0anno.At.BLAST.highscore %>%
  left_join(root.voom5.mod3,by=c("name"="genes")) %>%
  left_join(leaf.voom5.mod3,by=c("name"="genes"))
Br.v3.0anno.At.BLAST.highscore.moduleColors.signedhybrid %>% View()
# save the object for a while
#save(Br.v3.0anno.At.BLAST.highscore.moduleColors.signedhybrid,file=file.path("RNA","output","WGCNA.signedhybrid.moduleColors.voom5.Rdata"))
## 
write_csv(Br.v3.0anno.At.BLAST.highscore.moduleColors.signedhybrid,file=file.path("RNA","output","WGCNA.signedhybrid.moduleColors.voom5.new.csv.gz")) # 05012023

# Julin found the prolem (March 31, 2023)
load(file=file.path("RNA","output","WGCNA.signedhybrid.moduleColors.voom5.Rdata"))
# Br.v3.0anno.At.BLAST.highscore.moduleColors.signedhybrid[9460,]$At_Curator_summary
# Br.v3.0anno.At.BLAST.highscore.moduleColors.signedhybrid[9461,]$At_Curator_summary # problem 
# Br.v3.0anno.At.BLAST.highscore.moduleColors.signedhybrid[9462,]$At_Curator_summary
# # so likely Br.v3.0anno.At.BLAST.highscore has problem
# Br.v3.0anno.At.BLAST.highscore[9460,]$At_Curator_summary
# Br.v3.0anno.At.BLAST.highscore[9461,]$At_Curator_summary # problem
# Br.v3.0anno.At.BLAST.highscore[9461,]$name

# 
#Br.v3.0.At.BLAST <- read_csv(file.path("annotation","output","Brapa_v3.0_annotated.csv")) 
#Br.v3.0.At.BLAST[Br.v3.0.At.BLAST$name==Br.v3.0anno.At.BLAST.highscore.moduleColors.signedhybrid[9461,]$name,]$At_Curator_summary # problem
#Br.v3.0.At.BLAST[Br.v3.0.At.BLAST$name==Br.v3.0anno.At.BLAST.highscore.moduleColors.signedhybrid[9460,]$name,]$At_Curator_summary
# 
Br.v3.0.At.BLAST <- read_csv(file.path("annotation","output","Brapa_V3.0_annotated.noAt_Currator_summary.csv")) # April 3, 2023. NoAT_currator_summary column that caused problems.
Br.v3.0.At.BLAST[Br.v3.0.At.BLAST$name==Br.v3.0anno.At.BLAST.highscore.moduleColors.signedhybrid[9461,]$name,] # no problem


```

# GOseq ORA
# load GO.ORA function (copy from "04b_Br.mbio.e3.subset.v3.)
# GOseq function for Brassica rapa (v3.0)
# load Brgo.v3.0anno.Atgoslim.BP.list for GOseq analysis (GO terms) (using split() and map() to avoid long loop)
# Update GO slim (April 2020), but I totally forgot to use it ("Brgo.v3.0anno.Atgoslim.BP.list.unique.Rdata") (old one is ("Brgo.v3.0anno.Atgoslim.BP.list.Rdata"))
```{r eval=TRUE,error=TRUE}
# old version
load(file.path("annotation","output","Brgo.v3.0anno.Atgoslim.BP.list.Rdata"))
# new version
load(file.path("annotation","output","Brgo.v3.0anno.Atgoslim.BP.list.unique.Rdata"))

```

```{r eval=TRUE,error=TRUE}
# GOseq
library(ShortRead);library(goseq);library(GO.db);library("annotate")
# for ggplot heatmap
## uncompress gz file
system(paste("gunzip -c ",file.path("annotation","input","Brapa_genome_v3.0_cds.gz")," > ",file.path("annotation","input","Brapa_genome_v3.0_cds.fa")))
## read cDNA fasta file 
Bra.v3.0_cdna<-readDNAStringSet(file.path("annotation","input","Brapa_genome_v3.0_cds.fa")) # copied from /Volumes/data_work/Data8/NGS_related/Brassica_rapa_Upendra/G3
Bra.v3.0_cdna
## remove fasta file
system(paste("rm ",file.path("annotation","input","Brapa_genome_v3.0_cds.fa"),sep=""))

# GOseq function
GOseq.Brgo.v3.0.Atgoslim.BP.list.unique.ORA2<-function(genelist,padjust=0.05,ontology="BP",custom.category.list=Brgo.v3.0anno.Atgoslim.BP.list.unique,Br_cdna=Bra.v3.0_cdna) { # return GO enrichment table, padjus, padjust=0.05. 
  print(genelist)
  bias<-nchar(Br_cdna)
  names(bias)<-names(Br_cdna)
  # convert list of genelist into vector
  genelist <- genelist %>% unlist()
  TF<-(names(bias) %in% genelist)*1
  names(TF)<-names(bias)
  #print(TF)
  pwf<-nullp(TF,bias.data=bias,plot.fit=FALSE)
  #print(pwf$DEgenes)
  GO.pval <- goseq(pwf,gene2cat=custom.category.list,use_genes_without_cat=TRUE) # format became different in new goseq version (021111). Does not work (042716)
  #GO.pval <- goseq(pwf,gene2cat=Brgo.DF3,use_genes_without_cat=TRUE) # format became different in new goseq version (021111)
  
  #head(GO.pval) 
  if(ontology=="BP") {
    GO.pval2<-subset(GO.pval,ontology=="BP")
  } else if(ontology=="CC") {
    GO.pval2<-subset(GO.pval,ontology=="CC")
  } else if(ontology=="MF") {
    GO.pval2<-subset(GO.pval,ontology=="MF")
  } else {
    GO.pval2<-GO.pval
  }
    
  GO.pval2$over_represented_padjust<-p.adjust(GO.pval2$over_represented_pvalue,method="BH")
  if(GO.pval2$over_represented_padjust[1]>padjust) return(tibble(category="no enriched GO"))
  else {
    enriched.GO<-GO.pval2[GO.pval2$over_represented_padjust<padjust,] 
    print("enriched.GO is")
    print(enriched.GO)
    
    ## write Term and Definition 
    if(ontology=="BP"|ontology=="CC"|ontology=="MF") {
    for(i in 1:dim(enriched.GO)[1]) {
      if(is.null(Term(GOTERM[enriched.GO[i,"category"]]))) {next} else {
      enriched.GO$Term[i]<-Term(GOTERM[[enriched.GO[i,"category"]]])
      enriched.GO$Definition[i]<-Definition(GOTERM[[enriched.GO[i,"category"]]])
      }
    }
    }
    enriched.GO.tb <- as_tibble(enriched.GO)
    print("As tibble enriched.GO is")
    print(enriched.GO.tb)
    return(enriched.GO.tb)
  }
}
#
head(Bra.v3.0_cdna)
# length(bias) # 44239 > 45019 where the bias come from?
#  bias.data vector must have the same length as DEgenes vector!
```


# convert gene list into list object and use map() (I gave up)
# Instad split data frame to make list and apply GOseq function to each list.
```{r eval=FALSE}
# # learning
# mtcars %>%
#   split(.$cyl) %>% # making list?
#   map(~ lm(mpg ~ wt, data = .x)) %>%
#   map_dfr(~ as.data.frame(t(as.matrix(coef(.)))))
# # 
# GOseq.temp <- Br.v3.0anno.At.BLAST.highscore.modules2.voom5.FDR0.25 %>% ungroup() %>% dplyr::select(name,voom.module) %>%
#   split(.$voom.module) %>% map(~GOseq.Brgo.v3.0.Atgoslim.BP.list.ORA2(genelist=.$name))
# GOseq.temp %>% enframe() %>% unnest(value) %>% write_csv(file.path("..","output","voom5.FDR0.25.GOseq.csv"))
```

# GOseq of all WGCNA modules (leaf.voom5.signedbhydrid, root.voom5.signedhybrid version) with GO terms -> WGCNA.signedhybrid.moduleColors.voom5.csv.gz (071921)
## why "WGCNA.signedhybrid.moduleColors.voom5.csv.gz" still has At_Currator_summary? should be removed (May 1, 2023)
```{r}
#Br.v3.0anno.At.BLAST.highscore.modules2 <- read_csv(file.path("..","output","WGCNA.modules2.voom5.csv.gz")) # needs to update
Br.v3.0anno.At.BLAST.highscore.moduleColors.signedhybrid <- read_csv(file.path("RNA","output","WGCNA.signedhybrid.moduleColors.voom5.new.csv.gz")) # 050123 new without At_Currator_summary.
# old version of GO term (not unique)
GOseq.temp.old <- Br.v3.0anno.At.BLAST.highscore.moduleColors.signedhybrid %>% ungroup() %>% dplyr::select(name,root.voom5.module.signedhybrid,leaf.voom5.module.signedhybrid) %>%
pivot_longer(cols=-1,names_to = "voom") %>% drop_na(value) %>% unite(voom.module,c("voom","value")) %>% split(.$voom.module) %>% map(~GOseq.Brgo.v3.0.Atgoslim.BP.list.unique.ORA2(genelist=.$name,custom.category.list=Brgo.v3.0anno.Atgoslim.BP.list)) # ,custom.category.list=GSEA_cust_rootshoot_updown2.list
# error in unnest(value) below (Aug 17, 2022) because of "no enriched GO" as characters. I changed it to tibble in GOseq.Brgo.v3.0.Atgoslim.BP.list.unique.ORA2()

GOseq.temp.old %>% enframe() %>% unnest(value) %>% write_csv(file.path("RNA","output","WGCNA.voom5.signedhybrid.all.moduleColors.GOseq.081822.csv"))
# new version of GO term (unique)
GOseq.temp <- Br.v3.0anno.At.BLAST.highscore.moduleColors.signedhybrid %>% ungroup() %>% dplyr::select(name,root.voom5.module.signedhybrid,leaf.voom5.module.signedhybrid) %>%
pivot_longer(cols=-1,names_to = "voom") %>% drop_na(value) %>% unite(voom.module,c("voom","value")) %>% split(.$voom.module) %>% map(~GOseq.Brgo.v3.0.Atgoslim.BP.list.unique.ORA2(genelist=.$name)) # ,custom.category.list=GSEA_cust_rootshoot_updown2.list
# error in unnest(value) below (Aug 17, 2022) because of "no enriched GO" as characters. I changed it to tibble.

GOseq.temp %>% enframe() %>% unnest(value) %>% write_csv(file.path("RNA","output","WGCNA.voom5.signedhybrid.all.moduleColors.GOseq.unique.new.csv"))

```

# compare old and new GO ORA (Aug 2022)
## Update to "WGCNA.voom5.signedhybrid.all.moduleColors.GOseq.unique.csv" (although there are no changes).
```{r}
old <- GOseq.temp.old %>% enframe() %>% unnest(value) 
new <- GOseq.temp %>% enframe() %>% unnest(value) 
library(waldo)
waldo::compare(old,new) # no difference
# another way
colnames(old)
addition <- slice(old,2)
addition[1,1] <- "test"
addition
# add "addition" to the "old" 
old.mod <- bind_rows(old,addition)
# testing setdiff()
dplyr::setdiff(old, old.mod) # none
setdiff(old.mod, old)        # detecting "addition"
# real data
setdiff(old,new) # no change
setdiff(new,old) # no change
```


# formatting custom categories for GOseq (Sep 29, 2020) (run once)
## needs to change .Rdata object name (not using GSEA) (080421)
```{r eval=FALSE}
#filtered.custom.data <- read_csv(file.path("annotation","output","filtered.custom.data.rootshoot.csv.gz"),guess_max = 3000)
filtered.custom.data <- read_csv(file.path("annotation","output","filtered.custom.data.rootshoot.csv.gz"),guess_max = 3000) # needs to update for plusN_Wang2003 update (081021)


# howmany genes in the list?
summary(!is.na(filtered.custom.data))
# how many categories?
TF <- names(filtered.custom.data) %>% str_detect(c("plus","minus"))
categories.up.down <- names(filtered.custom.data)[TF] %>% gsub("_FC","",.) %>% gsub("_FDR","",.) %>% unique() # 58 categories
categories.up.down.combined <- categories.up.down %>% gsub("_up","",.) %>% gsub("_down","",.) %>% unique() # 30 categories

# convert into list object
 GSEA_cust_rootshoot_updown2 <- filtered.custom.data %>%
  dplyr::select(genes, ends_with("up"), ends_with("down"), -starts_with("exp")) %>%
  pivot_longer(-genes) 
#GSEA_cust_rootshoot_updown2 %>% drop_na(value) %>% View()
GSEA_cust_rootshoot_updown2.list <- GSEA_cust_rootshoot_updown2 %>% drop_na(value) %>% filter(str_detect(name,"FC")) %>% dplyr::select(-value) %>%split(.$genes) %>%   map(~dplyr::select(.,name)) %>% map(~as_vector(.))
save(GSEA_cust_rootshoot_updown2.list,file=file.path("annotation","output","GSEA_cust_rootshoot_updown2.list.Rdata"))
```

# using consolidatd custom categories
```{r}
load(file.path("annotation","output","cust_rootshoot_updown5.consolidated.FDR0.001.list.Rdata"))
cust_rootshoot_updown5.consolidated.FDR0.001.list
```


# GOseq with custom categories (Sep 30, 2020; Oct 28, 2020) -> WGCNA.signedhybrid.moduleColors.voom5.csv.gz (071921) -> cust_rootshoot_updown5.consolidated.FDR0.001.list (081721)
```{r}
# custom categories list
#load(file.path("annotation","output","GSEA_cust_rootshoot_updown2.list.Rdata")) # GSEA_cust_rootshoot_updown2.list
load(file.path("annotation","output","cust_rootshoot_updown5.consolidated.FDR0.001.list.Rdata"))
cust_rootshoot_updown5.consolidated.FDR0.001.list
# all modules
GOseq.temp <- Br.v3.0anno.At.BLAST.highscore.moduleColors.signedhybrid %>% ungroup() %>% dplyr::select(name,root.voom5.module.signedhybrid,leaf.voom5.module.signedhybrid) %>%
pivot_longer(cols=-1,names_to = "voom")  %>% drop_na(value) %>% unite(voom.module,c("voom","value")) %>%  split(.$voom.module) %>% map(~GOseq.Brgo.v3.0.Atgoslim.BP.list.ORA2(genelist=.$name,custom.category.list=cust_rootshoot_updown5.consolidated.FDR0.001.list,ontology="none")) # ,custom.category.list=cust_rootshoot_updown5.consolidated.FDR0.001.list
GOseq.temp %>% enframe() %>% unnest(value) %>% write_csv(file.path("RNA","output","WGCNA.voom5.signedhybrid.all.moduleColors.customcat.consolidated1.GOseq.csv"))

# check results on "root.voom5.module.signedhybrid_lightcyan1", which has "iron" in GO ORA which modules?
### under construction ####
GOseq.temp2 <- Br.v3.0anno.At.BLAST.highscore.moduleColors.signedhybrid %>% ungroup() %>% dplyr::select(name,root.voom5.module.signedhybrid,leaf.voom5.module.signedhybrid) %>%
pivot_longer(cols=-1,names_to = "voom")  %>% drop_na(value) %>% unite(voom.module,c("voom","value")) %>% filter(voom.module=="root.voom5.module.signedhybrid_lightcyan1") %>% 
  split(.$voom.module) %>% map(~GOseq.Brgo.v3.0.Atgoslim.BP.list.ORA2(genelist=.$name,custom.category.list=GSEA_cust_rootshoot_updown2.list,ontology="none")) # OK. Good. Fe categoreis enriched
```









