---
title: "10_WGCNA_ML_selected_modules"
author: "Kazu"
date: "12/2/2020"
output: html_document
editor_options: 
  chunk_output_type: console
---
# History
* For Julin's machine learning selected WGCNA modules correlated to standardized leaf growth ("leaf_avg_std").
* GSEA: soil treatment effects. RNAseq data is voom transformed by considering exp, gt, block effects (voom.root6 or voom.leaf6 data)
* WGCNA ORA: custom categories

# To Do
* Use "ElasticNet0.5_multiCV_Pct_var" instead of "ElasticNet0.5_LOO_prop_var" (031021)
* consistant format for tissue.module with lowercase (031221)

```{r setup}
rm(list=ls())
knitr::opts_chunk$set(error = TRUE,warning=FALSE)
library(tidyverse);library(readr);library(readxl);library(openxlsx)
# The following setting is important for WGCNA, do not omit.
 options(stringsAsFactors = FALSE)
# install.packages(c("dynamicTreeCut", "cluster", "flashClust", "Hmisc", "reshape", "foreach", "doParallel") ) 
library(WGCNA);library(flashClust) # for WGCNA in Whitney
#allowWGCNAThreads(nThreads = NULL) # for Whitney
#enableWGCNAThreads(4) # for Whitney
library(scales) # for muted function
library(ggdendro) # for dendrogram
library(cowplot)
library(ggnewscale)

```

# Julin's Machine Learning selected modules
```{r}
# reading Julin's results of penalized regression (see slide)
# ElasticNet0.5_LOO_prop_var <- read_csv(file.path("plant","output","ElasticNet0.5_LOO_prop_var.csv"))
# ElasticNet0.5_LOO_prop_var <- ElasticNet0.5_LOO_prop_var %>% mutate(J.module=str_remove(ME,"ME")) %>% separate(J.module,into=c("module","tissue")) %>% mutate(module=str_to_upper(module)) %>% unite(tissue.module,c("tissue","module"),sep=".") 
ElasticNet0.5_multiCV_Pct_var <- read_csv(file.path("plant","output","ElasticNet0.5_multiCV_Pct_var.csv")) # Updated version from 11_take2_Machine_Learning
#
ElasticNet0.5_multiCV_Pct_var <- ElasticNet0.5_multiCV_Pct_var %>% unite(tissue.module,c("Module_Organ","Module_Name"),sep=".")  # all lower case
```


# reading GSEA results (all WGCNA modules)
# GSEA analysis with WGCNA modules
## read reports 
## function for oct30 reports copied from "10_WGCNA_including_R500.Rmd"
```{r}
read.gsea_report_files_WGCNA <- function(gsea_report_path,GSEA.FDR.cutoff,tp=c("resid","removeblock"),data="exp_gt_block") {
# open xsl files and save as csvs
# gsea_report
gsea_report_files <- list.files(path=gsea_report_path,full.names=FALSE,include.dirs = FALSE,recursive=TRUE, pattern="(gsea_report_for)([[:print:]]+)(\\.xls)")
# ifelse(setCutoff=="Total", x <- "Total",if(setCutoff<10) {x <-  str_c("In0",setCutoff)} else {x <- str_c("In",setCutoff)})

# gsea_report_files <- gsea_report_files[str_detect(gsea_report_files,x)]
# print(str_c(x,"orless"))

gsea_report_files
# gsea_report (full name)
gsea_report_files2 <- list.files(path=gsea_report_path,full.names=TRUE,recursive=TRUE, pattern="(gsea_report_for)([[:print:]]+)(\\.xls)")
#gsea_report_files2 <- gsea_report_files2[str_detect(gsea_report_files2,x)] 
#gsea_report_files2
# read xls file (file name is "xls", actually tsv. That is why read_xls did not work)
GSEA_reports<-lapply(gsea_report_files2, function(x) read_tsv(file=file.path(x)))
# name
names(GSEA_reports) <- gsea_report_files
  gsub(".xls","",gsub("([[:print:]]+)(/)([[:print:]]+)","\\3",gsea_report_files))
# convert list into one data.frame
GSEA_reports.DF <- GSEA_reports %>% enframe(name="report") %>% unnest(value)
#GSEA_reports.DF %>% View()
# filter only significnt `FDR q-val` (GSEA.FDR.cutoff)
GSEA_reports.DF <- GSEA_reports.DF %>% filter(`FDR q-val`< GSEA.FDR.cutoff)
# 
GSEA_reports.DF <- GSEA_reports.DF %>% separate(report,into=c("report2","report3"),sep="/") %>% mutate(report2=str_replace(report2,"remove.block","removeblock")) %>% separate(report2,into=c("exp","type","type2","tissue","categories"),sep="\\.",fill="left",extra="drop") 
#
GSEA_reports.DF <- GSEA_reports.DF %>% filter(type2==data) # works?
#GSEA_reports.DF$LorD <- gsub("(gsea_report_for_)(0|1)([[:print:]]+)","\\2",GSEA_reports.DF$report3)
#GSEA_reports.DF <- GSEA_reports.DF %>% mutate(LorD=ifelse(LorD==0,"Live","Dead"))
GSEA_reports.DF %>% View()
ifelse(dim(GSEA_reports.DF)[1]==0,print("no significnat categories"),return(GSEA_reports.DF))
}
```

# data: leaf.voom5.module or root,voom5.module (WGCNA modules calculated either from voom_expression.e1and3.resid.exp_gt.root.plusR500 or voom_expression.e1and3.resid.exp_gt.leaf.plusR500). Expression data is "voom_expression.e1and3.resid.exp_gt_block.root.plusR500.txt" or "voom_expression.e1and3.resid.exp_gt_block.leaf.plusR500.txt
```{r}
type="resid";cutoff <- 1;
gsea_report_path=file.path("~","Box","gsea_home","output","oct30")
GSEA_reports.DF <- read.gsea_report_files_WGCNA(gsea_report_path,GSEA.FDR.cutoff=cutoff) # 
# select root.voom5 and leaf.voom5 data for categories
GSEA_reports.DF <- GSEA_reports.DF %>% filter(str_detect(categories,"voom5")) #%>% View()

# rename "NAME" column (for module names)
GSEA_reports.DF <- GSEA_reports.DF %>% mutate(NAME= str_remove(NAME,"MODULE.")) #%>%
  #str_replace("ROOT.VOOM5","voom_expression.e1and3.resid.exp_gt.root.plusR500") %>%
  #str_replace("LEAF.VOOM5","voom_expression.e1and3.resid.exp_gt.leaf.plusR500"))  #%>% View()
# adding L or D info
GSEA_reports.DF$LorD <- gsub("(gsea_report_for_)(0|1)([[:print:]]+)","\\2",GSEA_reports.DF$report3)
GSEA_reports.DF <- GSEA_reports.DF %>% mutate(LorD=ifelse(LorD==0,"Live","Dead"))
GSEA_reports.DF %>% View()
```

# selecting GSEA results for Machine Learning selected modules
```{r}
GSEA_reports.DF.MLselected <- 
GSEA_reports.DF %>% mutate(NAME=NAME %>% str_remove_all(".VOOM5") %>% str_to_lower()) %>% inner_join(ElasticNet0.5_multiCV_Pct_var,by=c("NAME"="tissue.module")) #%>% View()
# write csv
write_csv(GSEA_reports.DF.MLselected,file=file.path("RNA","output","GSEA_reports.DF.MLselected.csv"))
```

# calculate numbers of genes in each module
```{r}
GSEA_cust.root <- read_tsv(file.path("annotation","output","root.voom5.module.gmx")) %>% dplyr::slice(-1) # # remove row1 that has "na"
GSEA_cust.leaf <- read_tsv(file.path("annotation","output","leaf.voom5.module.gmx")) %>% dplyr::slice(-1) # # remove row1 that has "na"
genes.count.root <- GSEA_cust.root %>% 
  pivot_longer(cols=colnames(GSEA_cust.root)) %>% 
  #mutate(name=str_to_upper(name)) %>%
  group_by(name) %>% drop_na() %>% summarise(n=n()) %>% 
  dplyr::rename(NAME=name)
genes.count.leaf <- GSEA_cust.leaf %>% 
  pivot_longer(cols=colnames(GSEA_cust.leaf)) %>% 
  #mutate(name=str_to_upper(name)) %>%
  group_by(name) %>% drop_na() %>% summarise(n=n()) %>% 
  dplyr::rename(NAME=name)
genes.count <- bind_rows(genes.count.root,genes.count.leaf) %>% mutate(NAME=str_remove(NAME,".voom5.module")) #%>% str_replace("ROOT","root") %>% str_replace("LEAF","leaf")) # 031221
#GSEA_reports.DF <- GSEA_reports.DF %>% filter(`FDR q-val` < 0.3)
```


# modify GSEA plot function
```{r}
# plot function
GSEA_reports.DF.plot.dendrogram.v9.WGCNAmodule.notitle.ML <- function(data = GSEA_reports.DF,
                                            #setCutoff = set, 
                                            # ts="root",
                                            # LD="Dead",
                                            #R500="minusR500",
                                            GSEA.FDR.cutoff = cutoff,
                                            #tp=c("residexp_only","uncorrecected"),
                                            leaf_avg_std.ML.data = ElasticNet0.5_multiCV_Pct_var,
                                            top.margin=830,
                                            bottom.margin=40,
                                            dendrogram=c("TRUE", "FALSE"),
                                            legend.height=45,
                                            legend.width=25,
                                            modules=c("fullmodule","selected")) {
  # Scale each measurement (independently) to have a mean of 0 and variance of 1
  # 
# GSEA_reports.DF.formatted <- data %>% filter(type==tp,tissue==ts,LorD==LD) %>%  dplyr::select(setCutoff,tissue,LorD,NAME,`FDR q-val`) %>% mutate(`-log10FDR`=-log10(`FDR q-val`)) %>% dplyr::select(-`FDR q-val`) %>% unite(tissue_LorD_set,tissue, LorD,setCutoff,remove=TRUE) %>%  pivot_wider(names_from=tissue_LorD_set,values_from=`-log10FDR`) %>% group_by(NAME) %>% arrange(NAME) 
# new version (zero in `FDR q-val` is arbitrally converted into 0.000000001 because -log10(0) gave me "Inf")
GSEA_reports.DF.formatted <- data  %>%  dplyr::select(tissue,LorD,NAME,`FDR q-val`) %>%
  mutate(`FDR q-val`=as.numeric(str_replace(`FDR q-val`,"^0$","0.01"))) %>% #View()
  mutate(`-log10FDR`=-log10(`FDR q-val`)) %>% dplyr::select(-`FDR q-val`) %>% unite(tissue_LorD,tissue, LorD,remove=TRUE) %>%  pivot_wider(names_from=tissue_LorD,values_from=`-log10FDR`)  %>% mutate(NAME=str_remove(NAME,"voom_expression.e1and3.resid.exp_gt.")) %>% 
  mutate(NAME=str_remove(NAME,"plusR500.")) %>%
  group_by(NAME) %>% arrange(NAME) 
GSEA_reports.DF.formatted %>% View()
#### think about how to treat with "NA" #########
# GSEA_reports.DF.formatted[is.na(GSEA_reports.DF.formatted)] <- 0
GSEA_reports.DF.formatted <- GSEA_reports.DF.formatted %>% as.data.frame()
rownames(GSEA_reports.DF.formatted) <- GSEA_reports.DF.formatted[,"NAME"]
# add missing columns
#   ## leaf_Live
TF <- sum(str_detect(colnames(GSEA_reports.DF.formatted),"leaf_Live")*1);ifelse(TF==0,GSEA_reports.DF.formatted$`leaf_Live` <- 0,print("There is leaf_Live."))
#   ## leaf_Dead
TF <- sum(str_detect(colnames(GSEA_reports.DF.formatted),"leaf_Dead")*1);ifelse(TF==0,GSEA_reports.DF.formatted$`leaf_Dead` <- 0,print("There is leaf_Dead."))
#   ## root_Live
TF <- sum(str_detect(colnames(GSEA_reports.DF.formatted),"root_Live")*1);ifelse(TF==0,GSEA_reports.DF.formatted$`root_Live` <- 0,print("There is root_Live."))
#   ## root_Dead
TF <- sum(str_detect(colnames(GSEA_reports.DF.formatted),"root_Dead")*1);ifelse(TF==0,GSEA_reports.DF.formatted$`root_Dead` <- 0,print("There is root_Dead."))

## check
print("GSEA_reports.DF.formatted")
GSEA_reports.DF.formatted %>% View()
# replace "NA" to 0
GSEA_reports.DF.formatted[is.na(GSEA_reports.DF.formatted)] <- 0
# remove all colums are zero (otherwise resulting "NaN" after scale())
vars <- colnames(GSEA_reports.DF.formatted) #%>% str_subset(ts) 
GSEA_reports.DF.formatted <- GSEA_reports.DF.formatted %>% filter(!(.data[[vars[[2]]]]==0 &.data[[vars[[3]]]]==0 & .data[[vars[[4]]]]==0 & .data[[vars[[5]]]]==0)) # see ?filter
# which categories are exclused from GSEA_reports.DF.formatted
missing.categories <- genes.count %>% dplyr::select(NAME) %>% filter(!NAME %in% GSEA_reports.DF.formatted$NAME)
# scaling for dendrogram calculation
GSEA_reports.DF.scaled <- t(scale(t(GSEA_reports.DF.formatted[-1]))) %>% as.data.frame()
# add rownames
GSEA_reports.DF.scaled <- GSEA_reports.DF.scaled %>% mutate(NAME=GSEA_reports.DF.formatted[,"NAME"]) %>% column_to_rownames(var="NAME")
# Run clustering
GSEA_reports.DF.scaled.dendro <- as.dendrogram(hclust(d = dist(x = GSEA_reports.DF.scaled)))
# Create dendrogram plot
dendro.plot <- ggdendrogram(data = GSEA_reports.DF.scaled.dendro, rotate = TRUE) + 
#  theme(axis.text.y = element_text(size = 6))
  theme(axis.text.y = element_blank())
print(dendro.plot)
#
dendro.y.NAME <- tibble(NAME=dendro_data(GSEA_reports.DF.scaled.dendro)$label$label) 
save(dendro.y.NAME,file=file.path("RNA","output","dendro.y.NAME.Rdata"))
if(modules=="fullmodule") {
  # adding missing categories to heatmap 
  GSEA_reports.DF.formatted2 <- GSEA_reports.DF.formatted %>% as_tibble() %>%   pivot_longer(cols=-1,names_to="tissue_LorD_set",values_to="-log10 FDR q-val") %>% bind_rows(tibble(NAME=rep(missing.categories$NAME,4),tissue_LorD_set=rep(vars[-1],each=dim(missing.categories)[1]),`-log10 FDR q-val`=0)) #%>% View() # missing categories in GSEA reports
## relevel
  GSEA_reports.DF.formatted2 <- GSEA_reports.DF.formatted2 %>% 
 mutate(NAME=factor(NAME,levels=c(as.vector(dendro.y.NAME$NAME),as.vector(missing.categories$NAME)))) 
  str(GSEA_reports.DF.formatted2)
  GSEA_reports.DF.formatted2 %>% View()
} else if(modules=="selected") {
  GSEA_reports.DF.formatted2 <- GSEA_reports.DF.formatted %>% as_tibble() %>%   pivot_longer(cols=-1,names_to="tissue_LorD_set",values_to="-log10 FDR q-val") %>% mutate(NAME=factor(NAME,levels=as.vector(dendro.y.NAME$NAME))) 
  str(GSEA_reports.DF.formatted2)
  GSEA_reports.DF.formatted2 %>% View()
}
# plot (base)
GSEA_reports.DF.plot <- GSEA_reports.DF.formatted2 %>% 
  ggplot(aes(x=tissue_LorD_set,y=NAME)) + geom_tile(aes(fill=`-log10 FDR q-val`),colour="black")  + scale_fill_gradient2(limit=c(0,3), high=muted("magenta")) # + facet_grid(~tissue) # not OK

# + scale_x_discrete(sec.axis=dup_axis())# not working
GSEA_reports.DF.plot <- GSEA_reports.DF.plot +     theme(axis.text.x=element_text(size=10,angle=90),
      axis.text.y=element_text(size=10),
      axis.title=element_text(size=10),
      axis.ticks = element_blank(),
      strip.text.y = element_text(angle=0),
      panel.background = element_rect(fill = "white",colour="black"),
      plot.title=element_text(size=20),
      axis.line=element_blank()) + 
labs(x="",y="",fill="-log10\n FDR")
GSEA_reports.DF.plot
##  gene number, custom category, etc. shape is growth increase/decrease (instead of UP/DOWN), plus correlation of modules to leaf_avg_std from modTrait.leaf_avg_std.RP2.cor.summary.voom5
# format tissue.module in leaf_avg_std.ML.data (correct?)
#leaf_avg_std.ML.data <- leaf_avg_std.ML.data %>% dplyr::rename(tissue.module=voom) %>% #mutate(module=str_to_upper(module)) %>% #mutate(NAME=str_remove(NAME,"voom_expression.e1and3.resid.exp_gt.")) %>% 
#  mutate(NAME=str_remove(NAME,".plusR500")) %>% unite(NAME,c(NAME,module),sep=".")

if(modules=="fullmodule") {
  genes.count2 <- genes.count  %>% # filter(NAME %in% dendro.y.NAME$NAME) %>%
  mutate(NAME=factor(NAME,levels=c(as.vector(dendro.y.NAME$NAME),as.vector(missing.categories$NAME))))  %>% separate(NAME,into=c("tissue2","module"),sep="\\.",remove=FALSE) %>% #View()
 arrange(desc(NAME)) %>% left_join(leaf_avg_std.ML.data,by=c("NAME"="tissue.module")) # %>%    mutate(sign=ifelse(PropVar_0.5_LOO >0,"positive","negative")) # adding leaf_avg_std.ML.data

} else if(modules=="selected") {
  genes.count2 <- genes.count  %>% # filter(NAME %in% dendro.y.NAME$NAME) %>%
  separate(NAME,into=c("tissue2","module"),sep="\\.",remove=FALSE) %>% #View()
 arrange(desc(NAME)) %>% inner_join(leaf_avg_std.ML.data,by=c("NAME"="tissue.module")) %>% mutate(NAME=factor(NAME,levels=as.vector(leaf_avg_std.ML.data$tissue.module))) # %>% mutate(sign=ifelse(PropVar_0.5_LOO >0,"positive","negative"))# adding leaf_avg_std.ML.data
}

# check genes.count2
print("genes.count2 after modules if/else")
print(genes.count2)
str(genes.count2)
print("dim(genes.count2)")
print(dim(genes.count2))
genes.count2 <- genes.count2 %>% mutate(y=seq(dim(genes.count2)[1],1)) # bottom is y=1, 
# draw gene number inside heatmap
    GSEA_reports.DF.plot.alt <- GSEA_reports.DF.plot + 
      # for tissue for modules
      geom_point(aes(x=-0.35,y=y,color=tissue2),stroke=3, size=.8,inherit.aes = FALSE, data=genes.count2) +
      scale_color_manual(name = "", values = c("root"="brown", "leaf"="lawngreen")) + 
      #scale_shape_manual(values=c("increase"=2,"decrease"=6)) +
      labs(shape="") + #scale_shape_identity(name="") + 
      # for gene number (="n")
  geom_point(aes(x=0,y=y,size=n),color = "red",  inherit.aes = FALSE, data=genes.count2) +
      # for PropVar_0.5_LOO (120820), changed for leaf_avg_std.ML.data = ElasticNet0.5_multiCV_Pct_var (031221)
      new_scale_fill() + # error
      geom_point(aes(x=0.35,y=y,fill=Pct_Var_Explained,shape=Effect), size=3,inherit.aes = FALSE, data=genes.count2)  +
      scale_fill_gradient(limit=c(0,10)) +
      scale_shape_manual(values=c("Increase"=24,"Decrease"=25)) + # trial to heve different shape 
    coord_cartesian(xlim = c(0,4)) 
    GSEA_reports.DF.plot.alt # needs to work on abs(FC.mean) scale

### combine heatmap and dendrogram by cowplot. Adjust legend.
   
GSEA_reports.DF.plot.alt <- GSEA_reports.DF.plot.alt + labs(title=NULL) + theme(
  legend.position = "top",legend.title=element_text(size=10),legend.text=element_text(size=10,angle=90),legend.key.height=unit(legend.height,"pt"),legend.key.width = unit(legend.width,"pt"),
  plot.margin = margin(0, 0, 0, 0, "pt")) # Top, right, bottom, left
dendro.plot <- dendro.plot + theme(
  axis.text.x = element_blank(),
  #plot.margin = margin(470, 0, 40, 0, "pt")) # Top, right, bottom, left. 
  plot.margin = margin(top.margin, 0, bottom.margin, 0, "pt")) # Top, right, bottom, left.
# adding dendrogram or not
if(dendrogram=="TRUE") { 
# combine plots
GSEA_reports.plot <- plot_grid(GSEA_reports.DF.plot.alt, dendro.plot,rel_widths=c(4,1)) # worked!
} else {
  GSEA_reports.plot <- GSEA_reports.DF.plot.alt
}
GSEA_reports.plot
return(GSEA_reports.plot)
}
```


# WGCNA modules vs custom categories
# load GO.ORA function (copy from "04b_Br.mbio.e3.subset.v3.)
# GOseq function for Brassica rapa (v3.0)
```{r eval=FALSE,error=TRUE}
# GOseq
library(ShortRead);library(goseq);library(GO.db);library("annotate")
# for ggplot heatmap
## uncompress gz file
system(paste("gunzip -c ",file.path("annotation","input","Brapa_genome_v3.0_cds.gz")," > ",file.path("annotation","input","Brapa_genome_v3.0_cds.fa")))
## read cDNA fasta file 
Bra.v3.0_cdna<-readDNAStringSet(file.path("annotation","input","Brapa_genome_v3.0_cds.fa")) # copied from /Volumes/data_work/Data8/NGS_related/Brassica_rapa_Upendra/G3
Bra.v3.0_cdna
## remove fasta file
system(paste("rm ",file.path("annotation","input","Brapa_genome_v3.0_cds.fa"),sep=""))

#
Br.v3.0anno.At.BLAST.highscore.modules2 <- read_csv(file.path("RNA","output","WGCNA.modules2.voom5.csv.gz"))

# GOseq function
GOseq.Brgo.v3.0.Atgoslim.BP.list.ORA2<-function(genelist,padjust=0.05,ontology="BP",custom.category.list=Brgo.v3.0anno.Atgoslim.BP.list,Br_cdna=Bra.v3.0_cdna) { # return GO enrichment table, padjus, padjust=0.05. 
  print(genelist)
  bias<-nchar(Br_cdna)
  names(bias)<-names(Br_cdna)
  # convert list of genelist into vector
  genelist <- genelist %>% unlist()
  TF<-(names(bias) %in% genelist)*1
  names(TF)<-names(bias)
  #print(TF)
  pwf<-nullp(TF,bias.data=bias,plot.fit=FALSE)
  #print(pwf$DEgenes)
  GO.pval <- goseq(pwf,gene2cat=custom.category.list,use_genes_without_cat=TRUE) # format became different in new goseq version (021111). Does not work (042716)
  #GO.pval <- goseq(pwf,gene2cat=Brgo.DF3,use_genes_without_cat=TRUE) # format became different in new goseq version (021111)
  
  #head(GO.pval) 
  if(ontology=="BP") {
    GO.pval2<-subset(GO.pval,ontology=="BP")
  } else if(ontology=="CC") {
    GO.pval2<-subset(GO.pval,ontology=="CC")
  } else if(ontology=="MF") {
    GO.pval2<-subset(GO.pval,ontology=="MF")
  } else {
    GO.pval2<-GO.pval
  }
    
  GO.pval2$over_represented_padjust<-p.adjust(GO.pval2$over_represented_pvalue,method="BH")
  if(GO.pval2$over_represented_padjust[1]>padjust) return("no enriched GO")
  else {
    enriched.GO<-GO.pval2[GO.pval2$over_represented_padjust<padjust,] 
    print("enriched.GO is")
    print(enriched.GO)
    
    ## write Term and Definition 
    if(ontology=="BP"|ontology=="CC"|ontology=="MF") {
    for(i in 1:dim(enriched.GO)[1]) {
      if(is.null(Term(GOTERM[enriched.GO[i,"category"]]))) {next} else {
      #enriched.GO$Term[i]<-Term(GOTERM[[enriched.GO[i,"category"]]])
      enriched.GO$Definition[i]<-Definition(GOTERM[[enriched.GO[i,"category"]]])
      }
    }
    }
    enriched.GO.tb <- as_tibble(enriched.GO)
    print("As tibble enriched.GO is")
    print(enriched.GO.tb)
    return(enriched.GO.tb)
  }
}
#
head(Bra.v3.0_cdna)
# length(bias) # 44239 > 45019 where the bias come from?
#  bias.data vector must have the same length as DEgenes vector!
```


# GOseq with custom categories (Sep 30, 2020; Oct 28, 2020)
```{r eval=FALSE}
# annotation
Br.v3.0anno.At.BLAST.highscore.modules2 <- read_csv(file.path("RNA","output","WGCNA.modules2.voom5.csv.gz"))
# custom categories list
load(file.path("annotation","output","GSEA_cust_rootshoot_updown2.list.Rdata"))
#modTrait.leaf_avg_std.P2.summary.FDR0.25 %>% View()
# all modules
GOseq.temp <- Br.v3.0anno.At.BLAST.highscore.modules2 %>% ungroup() %>% dplyr::select(name,root.voom5.module,leaf.voom5.module) %>%
pivot_longer(cols=-1,names_to = "voom")  %>% drop_na(value) %>% unite(voom.module,c("voom","value")) %>%  split(.$voom.module) %>% map(~GOseq.Brgo.v3.0.Atgoslim.BP.list.ORA2(genelist=.$name,custom.category.list=GSEA_cust_rootshoot_updown2.list,ontology="none")) # ,custom.category.list=GSEA_cust_rootshoot_updown2.list
GOseq.temp %>% enframe() %>% unnest(value) %>% write_csv(file.path("RNA","output","WGCNA.voom5.all.modules.customcat.GOseq.csv"))

# check results on "voom_expression.e1and3.resid.exp_only.root_plum1", which has "iron" in GOseq? which modules?
GOseq.temp2 <- Br.v3.0anno.At.BLAST.highscore.modules2 %>% ungroup() %>% dplyr::select(name,root.voom5.module,leaf.voom5.module) %>%
pivot_longer(cols=-1,names_to = "voom")  %>% drop_na(value) %>% unite(voom.module,c("voom","value")) %>% filter(voom.module=="root.voom5.module_plum1") %>% 
  split(.$voom.module) %>% map(~GOseq.Brgo.v3.0.Atgoslim.BP.list.ORA2(genelist=.$name,custom.category.list=GSEA_cust_rootshoot_updown2.list,ontology="none")) # OK. Good. Fe categoreis enriched
```

# GOseq of all WGCNA modules (leaf.voom5, root.voom5 version) with custom categories 
```{r}
GOseq.temp <- Br.v3.0anno.At.BLAST.highscore.modules2 %>% ungroup() %>% dplyr::select(name,root.voom5.module,leaf.voom5.module) %>%
pivot_longer(cols=-1,names_to = "voom") %>% drop_na(value) %>% unite(voom.module,c("voom","value")) %>% split(.$voom.module) %>% map(~GOseq.Brgo.v3.0.Atgoslim.BP.list.ORA2(genelist=.$name,custom.category.list=GSEA_cust_rootshoot_updown2.list,ontology="none")) # ,custom.category.list=GSEA_cust_rootshoot_updown2.list
GOseq.temp %>% enframe() %>% unnest(value) %>% write_csv(file.path("RNA","output","WGCNA.voom5.all.modules.customcat.GOseq.csv"))

```

# Signature of WGCNA modules by looking enrichement of custom categories
```{r}
WGCNA.voom5.all.modules.customcat.GOseq <- read_csv(file.path("RNA","output","WGCNA.voom5.all.modules.customcat.GOseq.csv"))
#
# genes.count # does not use
# colnames(GSEA_cust.root)
# colnames(GSEA_cust.leaf)
# GSEA_reports.DF.formatted2 

WGCNA.voom5.all.modules.leaf_avg_std.correlated.customcat.GOseq <- WGCNA.voom5.all.modules.customcat.GOseq %>% mutate(name=str_replace(name,"_",".")) %>% filter(name %in% c(colnames(GSEA_cust.root) %>% str_remove(".voom5.module"),colnames(GSEA_cust.leaf))) %>% dplyr::rename(NAME=name) 
                         
WGCNA.voom5.all.modules.leaf_avg_std.correlated.customcat.GOseq %>% View()
str(WGCNA.voom5.all.modules.leaf_avg_std.correlated.customcat.GOseq)
```
# WGCNA module ORA heatmap
```{r}
WGCNA.voom5.all.modules.leaf_avg_std.correlated.customcat.GOseq.2 <- WGCNA.voom5.all.modules.leaf_avg_std.correlated.customcat.GOseq  %>%  mutate(category2=
  str_to_upper(category) %>%
  str_replace_all("KAILASAM2019", "KAILASAM_2019") %>%
 str_replace_all("MINUSPI_1D","MINUSPI-1D") %>%
  str_replace_all("MINUSPI_3D","MINUSPI-3D") %>%
  str_replace_all("AARABI2016","AARABI_2016") %>%
  str_replace_all("KIM2019","KIM_2019") %>%
  str_replace_all("WANG2003","WANG_2003") %>%
   str_replace_all("COLDKILIAN","COLD_KILIAN") %>%
  str_replace_all("BL_SEEDLINGS","plusBL_Nemhauser_2006_seedlings") %>%
  str_replace_all("IAA_SEEDLINGS","plusIAA_Nemhauser_2006_seedlings") ) %>% 
  separate(category2,into=c("category3","au","yr","tissue3","FC","up_down"),sep="_") %>% # View()
  mutate(category3=category3 %>% 
           str_replace_all("MINUSMG","MG-") %>%
           str_replace_all("PLUSMG","MG+") %>%
           str_replace_all("MINUSFE","FE-") %>%
    str_replace_all("PLUSFE","FE_PLUS") %>%
  str_replace_all("MINUSPI-1D","PI-1D-") %>%
    str_replace_all("PLUSPI-1D","PI-1D+") %>%
  str_replace_all("MINUSPI-3D","PI-3D+") %>%
    str_replace_all("PLUSPI-3D","PI-3D+") %>%
  str_replace_all("MINUSN","N-") %>%
  str_replace_all("PLUSN","N+") %>%
  str_replace_all("MINUSMN","MN-") %>%
  str_replace_all("PLUSMN","MN+") %>% 
  str_replace_all("MINUSP","P-") %>% 
    str_replace_all("PLUSP","P+") %>% 
    str_replace_all("PLUSIAA","IAA+") %>% 
    str_replace_all("PLUSAL","AL+") %>%
    str_replace_all("MINUSCA","CA-") %>% 
      str_replace_all("PLUSCA","CA+") %>%
    str_replace_all("MINUSK","K-") %>% 
    str_replace_all("PLUSK","K+") %>% 
    str_replace_all("MINUSS","S-") %>% 
    str_replace_all("PLUSS","S+")) %>%
  drop_na(category3) %>%
  unite(category3.au,c("category3","au"),remove=FALSE) %>% 
  mutate(au2 = str_replace_all(au,"KAILASAM","(1)") %>% 
           str_replace_all("KIM","(2)") %>% 
           str_replace_all("NISHIDA","(3)") %>%
           str_replace_all("NIU","(4)") %>%
           str_replace_all("KILIAN","(5)") %>%
           str_replace_all("RODRIGUEZ-CELMA","(6)") %>%
           str_replace_all("LIU","(7)") %>%
           str_replace_all("PENG","(8)") %>%
           str_replace_all("AARABI","(9)") %>%
           str_replace_all("LIGABA-OSENA","(10)") %>%
           str_replace_all("NEMHAUSER","(11)") %>%
           str_replace_all("WANG","(12)")) %>%
  unite(category3.au.up_down,c("category3","up_down","au2"),sep=" ",remove=FALSE) 
# check
WGCNA.voom5.all.modules.leaf_avg_std.correlated.customcat.GOseq.2 %>% View()
# arbitrally convert value of 0 into 10^-10 for scale visualization
WGCNA.voom5.all.modules.leaf_avg_std.correlated.customcat.GOseq.2[WGCNA.voom5.all.modules.leaf_avg_std.correlated.customcat.GOseq.2$over_represented_padjust < 10^-10,"over_represented_padjust"] <- 10^-10
# select modules
WGCNA.voom5.all.modules.leaf_avg_std.correlated.customcat.GOseq.2 %>% View()
WGCNA.voom5.all.modules.leaf_avg_std.correlated.customcat.GOseq.2.selected <- WGCNA.voom5.all.modules.leaf_avg_std.correlated.customcat.GOseq.2 %>% mutate(NAME=str_remove(NAME,".voom5.module")) %>%   inner_join(ElasticNet0.5_multiCV_Pct_var, by=c("NAME"="tissue.module")) %>% arrange(Pct_Var_Explained) %>% drop_na(Pct_Var_Explained) #%>% View() # sorting modules
# which categories are exclused from WGCNA.voom5.all.modules.leaf_avg_std.correlated.customcat.GOseq.2.selected
missing.categories <- ElasticNet0.5_multiCV_Pct_var %>% dplyr::select(tissue.module) %>% filter(!tissue.module %in% WGCNA.voom5.all.modules.leaf_avg_std.correlated.customcat.GOseq.2.selected$NAME) %>% dplyr::rename(NAME=tissue.module)
# format data to add missing categories to heatmap 
WGCNA.voom5.all.modules.leaf_avg_std.correlated.customcat.GOseq.2.selected2 <- WGCNA.voom5.all.modules.leaf_avg_std.correlated.customcat.GOseq.2.selected %>% dplyr::select(NAME,category3.au.up_down,over_represented_padjust,tissue3)  
# names of category3.au.up_down
num.category3.au.up_down <- names(table(WGCNA.voom5.all.modules.leaf_avg_std.correlated.customcat.GOseq.2.selected2$category3.au.up_down))
# adding missing categories to heatmap
WGCNA.voom5.all.modules.leaf_avg_std.correlated.customcat.GOseq.2.selected2 <- 
WGCNA.voom5.all.modules.leaf_avg_std.correlated.customcat.GOseq.2.selected2 %>% bind_rows(tibble(NAME=rep(missing.categories$NAME,length(num.category3.au.up_down)),category3.au.up_down=rep(num.category3.au.up_down,each=dim(missing.categories)[1]),over_represented_padjust=1,tissue3="ROOT")) #%>% View() # missing categories in GSEA reports. tissue3="root" is fake
# check this is what I want
WGCNA.voom5.all.modules.leaf_avg_std.correlated.customcat.GOseq.2.selected2 %>% group_by(NAME,category3.au.up_down) %>% summarize(n()) %>% View() # none, why?

# relevel using dendrogram in GSEA plot
load(file.path("RNA","output","dendro.y.NAME.Rdata"))
WGCNA.voom5.all.modules.leaf_avg_std.correlated.customcat.GOseq.2.selected2 <-  WGCNA.voom5.all.modules.leaf_avg_std.correlated.customcat.GOseq.2.selected2 %>%
  mutate(NAME=factor(NAME,levels=dendro.y.NAME$NAME)) 
str(WGCNA.voom5.all.modules.leaf_avg_std.correlated.customcat.GOseq.2.selected2)

# plot 
WGCNA.voom5.all.modules.customcat.GOseq.plot.selected2 <- ggplot(WGCNA.voom5.all.modules.leaf_avg_std.correlated.customcat.GOseq.2.selected2, aes(x=category3.au.up_down,y=NAME)) + geom_tile(aes(fill=-log10(over_represented_padjust)),colour="white") + scale_fill_gradient2(limit=c(0,10), high=muted("magenta")) + facet_grid(.~tissue3,scales="free_x",space="free")

WGCNA.voom5.all.modules.customcat.GOseq.plot.selected2 <- WGCNA.voom5.all.modules.customcat.GOseq.plot.selected2 + theme(axis.text.x=element_text(size=8,angle=90,hjust=1),
                                      axis.text.y=element_text(size=8),
                                      axis.title=element_text(size=8),
                                      axis.ticks = element_blank(),
                                      strip.text.y = element_text(angle=0),
                                      strip.text.x = element_text(size=8,angle=90),
                                      panel.background = element_rect(fill = "white",colour="black"),
                                      plot.title=element_text(size=14),
                                      axis.line=element_blank()) + 
      labs(x="",y="",fill="-log10\n FDR",title=" ")
    WGCNA.voom5.all.modules.customcat.GOseq.plot.selected2
ggsave(path=file.path("RNA","output"),filename="WGCNA.voom5.leaf_avg_std.correlated.modules.customcat.GOseq.4.selected.plot.png",height=8,width=11) 

```

# combine GSEA plot and WGCNA module ORA plot 
```{r}
# format GSEA plot
plot.GSEA <- GSEA_reports.DF.plot.dendrogram.v9.WGCNAmodule.notitle.ML(data = GSEA_reports.DF.MLselected,
                                            GSEA.FDR.cutoff = cutoff,
                                            top.margin=290,
                                            bottom.margin=30,
                                            dendrogram="FALSE",
                                            legend.height=30,
                                            legend.width=10,
                                            modules="fullmodule") 
#plot.GSEA  +  guides(fill = guide_legend(nrow=2)) # this is not what I want
plot.GSEA <- plot.GSEA + theme(legend.position="left",legend.key.height=unit(10,"pt"),legend.text=element_text(size=10,angle=0),plot.margin = margin(74, 0, 50, 0, "pt")) 
ggsave(plot.GSEA,filename="GSEA.RNAseq.plot.ML.full2.png",path=file.path("RNA","output"),width=8,height=15)
# only ML selected modules
plot.GSEA <- GSEA_reports.DF.plot.dendrogram.v9.WGCNAmodule.notitle.ML(data = GSEA_reports.DF.MLselected,
                                            GSEA.FDR.cutoff = cutoff,
                                            top.margin=290,
                                            bottom.margin=30,
                                            dendrogram="FALSE",
                                            legend.height=30,
                                            legend.width=10,
                                            modules="selected") 
#plot.GSEA  +  guides(fill = guide_legend(nrow=2)) # this is not what I want
plot.GSEA <- plot.GSEA + theme(legend.position="left",legend.key.height=unit(10,"pt"),legend.text=element_text(size=10,angle=0),plot.margin = margin(75, 0, 35, 0, "pt")) 
ggsave(plot.GSEA,filename="GSEA.RNAseq.plot.ML.selected.png",path=file.path("RNA","output"),width=6,height=8)
# format WGCNA ORA plots
WGCNA.voom5.all.modules.customcat.GOseq.plot.selected2 <- WGCNA.voom5.all.modules.customcat.GOseq.plot.selected2 + theme(plot.margin = margin(0, 0, 0, 0, "pt"),legend.position = "right",legend.key.height=unit(30,"pt"),axis.text.y=element_blank(),axis.text.x=element_text(size=10,hjust=1)) + # Top, right, bottom, left.)
  labs(title="") 
# combine two plots
library(cowplot)
plot.combined <- plot_grid(plot.GSEA,WGCNA.voom5.all.modules.customcat.GOseq.plot.selected2,rel_widths=c(1,1),labels=c("A","B"))
#plot.combined
ggsave(plot.combined,path=file.path("RNA","output"),filename="WGCNA.voom5.leaf_avg_std.correlated.modules.ML.customcat.GSEA.combined2.png",height=8,width=11)
# wider version
plot.combined <- plot_grid(plot.GSEA,WGCNA.voom5.all.modules.customcat.GOseq.plot.selected2,rel_widths=c(3,4),labels=c("A","B"))
#plot.combined
ggsave(plot.combined,path=file.path("RNA","output"),filename="WGCNA.voom5.leaf_avg_std.correlated.modules.customcat.MLGSEA.combined2.wider.png",height=8,width=15)

```

# GOseq of all WGCNA modules (leaf.voom5, root.voom5 version) with custom categories  (copied from "10_WGCNA_including_R500.Rmd")


# GOseq ORA prep
# load Brgo.v3.0anno.Atgoslim.BP.list for GOseq analysis (GO terms) (using split() and map() to avoid long loop)
```{r eval=TRUE,error=TRUE}
load(file.path("annotation","output","Brgo.v3.0anno.Atgoslim.BP.list.Rdata"))
```

# GOseq of all WGCNA modules (leaf.voom5, root.voom5 version) with GO terms
```{r}
#Br.v3.0anno.At.BLAST.highscore.modules2 <- read_csv(file.path("RNA","output","WGCNA.modules2.voom5.csv.gz"))
# stats of modules
Br.v3.0anno.At.BLAST.highscore.modules2 %>% group_by(root.voom5.module) %>% summarise(root.voom5.module.num=n()) %>% dim() # 41 2
Br.v3.0anno.At.BLAST.highscore.modules2 %>% group_by(leaf.voom5.module) %>% summarise(leaf.voom5.module.num=n()) %>% dim() # 66 2
GOseq.temp <- Br.v3.0anno.At.BLAST.highscore.modules2 %>% ungroup() %>% dplyr::select(name,root.voom5.module,leaf.voom5.module) %>%
pivot_longer(cols=-1,names_to = "voom") %>% drop_na(value) %>% unite(voom.module,c("voom","value")) %>% split(.$voom.module) %>% map(~GOseq.Brgo.v3.0.Atgoslim.BP.list.ORA2(genelist=.$name)) # ,custom.category.list=GSEA_cust_rootshoot_updown2.list
GOseq.temp %>% enframe() %>% unnest(value) %>% mutate(name = str_replace(name,".voom5.module_",".")) %>% write_csv(file.path("RNA","output","WGCNA.voom5.all.modules.GOseq.csv"))

```

# GO ORA with custom categories
```{r}
load(file.path("annotation","output","GSEA_cust_rootshoot_updown2.list.Rdata"))
GOseq.temp <- Br.v3.0anno.At.BLAST.highscore.modules2 %>% ungroup() %>% dplyr::select(name,root.voom5.module,leaf.voom5.module) %>%
pivot_longer(cols=-1,names_to = "voom") %>% drop_na(value) %>% unite(voom.module,c("voom","value")) %>% split(.$voom.module) %>% map(~GOseq.Brgo.v3.0.Atgoslim.BP.list.ORA2(genelist=.$name,custom.category.list=GSEA_cust_rootshoot_updown2.list,ontology="none")) # ,custom.category.list=GSEA_cust_rootshoot_updown2.list
GOseq.temp %>% enframe() %>% unnest(value) %>% mutate(name=str_replace(name,".voom5.module_",".")) %>%  write_csv(file.path("RNA","output","WGCNA.voom5.all.modules.customcat.GOseq.csv"))

```

# Only GO ORA Table for growth correlated modules
* needs to fix "ElasticNet0.5_multiCV_Pct_var" related part (031021)
```{r}
WGCNA.voom5.all.modules.GOseq <- read_csv(file.path("RNA","output","WGCNA.voom5.all.modules.GOseq.csv"))
WGCNA.voom5.all.modules.GOseq %>% dplyr::select(name)

ElasticNet0.5_multiCV_Pct_var  %>% inner_join(WGCNA.voom5.all.modules.GOseq,by=c("tissue.module"="name")) %>% separate(tissue.module,into=c("tissue","module")) %>% dplyr::select(tissue,module,category,over_represented_pvalue,over_represented_padjust,term) %>% drop_na(category) %>% #View() 
  write_csv(file.path("RNA","output","WGCNA.voom5.growth.related.modules.GOseq.csv"))
```

