---
title: "Eigen Gene Treatment Tests"
output: html_notebook
author: "Julin Maloof"
editor_options: 
  chunk_output_type: inline
---

This is an alternative approach to look for modules that are responsive to treatment.  Do a t-test or equivalent on eigen gene expression levels split by treatment.

```{r}
library(tidyverse)
```

## load sample info

### exp1
```{r}
exp1 <- readxl::read_excel("../input/wy001_RNAseq.xlsx") %>%
  filter(time_point=="wk2", autoclave %in% c("dead", "live")) %>%
  mutate(sampleID=str_c(genotype, 
                        str_pad(block, width = 2, pad = "0"),
                        str_pad(pot, width = 3, pad = "0"),
                        sep="_")) %>%
  select(sampleID, trt=autoclave)
exp1
```
### exp3 

```{r}
exp3 <- read_csv("../input/wy003-metadata.csv") %>%
  filter(CROWDING=="CR") %>%
  select(sampleID=`SAMPLE ID`, trt=TREATMENT)
exp3
```

combine it
```{r}
sample_info <- bind_rows(exp1, exp3)
sample_info
```

# roots

load root modules
```{r}
root.voom5 <- readr::read_tsv(file.path("..","output","voom_expression.e1and3.resid.exp_gt.root.plusR500.txt.gz"))
load("../output/voom_expression.e1and3.resid.exp_gt.root.plusR500.signedhybrid.txt.gz.WGCNA.softthresh10.signedhybrid.RData")

rootMEs <- MEs %>% as_tibble() %>%
  rename_all(~ str_remove(., "ME")) %>%
  rename_all(~ str_c("root.", .)) %>%
  mutate(sampleID=colnames(root.voom5)[-1],
         sampleID=str_remove(sampleID, "wyo_root_"),
         sampleID=str_remove(sampleID, "_S[0-9]{1,3}$")) %>%
  select(sampleID, everything())
rootMEs
rm(MEs)
```

```{r}
root_results <- inner_join(sample_info, rootMEs) %>%
  pivot_longer(starts_with("root"), names_to = "module", values_to = "eigen_value") %>%
  nest(data = c(sampleID, trt, eigen_value)) %>%
  mutate(lm=map(data, ~lm(eigen_value ~ trt, data=.)),
         lmglance=map(lm, broom::glance),
         lmtidy=map(lm, broom::tidy)) %>%
  #unnest(lmglance) %>% 
  #select(1:lmtidy, lm.p.value=p.value) %>% 
  #mutate(lmfdr=p.adjust(lm.p.value, method="fdr")) %>%
  unnest(lmtidy)

root_results <- root_results %>% 
  filter(term=="trtlive") %>%
  arrange(p.value) %>% 
  mutate(fdr=p.adjust(p.value, method="fdr")) %>%
  select(module, term, estimate, p.value, fdr)

root_results

write_csv(root_results, "../output/root_eigen_microbes_lm.csv")
```

# leaf

load leaf modules
```{r}
leaf.voom5 <- readr::read_tsv(file.path("..","output","voom_expression.e1and3.resid.exp_gt.leaf.plusR500.txt.gz"))
load("../output/voom_expression.e1and3.resid.exp_gt.leaf.plusR500.signedhybrid.txt.gz.WGCNA.softthresh10.signedhybrid.RData")
```

# checking module names in "MEs" (Kazu: July 18 - 19, 2021
```{r}
# number of modules
tibble(dynamicColors) %>% count(dynamicColors) # 68
dim(MEs) # 31, why different from dynamiccolors?
tibble(moduleColors) %>% count(moduleColors) # 31. Instead of dynamicColors, I should have used "moduleColors"... 
# module names in MEs are identical to ones in moduleColors?
MEs.colnames <- colnames(MEs) %>% str_remove("ME")
moduleColors.modules <- tibble(moduleColors) %>% count(moduleColors) %>% select(moduleColors) %>% as_vector()
MEs.colnames %in% moduleColors.modules # all TRUE. 
# the end of checking module names in "MEs" 
```

```{r}
#
leafMEs <- MEs %>% as_tibble() %>%
  rename_all(~ str_remove(., "ME")) %>%
  rename_all(~ str_c("leaf.", .)) %>%
  mutate(sampleID=colnames(leaf.voom5)[-1],
         sampleID=str_remove(sampleID, "wyo_leaf_"),
         sampleID=str_remove(sampleID, "_S[0-9]{1,3}$")) %>%
  select(sampleID, everything())
leafMEs
rm(MEs)
```

```{r}
leaf_results <- inner_join(sample_info, leafMEs) %>%
  pivot_longer(starts_with("leaf"), names_to = "module", values_to = "eigen_value") %>%
  nest(data = c(sampleID, trt, eigen_value)) %>%
  mutate(lm=map(data, ~lm(eigen_value ~ trt, data=.)),
         lmglance=map(lm, broom::glance),
         lmtidy=map(lm, broom::tidy)) %>%
  unnest(lmtidy)

leaf_results <- leaf_results %>% 
  filter(term=="trtlive") %>%
  arrange(p.value) %>% 
  mutate(fdr=p.adjust(p.value, method="fdr")) %>%
  select(module, term, estimate, p.value, fdr)

leaf_results

write_csv(leaf_results, "../output/leaf_eigen_microbes_lm.csv")
```
# combine with leaf.length and GO results

```{r}
combined <- rbind(leaf_results, root_results) %>%
  select(module, term, microbe.estimate=estimate, microbe.fdr=fdr)
combined
```


```{r}
leaf.length <- read_csv("../output/ElasticNet0.5_multiCV_Pct_var.signed.hybrid.csv") %>%
  mutate(module=str_c(Module_Organ,".",Module_Name)) %>%
  select(module, leaf_length_pct_var=Pct_Var_Explained, leaf_length_effect=Effect)
head(leaf.length)
```

```{r}
combined <- full_join(combined, leaf.length, by="module")
combined
```

GO terms
```{r}
GO <- read_csv("../output/WGCNA.voom5.signedhybrid.all.moduleColors.GOseq.csv")
max(GO$over_represented_padjust,na.rm = TRUE)
GO
```

Get names to match
```{r}
GO <- GO %>% mutate(module=str_remove(name,"voom5.module.signedhybrid_"))
unique(GO$module)
```

Consolidate GOs to single line per module
```{r}
GO_summary <- GO %>% select(module, term)%>%
  mutate(term=ifelse(is.na(term), "no enriched GO", term)) %>%
  group_by(module) %>%
  summarize(terms= str_c(term, collapse="; "))

GO_summary
```

```{r}
combined <- full_join(combined, GO_summary)
combined
```

```{r}
write_csv(combined, "../output/voom5.signedhybrid.GO_leaf_length_microbe.csv")
```

