---
title: "custom_categories_consolidate1_forV5_WGCNA_ORA"
author: "Kazu"
date: "5/9/2021"
output: html_document
editor_options: 
  chunk_output_type: console
---

Summary

History
* copied from "06_f_cor_table.Rmd"
* add GSEA_cust_rootshoot_updown3.FDR0.001.list (051221)
*  Expand DEGs with FC threshold for all custom categoreis
* root data including minusN_Peng_2007 and plusIAA_Nembauser_2006
* adding Nishida_2017 to root categories (070720)
* making top20, top50, and top100 for (a) among root/leaf categories, (b) among minusX_up, (c) among minusX_down
* More stringent FDR for custom category ORA of root.ivory module (050921)
* "In only" approach for custom ORA (051821)
* Fair "In only" approach for custom ORA using consolidated categoreis (052821)
* Using consolidated categories in InX selection process and ORA and heatmap with the consolidated categories 
* Another version (consolidate2) is using the consolidated categoreis to generate InX, but use the original categories for ORA.
* plot heatmap of InX comparison for root.ivory module
* and more modules.
* Making v4 custom categories and use it for this script
* Modify for updated plusN_Wang2003 using v5 (080621-081021)
* Save as "custom_categories_consolidate1_forV5"
* heatmap coloaring by soil FC sign (positive vs negative) (Sep 2021)
* Cleaning up scripts (Jan 25, 2021)
* split scripts into two (Jan 27, 2021): 
** custom_categories_consolidate1_forV5.Rmd for consolidating custom categoreis.
** custom_categories_consolidate1_forV5_WGCNA_module_ORA.Rmd
*** WGCNA module ORA with InX
*** 

To Do
* Cleaning up scripts more
* having the same x-axis and y-axis for WGCNA heatmap

#### making root + shoot category with up and down and
# prep
```{r}
rm(list=ls())
knitr::opts_chunk$set(echo = TRUE,error=TRUE)
library(tidyverse);library(GGally);library(readxl)
library(readr)
library(stringr)
# https://bioconductor.org/packages/devel/workflows/vignettes/maEndToEnd/inst/doc/MA-Workflow.html
# functions for expression pattern
#source("../../../tools/Expression_pattern_graph.R",chdir=TRUE)
library(scales) # for muted function
library(ggdendro) # for dendrogram
library(cowplot)
library(ggnewscale)
```



reformat for for ORA (051821)
# Signature of WGCNA modules by looking enrichement of custom categories
```{r eval=TRUE,error=TRUE}
# GOseq
library(ShortRead);library(goseq);library(GO.db);library("annotate")
# for ggplot heatmap
## uncompress gz file
system(paste("gunzip -c ",file.path("annotation","input","Brapa_genome_v3.0_cds.gz")," > ",file.path("annotation","input","Brapa_genome_v3.0_cds.fa")))
## read cDNA fasta file 
Bra.v3.0_cdna<-readDNAStringSet(file.path("annotation","input","Brapa_genome_v3.0_cds.fa")) # copied from /Volumes/data_work/Data8/NGS_related/Brassica_rapa_Upendra/G3
Bra.v3.0_cdna
## remove fasta file
system(paste("rm ",file.path("annotation","input","Brapa_genome_v3.0_cds.fa"),sep=""))

# GOseq function
GOseq.Brgo.v3.0.Atgoslim.BP.list.ORA2<-function(genelist,padjust=0.05,ontology="BP",custom.category.list=Brgo.v3.0anno.Atgoslim.BP.list,Br_cdna=Bra.v3.0_cdna) { # return GO enrichment table, padjus, padjust=0.05. 
  print(genelist)
  bias<-nchar(Br_cdna)
  names(bias)<-names(Br_cdna)
  # convert list of genelist into vector
  genelist <- genelist %>% unlist()
  TF<-(names(bias) %in% genelist)*1
  names(TF)<-names(bias)
  #print(TF)
  pwf<-nullp(TF,bias.data=bias,plot.fit=FALSE)
  #print(pwf$DEgenes)
  GO.pval <- goseq(pwf,gene2cat=custom.category.list,use_genes_without_cat=TRUE) # format became different in new goseq version (021111). Does not work (042716)
  #GO.pval <- goseq(pwf,gene2cat=Brgo.DF3,use_genes_without_cat=TRUE) # format became different in new goseq version (021111)
  
  #head(GO.pval) 
  if(ontology=="BP") {
    GO.pval2<-subset(GO.pval,ontology=="BP")
  } else if(ontology=="CC") {
    GO.pval2<-subset(GO.pval,ontology=="CC")
  } else if(ontology=="MF") {
    GO.pval2<-subset(GO.pval,ontology=="MF")
  } else {
    GO.pval2<-GO.pval
  }
    
  GO.pval2$over_represented_padjust<-p.adjust(GO.pval2$over_represented_pvalue,method="BH")
  if(GO.pval2$over_represented_padjust[1]>padjust) return("no enriched GO")
  else {
    enriched.GO<-GO.pval2[GO.pval2$over_represented_padjust<padjust,] 
    print("enriched.GO is")
    print(enriched.GO)
    
    ## write Term and Definition 
    if(ontology=="BP"|ontology=="CC"|ontology=="MF") {
    for(i in 1:dim(enriched.GO)[1]) {
      if(is.null(Term(GOTERM[enriched.GO[i,"category"]]))) {next} else {
      enriched.GO$Term[i]<-Term(GOTERM[[enriched.GO[i,"category"]]])
      enriched.GO$Definition[i]<-Definition(GOTERM[[enriched.GO[i,"category"]]])
      }
    }
    }
    enriched.GO.tb <- as_tibble(enriched.GO)
    print("As tibble enriched.GO is")
    print(enriched.GO.tb)
    return(enriched.GO.tb)
  }
}
#
head(Bra.v3.0_cdna)
# length(bias) # 44239 > 45019 where the bias come from?
#  bias.data vector must have the same length as DEgenes vector!
```

# In01.forGSEA
```{r}
# gene list
Br.v3.0anno.At.BLAST.highscore.moduleColors.signedhybrid <- read_csv(file.path("RNA","output","WGCNA.signedhybrid.moduleColors.voom5.csv.gz"))
# load filtered.custom.data.consolidated.numericL
load(file.path("RNA","output","filtered.custom.v5.data.consolidated.numericL.Rdata"))
In01.consolidated.forGSEA <- filtered.custom.data.consolidated.numericL %>%
  filter(totalSetN <= 1, presentInSet ==1) %>%
  dplyr::select(genes, set) %>%
  mutate(id=1:nrow(.)) %>% 
  pivot_wider(names_from = set, values_from = genes, values_fill = list(genes = NA)) %>%
  dplyr::select(-id) %>%
  mutate_all(sort, na.last=TRUE) 
In01.consolidated.forGSEA <- In01.consolidated.forGSEA[apply(In01.consolidated.forGSEA,1,function(x) !all(is.na(x))),]
In01.consolidated.forGSEA %>% View()
### for ORA　(copied from AnnotateV3.0_ORA_categories.Rmd)
# trial
In01.consolidated.forGSEA %>% dplyr::select(1) %>% pivot_longer(1)
# 
#In01.consolidated.forGSEA %>% map(~pivot_longer(cols=name)) # does not work
##
In01.consolidated.forGSEA.list <- list()
for(i in 1:(dim(In01.consolidated.forGSEA)[2]-1)) {
In01.consolidated.forGSEA.list[[i]] <- In01.consolidated.forGSEA %>% dplyr::select(i) %>% pivot_longer(1) %>% dplyr::rename(category=name)
}
#
In01.consolidated.forGSEA.DF <- In01.consolidated.forGSEA.list %>% enframe() %>% unnest(value) %>% drop_na(value)# does not work if I did not rename "category" columns
# convert into list by gene name
In01.consolidated.forORA.list2 <- In01.consolidated.forGSEA.DF %>%split(.$value) %>%   map(~dplyr::select(.,category)) %>% map(~as_vector(.))
save(In01.consolidated.forORA.list2,file=file.path("annotation","output","In01.consolidated.forORA.list2.v5.Rdata"))
# ORA
# all modules
GOseq.temp <- Br.v3.0anno.At.BLAST.highscore.moduleColors.signedhybrid %>% ungroup() %>% dplyr::select(name,root.voom5.module.signedhybrid,leaf.voom5.module.signedhybrid) %>%
pivot_longer(cols=-1,names_to = "voom")  %>% drop_na(value) %>% unite(voom.module,c("voom","value")) %>%  split(.$voom.module) %>% map(~GOseq.Brgo.v3.0.Atgoslim.BP.list.ORA2(genelist=.$name,custom.category.list=In01.consolidated.forORA.list2,ontology="none")) # ,custom.category.list=GSEA_cust_rootshoot_updown2.list
GOseq.temp %>% enframe() %>% unnest(value) %>% write_csv(file.path("RNA","output","WGCNA.voom5.signedhybrid.all.moduleColors.customcatv5.In01.consolidated.FDR0.001.GOseq.csv"))

```

# In02orless.forGSEA
```{r}
In02orless.consolidated.forGSEA <- filtered.custom.data.consolidated.numericL %>%
  filter(totalSetN <= 2, presentInSet ==1) %>%
  dplyr::select(genes, set) %>%
  mutate(id=1:nrow(.)) %>% 
  pivot_wider(names_from = set, values_from = genes, values_fill = list(genes = NA)) %>%
  dplyr::select(-id) %>%
  mutate_all(sort, na.last=TRUE) 
In02orless.consolidated.forGSEA <- In02orless.consolidated.forGSEA[apply(In02orless.consolidated.forGSEA,1,function(x) !all(is.na(x))),]
In02orless.consolidated.forGSEA %>% View()
### for ORA　(copied from AnnotateV3.0_ORA_categories.Rmd)
# trial
In02orless.consolidated.forGSEA %>% dplyr::select(1) %>% pivot_longer(1)
# 
#In02orless.consolidated.forGSEA %>% map(~pivot_longer(cols=name)) # does not work
##
In02orless.consolidated.forGSEA.list <- list()
for(i in 1:(dim(In02orless.consolidated.forGSEA)[2]-1)) {
In02orless.consolidated.forGSEA.list[[i]] <- In02orless.consolidated.forGSEA %>% dplyr::select(i) %>% pivot_longer(1) %>% dplyr::rename(category=name)
}
#
In02orless.consolidated.forGSEA.DF <- In02orless.consolidated.forGSEA.list %>% enframe() %>% unnest(value) %>% drop_na(value)# does not work if I did not rename "category" columns
# convert into list by gene name
In02orless.consolidated.forORA.list2 <- In02orless.consolidated.forGSEA.DF %>%split(.$value) %>%   map(~dplyr::select(.,category)) %>% map(~as_vector(.))
save(In02orless.consolidated.forORA.list2,file=file.path("annotation","output","In02orless.consolidated.forORA.list2.v5.Rdata"))
# ORA
# all modules
GOseq.temp <- Br.v3.0anno.At.BLAST.highscore.moduleColors.signedhybrid %>% ungroup() %>% dplyr::select(name,root.voom5.module.signedhybrid,leaf.voom5.module.signedhybrid) %>%
pivot_longer(cols=-1,names_to = "voom")  %>% drop_na(value) %>% unite(voom.module,c("voom","value")) %>%  split(.$voom.module) %>% map(~GOseq.Brgo.v3.0.Atgoslim.BP.list.ORA2(genelist=.$name,custom.category.list=In02orless.consolidated.forORA.list2,ontology="none")) # ,custom.category.list=GSEA_cust_rootshoot_updown2.list
GOseq.temp %>% enframe() %>% unnest(value) %>% write_csv(file.path("RNA","output","WGCNA.voom5.signedhybrid.all.moduleColors.customcatv5.In02orless.consolidated.FDR0.001.GOseq.csv"))

```

# In03orless.forGSEA
```{r}
In03orless.consolidated.forGSEA <- filtered.custom.data.consolidated.numericL %>%
  filter(totalSetN <= 3, presentInSet ==1) %>%
  dplyr::select(genes, set) %>%
  mutate(id=1:nrow(.)) %>% 
  pivot_wider(names_from = set, values_from = genes, values_fill = list(genes = NA)) %>%
  dplyr::select(-id) %>%
  mutate_all(sort, na.last=TRUE) 
In03orless.consolidated.forGSEA <- In03orless.consolidated.forGSEA[apply(In03orless.consolidated.forGSEA,1,function(x) !all(is.na(x))),]
In03orless.consolidated.forGSEA %>% View()
### for ORA　(copied from AnnotateV3.0_ORA_categories.Rmd)
# trial
In03orless.consolidated.forGSEA %>% dplyr::select(1) %>% pivot_longer(1)
# 
#In03orless.consolidated.forGSEA %>% map(~pivot_longer(cols=name)) # does not work
##
In03orless.consolidated.forGSEA.list <- list()
for(i in 1:(dim(In03orless.consolidated.forGSEA)[2]-1)) {
In03orless.consolidated.forGSEA.list[[i]] <- In03orless.consolidated.forGSEA %>% dplyr::select(i) %>% pivot_longer(1) %>% dplyr::rename(category=name)
}
#
In03orless.consolidated.forGSEA.DF <- In03orless.consolidated.forGSEA.list %>% enframe() %>% unnest(value) %>% drop_na(value)# does not work if I did not rename "category" columns
# convert into list by gene name
In03orless.consolidated.forORA.list2 <- In03orless.consolidated.forGSEA.DF %>%split(.$value) %>%   map(~dplyr::select(.,category)) %>% map(~as_vector(.))
save(In03orless.consolidated.forORA.list2,file=file.path("annotation","output","In03orless.consolidated.forORA.list2.v5.Rdata"))
# ORA
# all modules
GOseq.temp <- Br.v3.0anno.At.BLAST.highscore.moduleColors.signedhybrid %>% ungroup() %>% dplyr::select(name,root.voom5.module.signedhybrid,leaf.voom5.module.signedhybrid) %>%
pivot_longer(cols=-1,names_to = "voom")  %>% drop_na(value) %>% unite(voom.module,c("voom","value")) %>%  split(.$voom.module) %>% map(~GOseq.Brgo.v3.0.Atgoslim.BP.list.ORA2(genelist=.$name,custom.category.list=In03orless.consolidated.forORA.list2,ontology="none")) # ,custom.category.list=GSEA_cust_rootshoot_updown2.list
GOseq.temp %>% enframe() %>% unnest(value) %>% write_csv(file.path("RNA","output","WGCNA.voom5.signedhybrid.all.moduleColors.customcatv5.In03orless.consolidated.FDR0.001.GOseq.csv"))

```

# In05orless.forGSEA
```{r}
In05orless.consolidated.forGSEA <- filtered.custom.data.consolidated.numericL %>%
  filter(totalSetN <= 5, presentInSet ==1) %>%
  dplyr::select(genes, set) %>%
  mutate(id=1:nrow(.)) %>% 
  pivot_wider(names_from = set, values_from = genes, values_fill = list(genes = NA)) %>%
  dplyr::select(-id) %>%
  mutate_all(sort, na.last=TRUE) 
In05orless.consolidated.forGSEA <- In05orless.consolidated.forGSEA[apply(In05orless.consolidated.forGSEA,1,function(x) !all(is.na(x))),]
In05orless.consolidated.forGSEA %>% View()
### for ORA　(copied from AnnotateV3.0_ORA_categories.Rmd)
# trial
In05orless.consolidated.forGSEA %>% dplyr::select(1) %>% pivot_longer(1)
# 
#In05orless.consolidated.forGSEA %>% map(~pivot_longer(cols=name)) # does not work
##
In05orless.consolidated.forGSEA.list <- list()
for(i in 1:(dim(In05orless.consolidated.forGSEA)[2]-1)) {
In05orless.consolidated.forGSEA.list[[i]] <- In05orless.consolidated.forGSEA %>% dplyr::select(i) %>% pivot_longer(1) %>% dplyr::rename(category=name)
}
#
In05orless.consolidated.forGSEA.DF <- In05orless.consolidated.forGSEA.list %>% enframe() %>% unnest(value) %>% drop_na(value)# does not work if I did not rename "category" columns
# convert into list by gene name
In05orless.consolidated.forORA.list2 <- In05orless.consolidated.forGSEA.DF %>%split(.$value) %>%   map(~dplyr::select(.,category)) %>% map(~as_vector(.))
save(In05orless.consolidated.forORA.list2,file=file.path("annotation","output","In05orless.consolidated.forORA.list2.v5.Rdata"))
# ORA
# all modules
GOseq.temp <- Br.v3.0anno.At.BLAST.highscore.moduleColors.signedhybrid %>% ungroup() %>% dplyr::select(name,root.voom5.module.signedhybrid,leaf.voom5.module.signedhybrid) %>%
pivot_longer(cols=-1,names_to = "voom")  %>% drop_na(value) %>% unite(voom.module,c("voom","value")) %>%  split(.$voom.module) %>% map(~GOseq.Brgo.v3.0.Atgoslim.BP.list.ORA2(genelist=.$name,custom.category.list=In05orless.consolidated.forORA.list2,ontology="none")) # ,custom.category.list=GSEA_cust_rootshoot_updown2.list
GOseq.temp %>% enframe() %>% unnest(value) %>% write_csv(file.path("RNA","output","WGCNA.voom5.signedhybrid.all.moduleColors.customcatv5.In05orless.consolidated.FDR0.001.GOseq.csv"))

```

# In10orless.forGSEA
```{r}
In10orless.consolidated.forGSEA <- filtered.custom.data.consolidated.numericL %>%
  filter(totalSetN <= 10, presentInSet ==1) %>%
  dplyr::select(genes, set) %>%
  mutate(id=1:nrow(.)) %>% 
  pivot_wider(names_from = set, values_from = genes, values_fill = list(genes = NA)) %>%
  dplyr::select(-id) %>%
  mutate_all(sort, na.last=TRUE) 
In10orless.consolidated.forGSEA <- In10orless.consolidated.forGSEA[apply(In10orless.consolidated.forGSEA,1,function(x) !all(is.na(x))),]
In10orless.consolidated.forGSEA %>% View()
### for ORA　(copied from AnnotateV3.0_ORA_categories.Rmd)
# trial
In10orless.consolidated.forGSEA %>% dplyr::select(1) %>% pivot_longer(1)
# 
#In10orless.consolidated.forGSEA %>% map(~pivot_longer(cols=name)) # does not work
##
In10orless.consolidated.forGSEA.list <- list()
for(i in 1:(dim(In10orless.consolidated.forGSEA)[2]-1)) {
In10orless.consolidated.forGSEA.list[[i]] <- In10orless.consolidated.forGSEA %>% dplyr::select(i) %>% pivot_longer(1) %>% dplyr::rename(category=name)
}
#
In10orless.consolidated.forGSEA.DF <- In10orless.consolidated.forGSEA.list %>% enframe() %>% unnest(value) %>% drop_na(value)# does not work if I did not rename "category" columns
# convert into list by gene name
In10orless.consolidated.forORA.list2 <- In10orless.consolidated.forGSEA.DF %>%split(.$value) %>%   map(~dplyr::select(.,category)) %>% map(~as_vector(.))
save(In10orless.consolidated.forORA.list2,file=file.path("annotation","output","In10orless.consolidated.forORA.list2.v5.Rdata"))
# ORA
# all modules
GOseq.temp <- Br.v3.0anno.At.BLAST.highscore.moduleColors.signedhybrid %>% ungroup() %>% dplyr::select(name,root.voom5.module.signedhybrid,leaf.voom5.module.signedhybrid) %>%
pivot_longer(cols=-1,names_to = "voom")  %>% drop_na(value) %>% unite(voom.module,c("voom","value")) %>%  split(.$voom.module) %>% map(~GOseq.Brgo.v3.0.Atgoslim.BP.list.ORA2(genelist=.$name,custom.category.list=In10orless.consolidated.forORA.list2,ontology="none")) # ,custom.category.list=GSEA_cust_rootshoot_updown2.list
GOseq.temp %>% enframe() %>% unnest(value) %>% write_csv(file.path("RNA","output","WGCNA.voom5.signedhybrid.all.moduleColors.customcatv5.In10orless.consolidated.FDR0.001.GOseq.csv"))

```

# In20orless.forGSEA
```{r}
In20orless.consolidated.forGSEA <- filtered.custom.data.consolidated.numericL %>%
  filter(totalSetN <= 20, presentInSet ==1) %>%
  dplyr::select(genes, set) %>%
  mutate(id=1:nrow(.)) %>% 
  pivot_wider(names_from = set, values_from = genes, values_fill = list(genes = NA)) %>%
  dplyr::select(-id) %>%
  mutate_all(sort, na.last=TRUE) 
In20orless.consolidated.forGSEA <- In20orless.consolidated.forGSEA[apply(In20orless.consolidated.forGSEA,1,function(x) !all(is.na(x))),]
In20orless.consolidated.forGSEA %>% View()
### for ORA　(copied from AnnotateV3.0_ORA_categories.Rmd)
# trial
In20orless.consolidated.forGSEA %>% dplyr::select(1) %>% pivot_longer(1)
# 
#In20orless.consolidated.forGSEA %>% map(~pivot_longer(cols=name)) # does not work
##
In20orless.consolidated.forGSEA.list <- list()
for(i in 1:(dim(In20orless.consolidated.forGSEA)[2]-1)) {
In20orless.consolidated.forGSEA.list[[i]] <- In20orless.consolidated.forGSEA %>% dplyr::select(i) %>% pivot_longer(1) %>% dplyr::rename(category=name)
}
#
In20orless.consolidated.forGSEA.DF <- In20orless.consolidated.forGSEA.list %>% enframe() %>% unnest(value) %>% drop_na(value)# does not work if I did not rename "category" columns
# convert into list by gene name
In20orless.consolidated.forORA.list2 <- In20orless.consolidated.forGSEA.DF %>%split(.$value) %>%   map(~dplyr::select(.,category)) %>% map(~as_vector(.))
save(In20orless.consolidated.forORA.list2,file=file.path("annotation","output","In20orless.consolidated.forORA.list2.v5.Rdata"))
# ORA
# all modules
GOseq.temp <- Br.v3.0anno.At.BLAST.highscore.moduleColors.signedhybrid %>% ungroup() %>% dplyr::select(name,root.voom5.module.signedhybrid,leaf.voom5.module.signedhybrid) %>%
pivot_longer(cols=-1,names_to = "voom")  %>% drop_na(value) %>% unite(voom.module,c("voom","value")) %>%  split(.$voom.module) %>% map(~GOseq.Brgo.v3.0.Atgoslim.BP.list.ORA2(genelist=.$name,custom.category.list=In20orless.consolidated.forORA.list2,ontology="none")) # ,custom.category.list=GSEA_cust_rootshoot_updown2.list
GOseq.temp %>% enframe() %>% unnest(value) %>% write_csv(file.path("RNA","output","WGCNA.voom5.signedhybrid.all.moduleColors.customcatv5.In20orless.consolidated.FDR0.001.GOseq.csv"))

```

# In05orMore.forGSEA
```{r}
In05orMore.consolidated.forGSEA <- filtered.custom.data.consolidated.numericL %>%
  filter(totalSetN >= 5, presentInSet ==1) %>%
  dplyr::select(genes, set) %>%
  mutate(id=1:nrow(.)) %>% 
  pivot_wider(names_from = set, values_from = genes, values_fill = list(genes = NA)) %>%
  dplyr::select(-id) %>%
  mutate_all(sort, na.last=TRUE) 
In05orMore.consolidated.forGSEA <- In05orMore.consolidated.forGSEA[apply(In05orMore.consolidated.forGSEA,1,function(x) !all(is.na(x))),]
In05orMore.consolidated.forGSEA %>% View()
### for ORA　(copied from AnnotateV3.0_ORA_categories.Rmd)
# trial
In05orMore.consolidated.forGSEA %>% dplyr::select(1) %>% pivot_longer(1)
# 
#In05orMore.consolidated.forGSEA %>% map(~pivot_longer(cols=name)) # does not work
##
In05orMore.consolidated.forGSEA.list <- list()
for(i in 1:(dim(In05orMore.consolidated.forGSEA)[2]-1)) {
In05orMore.consolidated.forGSEA.list[[i]] <- In05orMore.consolidated.forGSEA %>% dplyr::select(i) %>% pivot_longer(1) %>% dplyr::rename(category=name)
}
#
In05orMore.consolidated.forGSEA.DF <- In05orMore.consolidated.forGSEA.list %>% enframe() %>% unnest(value) %>% drop_na(value)# does not work if I did not rename "category" columns
# convert into list by gene name
In05orMore.consolidated.forORA.list2 <- In05orMore.consolidated.forGSEA.DF %>%split(.$value) %>%   map(~dplyr::select(.,category)) %>% map(~as_vector(.))
save(In05orMore.consolidated.forORA.list2,file=file.path("annotation","output","In05orMore.consolidated.forORA.list2.v5.Rdata"))
# ORA
# all modules
GOseq.temp <- Br.v3.0anno.At.BLAST.highscore.moduleColors.signedhybrid %>% ungroup() %>% dplyr::select(name,root.voom5.module.signedhybrid,leaf.voom5.module.signedhybrid) %>%
pivot_longer(cols=-1,names_to = "voom")  %>% drop_na(value) %>% unite(voom.module,c("voom","value")) %>%  split(.$voom.module) %>% map(~GOseq.Brgo.v3.0.Atgoslim.BP.list.ORA2(genelist=.$name,custom.category.list=In05orMore.consolidated.forORA.list2,ontology="none")) # ,custom.category.list=GSEA_cust_rootshoot_updown2.list
GOseq.temp %>% enframe() %>% unnest(value) %>% write_csv(file.path("RNA","output","WGCNA.voom5.signedhybrid.all.moduleColors.customcatv5.In05orMore.consolidated.FDR0.001.GOseq.csv"))

```

# In10orMore.forGSEA
```{r}
In10orMore.consolidated.forGSEA <- filtered.custom.data.consolidated.numericL %>%
  filter(totalSetN >= 10, presentInSet ==1) %>%
  dplyr::select(genes, set) %>%
  mutate(id=1:nrow(.)) %>% 
  pivot_wider(names_from = set, values_from = genes, values_fill = list(genes = NA)) %>%
  dplyr::select(-id) %>%
  mutate_all(sort, na.last=TRUE) 
In10orMore.consolidated.forGSEA <- In10orMore.consolidated.forGSEA[apply(In10orMore.consolidated.forGSEA,1,function(x) !all(is.na(x))),]
In10orMore.consolidated.forGSEA %>% View()
### for ORA　(copied from AnnotateV3.0_ORA_categories.Rmd)
# trial
In10orMore.consolidated.forGSEA %>% dplyr::select(1) %>% pivot_longer(1)
# 
#In10orMore.consolidated.forGSEA %>% map(~pivot_longer(cols=name)) # does not work
##
In10orMore.consolidated.forGSEA.list <- list()
for(i in 1:(dim(In10orMore.consolidated.forGSEA)[2]-1)) {
In10orMore.consolidated.forGSEA.list[[i]] <- In10orMore.consolidated.forGSEA %>% dplyr::select(i) %>% pivot_longer(1) %>% dplyr::rename(category=name)
}
#
In10orMore.consolidated.forGSEA.DF <- In10orMore.consolidated.forGSEA.list %>% enframe() %>% unnest(value) %>% drop_na(value)# does not work if I did not rename "category" columns
# convert into list by gene name
In10orMore.consolidated.forORA.list2 <- In10orMore.consolidated.forGSEA.DF %>%split(.$value) %>%   map(~dplyr::select(.,category)) %>% map(~as_vector(.))
save(In10orMore.consolidated.forORA.list2,file=file.path("annotation","output","In10orMore.consolidated.forORA.list2.v5.Rdata"))
# ORA
# all modules
GOseq.temp <- Br.v3.0anno.At.BLAST.highscore.moduleColors.signedhybrid %>% ungroup() %>% dplyr::select(name,root.voom5.module.signedhybrid,leaf.voom5.module.signedhybrid) %>%
pivot_longer(cols=-1,names_to = "voom")  %>% drop_na(value) %>% unite(voom.module,c("voom","value")) %>%  split(.$voom.module) %>% map(~GOseq.Brgo.v3.0.Atgoslim.BP.list.ORA2(genelist=.$name,custom.category.list=In10orMore.consolidated.forORA.list2,ontology="none")) # ,custom.category.list=GSEA_cust_rootshoot_updown2.list
GOseq.temp %>% enframe() %>% unnest(value) %>% write_csv(file.path("RNA","output","WGCNA.voom5.signedhybrid.all.moduleColors.customcatv5.In10orMore.consolidated.FDR0.001.GOseq.csv"))

```

# In20ormore.forGSEA (new, 061320) (zero)
```{r eval=FALSE}
In20orMore.consolidated.forGSEA <- filtered.custom.data.consolidated.numericL %>%
  filter(totalSetN >= 20, presentInSet ==1) %>%
  dplyr::select(genes, set) %>%
  mutate(id=1:nrow(.)) %>% 
  pivot_wider(names_from = set, values_from = genes, values_fill = list(genes = NA)) %>%
  dplyr::select(-id) %>%
  mutate_all(sort, na.last=TRUE) 
```

# total
```{r eval=FALSE}
Total.forGSEA <- filtered.custom.data.consolidated.numericL %>%
  filter(presentInSet ==1) %>%
  dplyr::select(genes, set) %>%
  mutate(id=1:nrow(.)) %>%
  pivot_wider(names_from = set, values_from = genes, values_fill = list(genes = NA)) %>%
  dplyr::select(-id) %>%
  mutate_all(sort, na.last=TRUE)
Total.forGSEA <- Total.forGSEA[apply(Total.forGSEA,1,function(x) !all(is.na(x))),]
Total.forGSEA %>% View()
# write csv file of consolidated categories
write_csv(Total.forGSEA,file=file.path("annotation","output","Total.consolidated1.v5.csv"))

# 
Total.consolidated.forGSEA.list <- list()
for(i in 1:(dim(Total.forGSEA)[2]-1)) {
Total.consolidated.forGSEA.list[[i]] <- Total.forGSEA %>% dplyr::select(i) %>% pivot_longer(1) %>% dplyr::rename(category=name)
}
#
Total.consolidated.forGSEA.DF <- Total.consolidated.forGSEA.list %>% enframe() %>% unnest(value) %>% drop_na(value)# does not work if I did not rename "category" columns
# use Total.consolidated.forGSEA.DF for later
# convert into list by gene name
Total.consolidated.forGSEA.list2 <- Total.consolidated.forGSEA.DF %>%split(.$value) %>%   map(~dplyr::select(.,category)) %>% map(~as_vector(.))

save(Total.consolidated.forGSEA.list2,file=file.path("annotation","output","Total.consolidated.forGSEA.list2.v5.Rdata"))

# all modules using filtered.custom.data.consolidated.list? why not Total.consolidated.forGSEA.list? (Sep 14, 2021)
GOseq.temp <- Br.v3.0anno.At.BLAST.highscore.moduleColors.signedhybrid %>% ungroup() %>% dplyr::select(name,root.voom5.module.signedhybrid,leaf.voom5.module.signedhybrid) %>%
pivot_longer(cols=-1,names_to = "voom")  %>% drop_na(value) %>% unite(voom.module,c("voom","value")) %>%  split(.$voom.module) %>% map(~GOseq.Brgo.v3.0.Atgoslim.BP.list.ORA2(genelist=.$name,custom.category.list=filtered.custom.data.consolidated.list,ontology="none")) # ,custom.category.list=GSEA_cust_rootshoot_updown2.list
GOseq.temp %>% enframe() %>% unnest(value) %>% write_csv(file.path("RNA","output","WGCNA.voom5.signedhybrid.all.moduleColors.customcatv5.Total.consolidated.FDR0.001.GOseq.csv"))
```



# annotate filtered.custom.data.numericL

```{r}
# annotation file for v3.0annotation
Br.v3.0.At.BLAST <- read_csv(file.path("annotation","output","Brapa_v3.0_annotated.csv")) 
# This annotation is redundant with name (Br grene). Eg 
Br.v3.0.At.BLAST %>% filter(name=="BraA01g040570.3C")
# reduce the redundancy (112418)
Br.v3.0anno.At.BLAST.highscore <- Br.v3.0.At.BLAST %>% group_by(name) %>% arrange(desc(score)) %>% dplyr::slice(1)
# reduce the redundancy for Br (051120) (do not use)
#Br.v3.0anno.At.BLAST.highscore.Br <- Br.v3.0.At.BLAST %>% group_by(AGI) %>% arrange(desc(score)) %>% slice(1) #%>% View()
```

```{r}
# bHLH039 ("") as an minusFe up marker
filtered.custom.data.consolidated.numericL %>% left_join(Br.v3.0anno.At.BLAST.highscore, by=c("genes"="name")) %>% filter(genes=="BraA04g003660.3C") %>% View() # not in In01. Found in In02 because minusFe has two datasets (minusFe_Kim2019_root_FC_up and minusFe_Kailasam2019_root_FC_up), so that In01 is not good.
```

# In onlyX strategy effects (052521)
## Combine all custom category ORA with WGCNA modules
## updated for v5
```{r}
filenames.csv <- list.files(path=file.path("RNA","output"),pattern="consolidated.FDR0.001.GOseq.csv",full.names=TRUE,recursive=TRUE)
#only v5
filenames.csv <- str_subset(filenames.csv,"v5")
# only containing FDR0.001.GOseq.csv
#filenames.csv <- tibble(filenames=filenames.csv) %>% filter(str_detect(filenames,"FDR0.001.GOseq.csv")) %>% as_vector()
filenames.csv2 <- list.files(path=file.path("RNA","output"),pattern="consolidated.FDR0.001.GOseq.csv",full.names=FALSE,recursive=FALSE)
# only v5
filenames.csv2 <- str_subset(filenames.csv2,"v5")
# read xls file (file name is "xls", actually tsv. That is why read_xls did not work)
customcat.consolidated.v5.InX.FDR0.001.ORA <- lapply(filenames.csv, function(x) read_csv(file=file.path(x)))
# name
names(customcat.consolidated.v5.InX.FDR0.001.ORA) <- filenames.csv2
# convert list into one data.frame
customcat.consolidated.v5.InX.FDR0.001.ORA.DF <- customcat.consolidated.v5.InX.FDR0.001.ORA %>% enframe(name="InX.FDR0.001") %>% unnest(value)
```

## focus on root.ivory module which is associated with leaf length as well as soil treatment. Stack heatmaps, i.e. y-axis is In X (1, 02orless, 03orless, etc)
```{r}
data <- customcat.consolidated.v5.InX.FDR0.001.ORA.DF %>% as_tibble()
# Scale each measurement (independently) to have a mean of 0 and variance of 1
#customcat.InX.FDR0.001.ORA.DF.formatted <- data %>% ifelse(value=="no enriched GO",over_represented_padjust <- 1) %>% View() # error
#ifelse(data$value=="no enriched GO",print("yes"),print("no")) # ???
# traditional if/else loop
for(i in 1:dim(data)[1]) {
  if(is.na(data[i,3])==FALSE) {data[i,"over_represented_padjust"] <- 1 } else if(is.na(data[i,3])){data[i,"over_represented_padjust"] <- data[i,"over_represented_padjust"]}
}
```

# Signature of WGCNA modules by looking enrichement of custom categories
```{r}
data2 <- data %>% mutate(name=str_replace(name,"_",".")) %>% dplyr::rename(NAME=name)  %>% 
  separate(category,into=c("category3","tissue3","FC","up_down"),sep="_") %>% # View()
  mutate(category3=category3 %>% 
           str_replace_all("MINUSMG","MG-") %>%
           str_replace_all("PLUSMG","MG+") %>%
           str_replace_all("MINUSFE","FE-") %>%
    str_replace_all("PLUSFE","FE_PLUS") %>%
  str_replace_all("MINUSPI-1D","PI-1D-") %>%
    str_replace_all("PLUSPI-1D","PI-1D+") %>%
  str_replace_all("MINUSPI-3D","PI-3D+") %>%
    str_replace_all("PLUSPI-3D","PI-3D+") %>%
  str_replace_all("MINUSN","N-") %>%
  str_replace_all("PLUSN","N+") %>%
  str_replace_all("MINUSMN","MN-") %>%
  str_replace_all("PLUSMN","MN+") %>% 
  str_replace_all("MINUSP","P-") %>% 
    str_replace_all("PLUSP","P+") %>% 
    str_replace_all("PLUSIAA","IAA+") %>% 
    str_replace_all("PLUSAL","AL+") %>%
    str_replace_all("MINUSCA","CA-") %>% 
      str_replace_all("PLUSCA","CA+") %>%
    str_replace_all("MINUSK","K-") %>% 
    str_replace_all("PLUSK","K+") %>% 
    str_replace_all("MINUSS","S-") %>% 
    str_replace_all("PLUSS","S+")) %>%
  drop_na(category3) %>%
  unite(category3.up_down,category3, up_down)
# check
data2 %>% View()
# note: "tissue3" and "category3" came from the same name (i.e. "tissue3" is tissues used for "category3")
# arbitrally convert value of 0 into 10^-10 for scale visualization
data2[data2$over_represented_padjust < 10^-10,"over_represented_padjust"] <- 10^-10
# clean up and sort "InX.FDR0.001" column
data3 <- data2 %>% mutate(InX=str_replace(InX.FDR0.001,"WGCNA.voom5.signedhybrid.all.moduleColors.customcatv5.Total.consolidatedv5.FDR0.001.GOseq.csv","total")) %>% mutate(InX=str_remove_all(InX,"WGCNA.voom5.signedhybrid.all.moduleColors.customcatv5.")) %>% mutate(InX=str_remove_all(InX,"\\.consolidated.FDR0.001.GOseq.csv")) #%>% View() # does not work
table(data3$InX)
data3 <- data3 %>% mutate(InX=factor(InX,levels=c("In10orMore","In05orMore","Total","In20orless","In10orless","In05orless","In03orless","In02orless","In01")))
table(data3$InX) # does work (no NA)
# save data3
write_csv(data3,file=file.path("RNA","output","data3.csv"))
```



#############################
# which genes found in root.ivory module?
```{r}
load(file.path("RNA","output","filtered.custom.data.consolidated.numericL.Rdata"))
filtered.custom.data.consolidated.numericLL <- filtered.custom.data.consolidated.numericL %>%
  pivot_longer(c(starts_with("In"),starts_with("Total", ignore.case = FALSE)), names_to = "group", values_to = "group_summary")
head(filtered.custom.data.consolidated.numericLL,20)

filtered.custom.data.consolidated.numericLL
Br.v3.0anno.At.BLAST.highscore.modules.signedhybrid
#
temp <- filtered.custom.data.consolidated.numericLL %>% filter(set %in% c("minusN_root_FC_down","plusMg_root_FC_down")) %>% filter(totalSetN <=20, group_summary=="1") %>% filter(presentInSet==1) %>% inner_join(Br.v3.0anno.At.BLAST.highscore.modules.signedhybrid %>% filter(root.voom5.module.signedhybrid=="ivory"), by=c("genes"="name")) %>% arrange(totalSetN) %>% dplyr::select("genes","totalSetN","set","AGI","At_symbol","At_short_description","root.voom5.module.signedhybrid","leaf.voom5.module.signedhybrid","presentInSet") %>% distinct()
# count
temp.count <- temp %>% count(genes) #%>% View()
temp %>% pivot_wider(names_from="set",values_from="presentInSet") %>% left_join(temp.count,by="genes") %>% group_by(plusMg_root_FC_down) %>% arrange(n,.group_by=TRUE) %>% write_csv(file.path("RNA","output","root.ivory.specific.genelist.csv")) #View()
```

# all in one plot (June 29, 2021) -> updated version (Aug 17, 2021, under construction)
## This is not good to check which X corresponds in each plot
```{r}
InX.comparison.plot3 <- function(data=data3) {
  #data <- data %>% filter(NAME==target.module)
  # plot 
WGCNA.voom5.signedhybrid.target.customcat.GOseq.plot.InXcomparison <- ggplot(data, aes(x=category3.up_down,y=InX)) + geom_tile(aes(fill=-log10(over_represented_padjust)),colour="white") + scale_fill_gradient2(limit=c(0,10), high=muted("magenta")) + facet_grid(NAME~tissue3,scales="free_x",space="free")

WGCNA.voom5.signedhybrid.target.customcat.GOseq.plot.InXcomparison <- WGCNA.voom5.signedhybrid.target.customcat.GOseq.plot.InXcomparison + theme(axis.text.x=element_text(size=12,angle=90,hjust=1,vjust=0.3),
                                      axis.text.y=element_text(size=12),
                                      axis.title=element_text(size=10),
                                      axis.ticks = element_blank(),
                                      strip.text.y = element_text(size=16,angle=0),
                                      strip.text.x = element_text(size=8,angle=90),
                                      panel.background = element_rect(fill = "white",colour="black"),
                                      plot.title=element_text(size=14),
                                      axis.line=element_blank()) + 
      labs(x="",y="",fill="-log10\n FDR",title=str_c("WGCNA.customcatv5.consolidated1.GOseq.plot.InXcomparison."))
# adding overlap numbers
WGCNA.voom5.signedhybrid.target.customcat.GOseq.plot.InXcomparison <- WGCNA.voom5.signedhybrid.target.customcat.GOseq.plot.InXcomparison + geom_text(aes(label=numDEInCat))

WGCNA.voom5.signedhybrid.target.customcat.GOseq.plot.InXcomparison
ggsave(path=file.path("RNA","output"),filename="WGCNA.customcatv5.consolidated1.GOseq.plot.InXcomparison.allmodules.pdf",height=200,width=20,limitsize=FALSE) 
}

# test
InX.comparison.plot3()

```

# new InX.comparion.plot4() for color coding FDR by microbiome FC (Aug 23, 2021) -> abundan this (Sep 14, 2021)

### start calculating microbiome responsiveness FC
# calculate microbe FC
# load normalized log2 expression data (Brassica genome v3.0 annotation) 
```{r}
# plus R500 in residual
root.voom5 <- readr::read_tsv(file.path("RNA","output","voom_expression.e1and3.resid.exp_gt.root.plusR500.txt.gz"))
leaf.voom5 <- readr::read_tsv(file.path("RNA","output","voom_expression.e1and3.resid.exp_gt.leaf.plusR500.txt.gz"))
```

# WGCNA modules
```{r}
Br.v3.0anno.At.BLAST.highscore.moduleColors.signedhybrid <- read_csv(file.path("RNA","output","WGCNA.signedhybrid.moduleColors.voom5.csv.gz"))
```

# make sample description data frame 
```{r}
sample.description.e1 <- read_csv(file.path("plant","output","Br.mbio.e1.sample.description.csv"))
sample.description.e3 <- read_csv(file.path("plant","output","Br.mbio.e3.sample.description.csv"))

sample.description.e1 %>% summarize(n_distinct(group))
sample.description.e1 %>% group_by(trt) %>% summarize(n())
##
root.voom5.e1 <- tibble(root.voom5 = colnames(root.voom5)[-1]) %>% inner_join(sample.description.e1[,c("sample","genotype","tissue","trt")],by=c("root.voom5"="sample"))
root.voom5.e3 <- tibble(root.voom5 = colnames(root.voom5)[-1]) %>% mutate(genotype="FPsc") %>% inner_join(sample.description.e3[,c("sample","tissue","trt")],by=c("root.voom5"="sample"))  
root.voom5.sample <- bind_rows(root.voom5.e1,root.voom5.e3) %>% mutate(trt=str_remove(trt,"5E_"))
# for Whitney
# root.voom5.sample <- bind_rows(root.voom5.e1,root.voom5.e3)
# root.voom5.sample$trt <- gsub("5E_","",root.voom5.sample$trt)
root.voom5.sample %>% group_by(genotype) %>% summarise(num=n())


# leaf.voom5.sample
leaf.voom5.e1 <- tibble(leaf.voom5 = colnames(leaf.voom5)[-1]) %>% inner_join(sample.description.e1[,c("sample","genotype","tissue","trt")],by=c("leaf.voom5"="sample"))
leaf.voom5.e3 <- tibble(leaf.voom5 = colnames(leaf.voom5)[-1]) %>% mutate(genotype="FPsc") %>% inner_join(sample.description.e3[,c("sample","tissue","trt")],by=c("leaf.voom5"="sample"))  
leaf.voom5.sample <- bind_rows(leaf.voom5.e1,leaf.voom5.e3) %>% mutate(trt=str_remove(trt,"5E_")) # str_remove did not work in Whitney (041721)
# for Whitney
# leaf.voom5.sample <- bind_rows(leaf.voom5.e1,leaf.voom5.e3)
# leaf.voom5.sample$trt <- gsub("5E_","",root.voom5.sample$trt)
leaf.voom5.sample %>% group_by(genotype) %>% summarise(num=n())
```

## gene annotation
```{r}
# annotation file for v3.0annotation
Br.v3.0.At.BLAST <- read_csv(file.path("annotation","output","Brapa_v3.0_annotated.csv")) 
# This annotation is redundant with name (Br grene). Eg 
Br.v3.0.At.BLAST %>% filter(name=="BraA01g040570.3C")
# reduce the redundancy (112418)
Br.v3.0anno.At.BLAST.highscore <- Br.v3.0.At.BLAST %>% group_by(name) %>% arrange(desc(score)) %>% dplyr::slice(1)
```
# calculate microbiome FC (root)
```{r}
root.voom5.trt.mean <- root.voom5 %>% pivot_longer(col=-1) %>% inner_join(root.voom5.sample,by=c("name"="root.voom5")) %>% group_by(genes,genotype,trt) %>% summarize(mean.soilFC=mean(value))
root.voom5.trt.mean # scaled to zero, i.e., value in "dead" and one in "live" are the same absolute value.
root.voom5.trt.mean.FPsc.live <- root.voom5.trt.mean %>% filter(trt=="live",genotype=="FPsc")
# add ATG
root.voom5.trt.mean.FPsc.live.ATG.moduleColors.signedhybrid <- root.voom5.trt.mean.FPsc.live %>% inner_join(Br.v3.0anno.At.BLAST.highscore.moduleColors.signedhybrid,by=c("genes"="name"))
```

# calculate microbiome responsiveness FC (leaf)
```{r}
leaf.voom5.trt.mean <- leaf.voom5 %>% pivot_longer(col=-1) %>% inner_join(leaf.voom5.sample,by=c("name"="leaf.voom5")) %>% group_by(genes,genotype,trt) %>% summarize(mean.soilFC=mean(value))
leaf.voom5.trt.mean # scaled to zero, i.e., value in "dead" and one in "live" are the same absolute value.
leaf.voom5.trt.mean.FPsc.live <- leaf.voom5.trt.mean %>% filter(trt=="live",genotype=="FPsc")
# add ATG
leaf.voom5.trt.mean.FPsc.live.ATG.moduleColors.signedhybrid <- leaf.voom5.trt.mean.FPsc.live %>% inner_join(Br.v3.0anno.At.BLAST.highscore.moduleColors.signedhybrid,by=c("genes"="name"))
```

### the end of calculation of microbe FC

#  calculate microbe FC in each InX module (run once. Use output csv file for further)
```{r}
load(file.path("annotation","output","In01.consolidated.forORA.list2.v5.Rdata"))
load(file.path("annotation","output","In02orless.consolidated.forORA.list2.v5.Rdata"))
load(file.path("annotation","output","In03orless.consolidated.forORA.list2.v5.Rdata"))
load(file.path("annotation","output","In05orless.consolidated.forORA.list2.v5.Rdata"))
load(file.path("annotation","output","In10orless.consolidated.forORA.list2.v5.Rdata"))
load(file.path("annotation","output","In20orless.consolidated.forORA.list2.v5.Rdata"))
load(file.path("annotation","output","Total.consolidated.forGSEA.list2.v5.Rdata"))
load(file.path("annotation","output","In05orMore.consolidated.forORA.list2.v5.Rdata"))
load(file.path("annotation","output","In10orMore.consolidated.forORA.list2.v5.Rdata"))
# check loaded cosolidated custom categories
In01.consolidated.forORA.list2
In02orless.consolidated.forORA.list2
In03orless.consolidated.forORA.list2
In05orless.consolidated.forORA.list2
In10orless.consolidated.forORA.list2
Total.consolidated.forGSEA.list2
In05orMore.consolidated.forORA.list2
In10orMore.consolidated.forORA.list2
# check WGCNA modules
root.voom5.trt.mean.FPsc.live.ATG.moduleColors.signedhybrid
leaf.voom5.trt.mean.FPsc.live.ATG.moduleColors.signedhybrid
# create function
overlap.genes.WGCNAmodule.category2 <- function(tissue=c("root","leaf"),custom.category.list=In01.consolidated.forORA.list2) {
  if(tissue=="root") {
    temp.root <- custom.category.list %>% enframe(value="category") %>% unnest(category) %>%  inner_join(root.voom5.trt.mean.FPsc.live.ATG.moduleColors.signedhybrid,by=c("name"="genes")) #%>% View()
    summary(temp.root)
# test
    temp.root %>% filter(root.voom5.module.signedhybrid=="bisque4",category=="cold_root_FC_down") %>% print()
# summarize() of mean FC by root modules and custom categories
    temp.root.summary <- temp.root %>% group_by(root.voom5.module.signedhybrid,category) %>% dplyr::summarise(soilFC=mean(mean.soilFC)) #%>% View() # values are positive to negative. What does that mean?
return(temp.root.summary)
  } else if(tissue=="leaf") {
  temp.leaf <- custom.category.list %>% enframe(value="category") %>% unnest(category) %>%  inner_join(leaf.voom5.trt.mean.FPsc.live.ATG.moduleColors.signedhybrid,by=c("name"="genes")) #%>% View()
    summary(temp.leaf)
# summarize() of mean FC by root modules and custom categories
    temp.leaf.summary <- temp.leaf %>% group_by(leaf.voom5.module.signedhybrid,category) %>% dplyr::summarise(soilFC=mean(mean.soilFC)) #%>% View() # values are positive to negative. What does that mean?
return(temp.leaf.summary)
  
  } else {print("Please specify tissue")}
}
# calc (error)
root.In01 <- overlap.genes.WGCNAmodule.category2(custom.category.list=In01.consolidated.forORA.list2,tissue="root") %>% mutate(tissue="root",InX="In01") %>% dplyr::rename(module=root.voom5.module.signedhybrid)
leaf.In01 <- overlap.genes.WGCNAmodule.category2(custom.category.list=In01.consolidated.forORA.list2,tissue="leaf") %>% mutate(tissue="leaf",InX="In01") %>% dplyr::rename(module=leaf.voom5.module.signedhybrid)
root.In02orless <- overlap.genes.WGCNAmodule.category2(custom.category.list=In02orless.consolidated.forORA.list2,tissue="root") %>% mutate(tissue="root",InX="In02orless") %>% dplyr::rename(module=root.voom5.module.signedhybrid)
leaf.In02orless <- overlap.genes.WGCNAmodule.category2(custom.category.list=In02orless.consolidated.forORA.list2,tissue="leaf") %>% mutate(tissue="leaf",InX="In02orless") %>% dplyr::rename(module=leaf.voom5.module.signedhybrid)

root.In03orless <- overlap.genes.WGCNAmodule.category2(custom.category.list=In03orless.consolidated.forORA.list2,tissue="root") %>% mutate(tissue="root",InX="In03orless") %>% dplyr::rename(module=root.voom5.module.signedhybrid)
leaf.In03orless <- overlap.genes.WGCNAmodule.category2(custom.category.list=In03orless.consolidated.forORA.list2,tissue="leaf") %>% mutate(tissue="leaf",InX="In03orless") %>% dplyr::rename(module=leaf.voom5.module.signedhybrid)

root.In05orless <- overlap.genes.WGCNAmodule.category2(custom.category.list=In05orless.consolidated.forORA.list2,tissue="root") %>% mutate(tissue="root",InX="In05orless") %>% dplyr::rename(module=root.voom5.module.signedhybrid)
leaf.In05orless <- overlap.genes.WGCNAmodule.category2(custom.category.list=In05orless.consolidated.forORA.list2,tissue="leaf") %>% mutate(tissue="leaf",InX="In05orless") %>% dplyr::rename(module=leaf.voom5.module.signedhybrid)

root.In10orless <- overlap.genes.WGCNAmodule.category2(custom.category.list=In10orless.consolidated.forORA.list2,tissue="root") %>% mutate(tissue="root",InX="In10orless") %>% dplyr::rename(module=root.voom5.module.signedhybrid)
leaf.In10orless <- overlap.genes.WGCNAmodule.category2(custom.category.list=In10orless.consolidated.forORA.list2,tissue="leaf") %>% mutate(tissue="leaf",InX="In10orless") %>% dplyr::rename(module=leaf.voom5.module.signedhybrid)

root.In20orless <- overlap.genes.WGCNAmodule.category2(custom.category.list=In20orless.consolidated.forORA.list2,tissue="root") %>% mutate(tissue="root",InX="In20orless") %>% dplyr::rename(module=root.voom5.module.signedhybrid)
leaf.In20orless <- overlap.genes.WGCNAmodule.category2(custom.category.list=In10orless.consolidated.forORA.list2,tissue="leaf") %>% mutate(tissue="leaf",InX="In20orless") %>% dplyr::rename(module=leaf.voom5.module.signedhybrid)

root.Total <- overlap.genes.WGCNAmodule.category2(custom.category.list=Total.consolidated.forGSEA.list2,tissue="root") %>% mutate(tissue="root",InX="Total") %>% dplyr::rename(module=root.voom5.module.signedhybrid)
leaf.Total <- overlap.genes.WGCNAmodule.category2(custom.category.list=Total.consolidated.forGSEA.list2,tissue="leaf") %>% mutate(tissue="leaf",InX="Total") %>% dplyr::rename(module=leaf.voom5.module.signedhybrid)

root.In05orMore <- overlap.genes.WGCNAmodule.category2(custom.category.list=In05orMore.consolidated.forORA.list2,tissue="root") %>% mutate(tissue="root",InX="In05orMore") %>% dplyr::rename(module=root.voom5.module.signedhybrid)
leaf.In05orMore <- overlap.genes.WGCNAmodule.category2(custom.category.list=In05orMore.consolidated.forORA.list2,tissue="leaf") %>% mutate(tissue="leaf",InX="In05orMore") %>% dplyr::rename(module=leaf.voom5.module.signedhybrid)

root.In10orMore <- overlap.genes.WGCNAmodule.category2(custom.category.list=In10orMore.consolidated.forORA.list2,tissue="root") %>% mutate(tissue="root",InX="In10orMore") %>% dplyr::rename(module=root.voom5.module.signedhybrid)
leaf.In10orMore <- overlap.genes.WGCNAmodule.category2(custom.category.list=In10orMore.consolidated.forORA.list2,tissue="leaf") %>% mutate(tissue="leaf",InX="In10oreMore") %>% dplyr::rename(module=leaf.voom5.module.signedhybrid)
# combine all
InX.all.soilFC <- bind_rows(root.In01,leaf.In01,
                            root.In02orless,leaf.In02orless,
                            root.In03orless,leaf.In03orless,
                            root.In05orless,leaf.In05orless,
                            root.In10orless,leaf.In10orless,
                            root.In20orless,leaf.In20orless,
                            root.Total,leaf.Total,
                            root.In05orMore,leaf.In05orMore,
                            root.In10orMore,leaf.In10orMore) 
summary(InX.all.soilFC)
write_csv(InX.all.soilFC,file=file.path("RNA","output","InX.all.soilFC.csv"))
```


# start from this line. All above are prep for below.
# new plot with color coding of heat plot according to soil FC.
#
# separate plot for each module (Dec 21, 2021): giving up to play with facet_wrap()
# another trial by modifying function InX.comparison.plot() (Jan 4, 2022) -> Jan 18, 2022
```{r}
# read soilFC
InX.all.soilFC <- read_csv(file.path("RNA","output","InX.all.soilFC.csv"))
# get data3
data3 <- read_csv(file.path("RNA","output","data3.csv"))
# full levels (for each tissue3 type?)
full.level.x. <- data3 %>% distinct(category3.up_down) %>% arrange(category3.up_down) %>% as_vector()
full.level.y <- data3 %>% distinct(InX) %>% arrange(InX) %>% as_vector()

# instead. Start from a complete x and y, and then combine with results
## where I can find full.level of category?
full.level.x.y <- tibble() # 




#
InX.comparison.plot1.v2 <- function(data=data3,target.module="root.voom5.module.signedhybrid.darkred",level.x = full.level.x, level.y=full.level.y,soilFC=InX.all.soilFC) {# data3.root.darkred <- data3 %>% filter(NAME=="root.voom5.module.signedhybrid.darkred")
#data.target <- data %>% filter(NAME==target.module) # 
# instead enter FDR = 1 in non target.module
data.target <- data %>% mutate(over_represented_padjust=ifelse(NAME==target.module,over_represented_padjust,1)) #%>% dplyr::select(over_represented_padjust,NAME)
data.target %>% View()
# modify data.target having full levels of x and y
#print("original level")
#str(data.target) # original
#data.target <- data.target %>% mutate(category3.up_down=factor(category3.up_down,levels=level.x)) %>% mutate(InX=factor(InX,levels=level.y))
#print("full level")
#str(data.target) # full level version 
# plot 
WGCNA.voom5.signedhybrid.target.customcat.GOseq.plot.InXcomparison <- ggplot(data.target, aes(x=category3.up_down,y=InX)) + geom_tile(aes(fill=-log10(over_represented_padjust)),colour="white") + scale_fill_gradient2(limit=c(0,10), high=muted("magenta")) + facet_grid(.~tissue3,scales="free_x",space="free")

WGCNA.voom5.signedhybrid.target.customcat.GOseq.plot.InXcomparison <- WGCNA.voom5.signedhybrid.target.customcat.GOseq.plot.InXcomparison + theme(axis.text.x=element_text(size=12,angle=90,hjust=1,vjust=0.3),
                                      axis.text.y=element_text(size=12),
                                      axis.title=element_text(size=10),
                                      axis.ticks = element_blank(),
                                      strip.text.y = element_text(angle=0),
                                      strip.text.x = element_text(size=8,angle=90),
                                      panel.background = element_rect(fill = "white",colour="black"),
                                      plot.title=element_text(size=14),
                                      axis.line=element_blank()) + 
      labs(x="",y="",fill="-log10\n FDR",title=str_remove(target.module,"voom5.module.signedhybrid."))
# adding overlap numbers
WGCNA.voom5.signedhybrid.target.customcat.GOseq.plot.InXcomparison <- WGCNA.voom5.signedhybrid.target.customcat.GOseq.plot.InXcomparison + geom_text(aes(label=numDEInCat)) + facet_grid(.~tissue3,scales="free_x",space="free")

WGCNA.voom5.signedhybrid.target.customcat.GOseq.plot.InXcomparison
ggsave(path=file.path("RNA","output"),filename=str_c("WGCNA.",target.module,".customcat.consolidatedv5.GOseq.plot.InXcomparison.png"),height=8,width=11) 
}
# test
InX.comparison.plot1.v2(data=data3,target.module="root.voom5.module.signedhybrid.darkred") # does not work as I expected
# 
InX.comparison.plot1.v2(data=data3,target.module="leaf.voom5.module.signedhybrid.lightgreen") # does not work as I expected

# how many mnodules we have?
data3 %>% distinct(NAME) # 53 modules

# 
#data3 %>% distinct(NAME)  %>% map(~InX.comparison.plot(target.module=.$NAME)) # does not owrk
# plot for all WGCNA modules
data3 %>% distinct(NAME) %>% apply(.,1,function(x) InX.comparison.plot(target.module=x))
# problem. each plot has different X and Y. (Jan 4, 2022)

```

# using soil FC
```{r}
InX.comparison.plot5 <- function(data=data3,soilFC=InX.all.soilFC) {
  data <- data %>% mutate(NAME2=str_remove_all(NAME,"voom5.module.signedhybrid.")) #%>% separate(NAME2,into=c("tissue","module"))
  # unite
    data <- data %>% unite(col=tissue.module.category.InX, c(NAME2,tissue3,category3.up_down,InX),sep=".",remove=FALSE) # this tissue3 is for category, not for module.
  # modify soilFC
    soilFC <- soilFC %>% separate(category,into=c("category3","tissue3","FC","up_down")) %>% unite(category3.up_down,c(category3,up_down),sep="_") %>% unite(col=tissue.module.category.InX, c(tissue,module,tissue3,category3.up_down,InX),sep=".",remove=FALSE) #%>% View(). what is this tissue? tissue for modules.
    # combine data and soilFC
    data.combine <- data %>% inner_join(soilFC,by="tissue.module.category.InX") #
    data.combine
    # modify over_represented_padjust according to soilFC
    data.combine <- data.combine %>% mutate(minuslog10.soilFC.over_represented_padjust=ifelse(soilFC>=0,-log10(over_represented_padjust),log10(over_represented_padjust))) #%>% View()
  # plot 
WGCNA.voom5.signedhybrid.target.customcat.GOseq.plot.InXcomparison <- ggplot(data.combine, aes(x=category3.up_down.x,y=InX.x)) + geom_tile(aes(fill=minuslog10.soilFC.over_represented_padjust),colour="white") + scale_fill_gradient2(limit=c(-10,10), low="green",high=muted("magenta")) + facet_wrap(vars(NAME,tissue3.x),scales="free_x",ncol=4)

WGCNA.voom5.signedhybrid.target.customcat.GOseq.plot.InXcomparison <- WGCNA.voom5.signedhybrid.target.customcat.GOseq.plot.InXcomparison + theme(axis.text.x=element_text(size=12,angle=90,hjust=1,vjust=0.3),
                                      axis.text.y=element_text(size=12),
                                      axis.title=element_text(size=10),
                                      axis.ticks = element_blank(),
                                      strip.text.y = element_text(size=16,angle=0),
                                      strip.text.x = element_text(size=8,angle=0),
                                      panel.background = element_rect(fill = "white",colour="black"),
                                      plot.title=element_text(size=14),
                                      axis.line=element_blank()) + 
      labs(x="",y="",fill="-log10\n FDR",title=str_c("WGCNA.customcatv5.consolidated1.GOseq.plot.InXcomparison."))
# adding overlap numbers
WGCNA.voom5.signedhybrid.target.customcat.GOseq.plot.InXcomparison <- WGCNA.voom5.signedhybrid.target.customcat.GOseq.plot.InXcomparison + geom_text(aes(label=numDEInCat))

WGCNA.voom5.signedhybrid.target.customcat.GOseq.plot.InXcomparison
#ggsave(path=file.path("RNA","output"),filename="WGCNA.customcatv5.consolidated1.GOseq.plot.InXcomparison.allmodules.soilFC.pdf",height=200,width=20,limitsize=FALSE) 
return(WGCNA.voom5.signedhybrid.target.customcat.GOseq.plot.InXcomparison)
}

# test
p.all2 <- InX.comparison.plot5()
ggsave(p.all2, path=file.path("RNA","output"),filename="WGCNA.customcatv5.consolidated1.GOseq.plot.InXcomparison.allmodules.soilFC.v2.pdf",height=800,width=20,limitsize=FALSE) # give up
```






############################################################
############################################################
# did not use following scripts (unused or old)
