---
title: "custom_categories_consolidate1_forV5_heatmap"
author: "Kazu"
date: "5/9/2021"
output: html_document
editor_options: 
  chunk_output_type: console
---

Summary

History
* copied from "06_f_cor_table.Rmd"
* add GSEA_cust_rootshoot_updown3.FDR0.001.list (051221)
*  Expand DEGs with FC threshold for all custom categoreis
* root data including minusN_Peng_2007 and plusIAA_Nembauser_2006
* adding Nishida_2017 to root categories (070720)
* making top20, top50, and top100 for (a) among root/leaf categories, (b) among minusX_up, (c) among minusX_down
* More stringent FDR for custom category ORA of root.ivory module (050921)
* "In only" approach for custom ORA (051821)
* Fair "In only" approach for custom ORA using consolidated categoreis (052821)
* Using consolidated categories in InX selection process and ORA and heatmap with the consolidated categories 
* Another version (consolidate2) is using the consolidated categoreis to generate InX, but use the original categories for ORA.
* plot heatmap of InX comparison for root.ivory module
* and more modules.
* Making v4 custom categories and use it for this script
* Modify for updated plusN_Wang2003 using v5 (080621-081021)
* Save as "custom_categories_consolidate1_forV5"
* heatmap coloaring by soil FC sign (positive vs negative) (Sep 2021)
* Cleaning up scripts (Jan 25, 2021)
* split scripts into two (Jan 27, 2021): 
** custom_categories_consolidate1_forV5.Rmd for consolidating custom categoreis.
** custom_categories_consolidate1_forV5_WGCNA_module_ORA.Rmd
*** WGCNA module ORA with InX
** custom_categories_consolidate1_forV5_heatmap
** InX.comparison.plot5.2() funciton for InX comparisons
** InX.onlyTotal.plot5.3() function only for Total (Jan 31, 2023)
** bug fix having "NA" in category (working with InX.onlyTotal.plot5.3()) (Feb 2023)

To Do
* Cleaning up scripts more
* having the same x-axis and y-axis for WGCNA heatmap in every WGCNA module
* Oops! Use WGCNA.signedhybrid.moduleColors.voom5.csv.gz instead of WGCNA.signedhybrid.module.voom5.csv.gz

#### making root + shoot category with up and down and
# prep
```{r}
rm(list=ls())
knitr::opts_chunk$set(echo = TRUE,error=TRUE)
library(tidyverse);library(GGally);library(readxl)
library(readr)
library(stringr)
# https://bioconductor.org/packages/devel/workflows/vignettes/maEndToEnd/inst/doc/MA-Workflow.html
# functions for expression pattern
#source("../../../tools/Expression_pattern_graph.R",chdir=TRUE)
library(scales) # for muted function
library(ggdendro) # for dendrogram
library(cowplot)
library(ggnewscale)
```


# which genes found in root.ivory module? (optional)
```{r}
# gene list
#Br.v3.0anno.At.BLAST.highscore.modules.signedhybrid <- read_csv(file.path("RNA","output","WGCNA.signedhybrid.modules.voom5.csv.gz"))#
Br.v3.0anno.At.BLAST.highscore.moduleColors.signedhybrid <- read_csv(file.path("RNA","output","WGCNA.signedhybrid.moduleColors.voom5.csv.gz"))
#  
load(file.path("RNA","output","filtered.custom.data.consolidated.numericL.Rdata"))
# 
filtered.custom.data.consolidated.numericLL <- filtered.custom.data.consolidated.numericL %>%
  pivot_longer(c(starts_with("In"),starts_with("Total", ignore.case = FALSE)), names_to = "group", values_to = "group_summary")
head(filtered.custom.data.consolidated.numericLL,20)

filtered.custom.data.consolidated.numericLL
Br.v3.0anno.At.BLAST.highscore.moduleColors.signedhybrid
#
temp <- filtered.custom.data.consolidated.numericLL %>% filter(set %in% c("minusN_root_FC_down","plusMg_root_FC_down")) %>% filter(totalSetN <=20, group_summary=="1") %>% filter(presentInSet==1) %>% inner_join(Br.v3.0anno.At.BLAST.highscore.moduleColors.signedhybrid %>% filter(root.voom5.module.signedhybrid=="ivory"), by=c("genes"="name")) %>% arrange(totalSetN) %>% dplyr::select("genes","totalSetN","set","AGI","At_symbol","At_short_description","root.voom5.module.signedhybrid","leaf.voom5.module.signedhybrid","presentInSet") %>% distinct()
# count
temp.count <- temp %>% dplyr::count(genes) #%>% View()
temp %>% pivot_wider(names_from="set",values_from="presentInSet") %>% left_join(temp.count,by="genes") %>% group_by(plusMg_root_FC_down) %>% arrange(n,.group_by=TRUE) %>% write_csv(file.path("RNA","output","root.ivory.specific.genelist.csv")) #View()
```


### start calculating microbiome responsiveness FC
# calculate microbe FC
# load normalized log2 expression data (Brassica genome v3.0 annotation) 
```{r}
# plus R500 in residual
root.voom5 <- readr::read_tsv(file.path("RNA","output","voom_expression.e1and3.resid.exp_gt.root.plusR500.txt.gz"))
leaf.voom5 <- readr::read_tsv(file.path("RNA","output","voom_expression.e1and3.resid.exp_gt.leaf.plusR500.txt.gz"))
```

# WGCNA modules (correct one)
```{r}
Br.v3.0anno.At.BLAST.highscore.moduleColors.signedhybrid <- read_csv(file.path("RNA","output","WGCNA.signedhybrid.moduleColors.voom5.csv.gz"))
```

# make sample description data frame 
```{r}
sample.description.e1 <- read_csv(file.path("plant","output","Br.mbio.e1.sample.description.csv"))
sample.description.e3 <- read_csv(file.path("plant","output","Br.mbio.e3.sample.description.csv"))

sample.description.e1 %>% summarize(n_distinct(group))
sample.description.e1 %>% group_by(trt) %>% summarize(n())
##
root.voom5.e1 <- tibble(root.voom5 = colnames(root.voom5)[-1]) %>% inner_join(sample.description.e1[,c("sample","genotype","tissue","trt")],by=c("root.voom5"="sample"))
root.voom5.e3 <- tibble(root.voom5 = colnames(root.voom5)[-1]) %>% mutate(genotype="FPsc") %>% inner_join(sample.description.e3[,c("sample","tissue","trt")],by=c("root.voom5"="sample"))  
root.voom5.sample <- bind_rows(root.voom5.e1,root.voom5.e3) %>% mutate(trt=str_remove(trt,"5E_"))
# for Whitney
# root.voom5.sample <- bind_rows(root.voom5.e1,root.voom5.e3)
# root.voom5.sample$trt <- gsub("5E_","",root.voom5.sample$trt)
root.voom5.sample %>% group_by(genotype) %>% summarise(num=n())


# leaf.voom5.sample
leaf.voom5.e1 <- tibble(leaf.voom5 = colnames(leaf.voom5)[-1]) %>% inner_join(sample.description.e1[,c("sample","genotype","tissue","trt")],by=c("leaf.voom5"="sample"))
leaf.voom5.e3 <- tibble(leaf.voom5 = colnames(leaf.voom5)[-1]) %>% mutate(genotype="FPsc") %>% inner_join(sample.description.e3[,c("sample","tissue","trt")],by=c("leaf.voom5"="sample"))  
leaf.voom5.sample <- bind_rows(leaf.voom5.e1,leaf.voom5.e3) %>% mutate(trt=str_remove(trt,"5E_")) # str_remove did not work in Whitney (041721)
# for Whitney
# leaf.voom5.sample <- bind_rows(leaf.voom5.e1,leaf.voom5.e3)
# leaf.voom5.sample$trt <- gsub("5E_","",root.voom5.sample$trt)
leaf.voom5.sample %>% group_by(genotype) %>% summarise(num=n())
```

## gene annotation
```{r}
# annotation file for v3.0annotation
Br.v3.0.At.BLAST <- read_csv(file.path("annotation","output","Brapa_v3.0_annotated.csv")) 
# This annotation is redundant with name (Br grene). Eg 
Br.v3.0.At.BLAST %>% filter(name=="BraA01g040570.3C")
# reduce the redundancy (112418)
Br.v3.0anno.At.BLAST.highscore <- Br.v3.0.At.BLAST %>% group_by(name) %>% arrange(desc(score)) %>% dplyr::slice(1)
```
# calculate microbiome FC (root)
```{r}
root.voom5.trt.mean <- root.voom5 %>% pivot_longer(col=-1) %>% inner_join(root.voom5.sample,by=c("name"="root.voom5")) %>% group_by(genes,genotype,trt) %>% summarize(mean.soilFC=mean(value))
root.voom5.trt.mean # scaled to zero, i.e., value in "dead" and one in "live" are the same absolute value.
root.voom5.trt.mean.FPsc.live <- root.voom5.trt.mean %>% filter(trt=="live",genotype=="FPsc")
# add ATG
root.voom5.trt.mean.FPsc.live.ATG.moduleColors.signedhybrid <- root.voom5.trt.mean.FPsc.live %>% inner_join(Br.v3.0anno.At.BLAST.highscore.moduleColors.signedhybrid,by=c("genes"="name"))
```

# calculate microbiome responsiveness FC (leaf)
```{r}
leaf.voom5.trt.mean <- leaf.voom5 %>% pivot_longer(col=-1) %>% inner_join(leaf.voom5.sample,by=c("name"="leaf.voom5")) %>% group_by(genes,genotype,trt) %>% summarize(mean.soilFC=mean(value))
leaf.voom5.trt.mean # scaled to zero, i.e., value in "dead" and one in "live" are the same absolute value.
leaf.voom5.trt.mean.FPsc.live <- leaf.voom5.trt.mean %>% filter(trt=="live",genotype=="FPsc")
# add ATG
leaf.voom5.trt.mean.FPsc.live.ATG.moduleColors.signedhybrid <- leaf.voom5.trt.mean.FPsc.live %>% inner_join(Br.v3.0anno.At.BLAST.highscore.moduleColors.signedhybrid,by=c("genes"="name"))
```

### the end of calculation of microbe FC

#  calculate microbe FC in each InX module (run once. Use output csv file for further)
```{r}
load(file.path("annotation","output","In01.consolidated.forORA.list2.v5.Rdata"))
load(file.path("annotation","output","In02orless.consolidated.forORA.list2.v5.Rdata"))
load(file.path("annotation","output","In03orless.consolidated.forORA.list2.v5.Rdata"))
load(file.path("annotation","output","In05orless.consolidated.forORA.list2.v5.Rdata"))
load(file.path("annotation","output","In10orless.consolidated.forORA.list2.v5.Rdata"))
load(file.path("annotation","output","In20orless.consolidated.forORA.list2.v5.Rdata"))
load(file.path("annotation","output","Total.consolidated.forGSEA.list2.v5.Rdata"))
load(file.path("annotation","output","In05orMore.consolidated.forORA.list2.v5.Rdata"))
load(file.path("annotation","output","In10orMore.consolidated.forORA.list2.v5.Rdata"))
# check loaded cosolidated custom categories
In01.consolidated.forORA.list2
In02orless.consolidated.forORA.list2
In03orless.consolidated.forORA.list2
In05orless.consolidated.forORA.list2
In10orless.consolidated.forORA.list2
Total.consolidated.forGSEA.list2
In05orMore.consolidated.forORA.list2
In10orMore.consolidated.forORA.list2
# check WGCNA modules
root.voom5.trt.mean.FPsc.live.ATG.moduleColors.signedhybrid
leaf.voom5.trt.mean.FPsc.live.ATG.moduleColors.signedhybrid
# create function
overlap.genes.WGCNAmodule.category2 <- function(tissue=c("root","leaf"),custom.category.list=In01.consolidated.forORA.list2) {
  if(tissue=="root") {
    temp.root <- custom.category.list %>% enframe(value="category") %>% unnest(category) %>%  inner_join(root.voom5.trt.mean.FPsc.live.ATG.moduleColors.signedhybrid,by=c("name"="genes")) #%>% View()
    summary(temp.root)
# test
    temp.root %>% filter(root.voom5.module.signedhybrid=="bisque4",category=="cold_root_FC_down") %>% print()
# summarize() of mean FC by root modules and custom categories
    temp.root.summary <- temp.root %>% group_by(root.voom5.module.signedhybrid,category) %>% dplyr::summarise(soilFC=mean(mean.soilFC)) #%>% View() # values are positive to negative. What does that mean?
return(temp.root.summary)
  } else if(tissue=="leaf") {
  temp.leaf <- custom.category.list %>% enframe(value="category") %>% unnest(category) %>%  inner_join(leaf.voom5.trt.mean.FPsc.live.ATG.moduleColors.signedhybrid,by=c("name"="genes")) #%>% View()
    summary(temp.leaf)
# summarize() of mean FC by root modules and custom categories
    temp.leaf.summary <- temp.leaf %>% group_by(leaf.voom5.module.signedhybrid,category) %>% dplyr::summarise(soilFC=mean(mean.soilFC)) #%>% View() # values are positive to negative. What does that mean?
return(temp.leaf.summary)
  
  } else {print("Please specify tissue")}
}
# calc (error)
root.In01 <- overlap.genes.WGCNAmodule.category2(custom.category.list=In01.consolidated.forORA.list2,tissue="root") %>% mutate(tissue="root",InX="In01") %>% dplyr::rename(module=root.voom5.module.signedhybrid)
leaf.In01 <- overlap.genes.WGCNAmodule.category2(custom.category.list=In01.consolidated.forORA.list2,tissue="leaf") %>% mutate(tissue="leaf",InX="In01") %>% dplyr::rename(module=leaf.voom5.module.signedhybrid)
root.In02orless <- overlap.genes.WGCNAmodule.category2(custom.category.list=In02orless.consolidated.forORA.list2,tissue="root") %>% mutate(tissue="root",InX="In02orless") %>% dplyr::rename(module=root.voom5.module.signedhybrid)
leaf.In02orless <- overlap.genes.WGCNAmodule.category2(custom.category.list=In02orless.consolidated.forORA.list2,tissue="leaf") %>% mutate(tissue="leaf",InX="In02orless") %>% dplyr::rename(module=leaf.voom5.module.signedhybrid)

root.In03orless <- overlap.genes.WGCNAmodule.category2(custom.category.list=In03orless.consolidated.forORA.list2,tissue="root") %>% mutate(tissue="root",InX="In03orless") %>% dplyr::rename(module=root.voom5.module.signedhybrid)
leaf.In03orless <- overlap.genes.WGCNAmodule.category2(custom.category.list=In03orless.consolidated.forORA.list2,tissue="leaf") %>% mutate(tissue="leaf",InX="In03orless") %>% dplyr::rename(module=leaf.voom5.module.signedhybrid)

root.In05orless <- overlap.genes.WGCNAmodule.category2(custom.category.list=In05orless.consolidated.forORA.list2,tissue="root") %>% mutate(tissue="root",InX="In05orless") %>% dplyr::rename(module=root.voom5.module.signedhybrid)
leaf.In05orless <- overlap.genes.WGCNAmodule.category2(custom.category.list=In05orless.consolidated.forORA.list2,tissue="leaf") %>% mutate(tissue="leaf",InX="In05orless") %>% dplyr::rename(module=leaf.voom5.module.signedhybrid)

root.In10orless <- overlap.genes.WGCNAmodule.category2(custom.category.list=In10orless.consolidated.forORA.list2,tissue="root") %>% mutate(tissue="root",InX="In10orless") %>% dplyr::rename(module=root.voom5.module.signedhybrid)
leaf.In10orless <- overlap.genes.WGCNAmodule.category2(custom.category.list=In10orless.consolidated.forORA.list2,tissue="leaf") %>% mutate(tissue="leaf",InX="In10orless") %>% dplyr::rename(module=leaf.voom5.module.signedhybrid)

root.In20orless <- overlap.genes.WGCNAmodule.category2(custom.category.list=In20orless.consolidated.forORA.list2,tissue="root") %>% mutate(tissue="root",InX="In20orless") %>% dplyr::rename(module=root.voom5.module.signedhybrid)
leaf.In20orless <- overlap.genes.WGCNAmodule.category2(custom.category.list=In10orless.consolidated.forORA.list2,tissue="leaf") %>% mutate(tissue="leaf",InX="In20orless") %>% dplyr::rename(module=leaf.voom5.module.signedhybrid)

root.Total <- overlap.genes.WGCNAmodule.category2(custom.category.list=Total.consolidated.forGSEA.list2,tissue="root") %>% mutate(tissue="root",InX="Total") %>% dplyr::rename(module=root.voom5.module.signedhybrid)
leaf.Total <- overlap.genes.WGCNAmodule.category2(custom.category.list=Total.consolidated.forGSEA.list2,tissue="leaf") %>% mutate(tissue="leaf",InX="Total") %>% dplyr::rename(module=leaf.voom5.module.signedhybrid)

root.In05orMore <- overlap.genes.WGCNAmodule.category2(custom.category.list=In05orMore.consolidated.forORA.list2,tissue="root") %>% mutate(tissue="root",InX="In05orMore") %>% dplyr::rename(module=root.voom5.module.signedhybrid)
leaf.In05orMore <- overlap.genes.WGCNAmodule.category2(custom.category.list=In05orMore.consolidated.forORA.list2,tissue="leaf") %>% mutate(tissue="leaf",InX="In05orMore") %>% dplyr::rename(module=leaf.voom5.module.signedhybrid)

root.In10orMore <- overlap.genes.WGCNAmodule.category2(custom.category.list=In10orMore.consolidated.forORA.list2,tissue="root") %>% mutate(tissue="root",InX="In10orMore") %>% dplyr::rename(module=root.voom5.module.signedhybrid)
leaf.In10orMore <- overlap.genes.WGCNAmodule.category2(custom.category.list=In10orMore.consolidated.forORA.list2,tissue="leaf") %>% mutate(tissue="leaf",InX="In10oreMore") %>% dplyr::rename(module=leaf.voom5.module.signedhybrid)
# combine all
InX.all.soilFC <- bind_rows(root.In01,leaf.In01,
                            root.In02orless,leaf.In02orless,
                            root.In03orless,leaf.In03orless,
                            root.In05orless,leaf.In05orless,
                            root.In10orless,leaf.In10orless,
                            root.In20orless,leaf.In20orless,
                            root.Total,leaf.Total,
                            root.In05orMore,leaf.In05orMore,
                            root.In10orMore,leaf.In10orMore) 
summary(InX.all.soilFC)
write_csv(InX.all.soilFC,file=file.path("RNA","output","InX.all.soilFC.csv"))
```

# start from this line. All above are prep for below.
# new plot with color coding of heat plot according to soil FC.
# separate plot for each module (Dec 21, 2021): giving up to play with facet_wrap()
# another trial by modifying function InX.comparison.plot() (Jan 4, 2022) -> Jan 18, 2022 -> Jan 28, 2022
# using soil FC
```{r}
# read soilFC
InX.all.soilFC <- read_csv(file.path("RNA","output","InX.all.soilFC.csv"))
# get data3
data3 <- read_csv(file.path("RNA","output","data3.csv"))
# full levels (for each tissue3 type?)
full.level.y <- c("In01","In02orless","In03orless","In05orless","In10orless","In20orless","Total","In05orMore","In10orMore","In20orMore")
# instead. Start from a complete x and y, and then combine with results
## where I can find full.level of category?
Total.categories <- read_csv(file=file.path("annotation","output","Total.consolidated1.v5.csv"))
##
full.level.x <- tibble(category=colnames(Total.categories))  %>% separate(category,into=c("category3","tissue3","FC","up_down"),sep="_") %>% 
  mutate(category3=category3 %>% 
           str_replace_all("MINUSMG","MG-") %>%
           str_replace_all("PLUSMG","MG+") %>%
           str_replace_all("MINUSFE","FE-") %>%
    str_replace_all("PLUSFE","FE_PLUS") %>%
  str_replace_all("MINUSPI-1D","PI-1D-") %>%
    str_replace_all("PLUSPI-1D","PI-1D+") %>%
  str_replace_all("MINUSPI-3D","PI-3D+") %>%
    str_replace_all("PLUSPI-3D","PI-3D+") %>%
  str_replace_all("MINUSN","N-") %>%
  str_replace_all("PLUSN","N+") %>%
  str_replace_all("MINUSMN","MN-") %>%
  str_replace_all("PLUSMN","MN+") %>% 
  str_replace_all("MINUSP","P-") %>% 
    str_replace_all("PLUSP","P+") %>% 
    str_replace_all("PLUSIAA","IAA+") %>% 
    str_replace_all("PLUSAL","AL+") %>%
    str_replace_all("MINUSCA","CA-") %>% 
      str_replace_all("PLUSCA","CA+") %>%
    str_replace_all("MINUSK","K-") %>% 
    str_replace_all("PLUSK","K+") %>% 
    str_replace_all("MINUSS","S-") %>% 
    str_replace_all("PLUSS","S+")) %>%
  drop_na(category3) %>%
  unite(category3.up_down,category3, up_down)
# check
full.level.x %>% View()
# heatmap plot function
InX.comparison.plot5.2 <- function(data=data3,soilFC=InX.all.soilFC,target.module="root.voom5.module.signedhybrid.darkred",level.x = full.level.x, level.y=full.level.y) {
  data <- data %>% mutate(NAME2=str_remove_all(NAME,"voom5.module.signedhybrid.")) #%>% separate(NAME2,into=c("tissue","module"))
  # unite
    data <- data %>% unite(col=tissue.module.category.InX, c(NAME2,tissue3,category3.up_down,InX),sep=".",remove=FALSE) # this tissue3 is for category, not for module.
  # modify soilFC
    soilFC <- soilFC %>% separate(category,into=c("category3","tissue3","FC","up_down")) %>% unite(category3.up_down,c(category3,up_down),sep="_") %>% unite(col=tissue.module.category.InX, c(tissue,module,tissue3,category3.up_down,InX),sep=".",remove=FALSE) #%>% View(). what is this tissue? tissue for modules.
    # combine data and soilFC
    data.combine <- data %>% inner_join(soilFC %>% dplyr::select(tissue.module.category.InX,soilFC),by="tissue.module.category.InX") #
    data.combine
    # modify over_represented_padjust according to soilFC
    data.combine <- data.combine %>% mutate(minuslog10.soilFC.over_represented_padjust=ifelse(soilFC>=0,-log10(over_represented_padjust),log10(over_represented_padjust))) #%>% View()
# target module
# instead enter FDR = 1 in non target.module
data.target <- data.combine %>% filter(NAME==target.module) #%>% dplyr::select(over_represented_padjust,NAME)
data.target %>% View()
# add missing categories + tissue3, which is 0 in minuslog10.soilFC.over_represented_padjust and FDR, 0 in numDEInCat (taht is overlap gene numbers)
temp <- level.x %>% mutate(minuslog10.soilFC.over_represented_padjust=0,over_represented_padjust=0,numDEInCat=0) 
temp.1 <- temp %>% mutate(InX=level.y[1])
temp.2 <- temp %>% mutate(InX=level.y[2])
temp.3 <- temp %>% mutate(InX=level.y[3])
temp.4 <- temp %>% mutate(InX=level.y[4])
temp.5 <- temp %>% mutate(InX=level.y[5])
temp.6 <- temp %>% mutate(InX=level.y[6])
temp.7 <- temp %>% mutate(InX=level.y[7])
temp.8 <- temp %>% mutate(InX=level.y[8])
temp.9 <- temp %>% mutate(InX=level.y[9])
temp.10 <- temp %>% mutate(InX=level.y[10])
level.all.InX <- bind_rows(temp.1,temp.2,temp.3,temp.4, temp.5, temp.6,temp.7,temp.8,temp.9,temp.10)
# using anti_join, unite category3.up_down,tissue3, and InX
data.target.mod <- data.target %>% unite(cat.tissue3.InX,category3.up_down,tissue3,InX,remove=FALSE)
# 
level.all.InX.mod <- level.all.InX %>% unite(cat.tissue3.InX,category3.up_down,tissue3,InX,remove=FALSE)
#
anti.temp <- level.all.InX.mod  %>% anti_join(data.target.mod,by="cat.tissue3.InX") %>% dplyr::select(colnames(level.all.InX.mod))
# combine
all.temp <- data.target.mod %>% dplyr::select(colnames(level.all.InX.mod)) %>% bind_rows(anti.temp)
# change InX levels
all.temp <- all.temp %>% mutate(InX=factor(InX,levels=level.y))
# check
all.temp %>% View()
str(all.temp)
# remove 0 in numDEInCat
all.temp <- all.temp %>% mutate(numDEInCat=ifelse(numDEInCat==0,"",numDEInCat)) #%>% View()
# plot 
WGCNA.voom5.signedhybrid.target.customcat.GOseq.plot.InXcomparison <- ggplot(all.temp, aes(x=category3.up_down,y=InX)) + geom_tile(aes(fill=minuslog10.soilFC.over_represented_padjust),colour="white") + scale_fill_gradient2(limit=c(-10,10), low="green",high=muted("magenta")) + facet_grid(.~tissue3,scales="free_x",space="free")
# 
WGCNA.voom5.signedhybrid.target.customcat.GOseq.plot.InXcomparison <- WGCNA.voom5.signedhybrid.target.customcat.GOseq.plot.InXcomparison + theme(axis.text.x=element_text(size=8,angle=90,hjust=1,vjust=0.3),
                                      axis.text.y=element_text(size=8),
                                      axis.title=element_text(size=9),
                                      axis.ticks = element_blank(),
                                      strip.text.y = element_text(size=16,angle=0),
                                      strip.text.x = element_text(size=8,angle=90),
                                      panel.background = element_rect(fill = "white",colour="black"),
                                      plot.title=element_text(size=8),
                                      axis.line=element_blank()) + 
      labs(x="",y="",fill="-log10\n FDR",title=target.module) # str_c("WGCNA.customcatv5.consolidated1.GOseq.plot.InXcomparison.\n",target.module))
# adding overlap numbers
WGCNA.voom5.signedhybrid.target.customcat.GOseq.plot.InXcomparison <- WGCNA.voom5.signedhybrid.target.customcat.GOseq.plot.InXcomparison + geom_text(aes(label=numDEInCat),size=2, angle=90)
# 
WGCNA.voom5.signedhybrid.target.customcat.GOseq.plot.InXcomparison
ggsave(path=file.path("RNA","output"),filename=paste("WGCNA.customcatv5.consol1.ORA.InXcomp.soilFC.",target.module,".pdf",sep=""),height=4,width=6.5) 
ggsave(path=file.path("RNA","output"),filename=paste("WGCNA.customcatv5.consol1.ORA.InXcomp.soilFC.",target.module,".png",sep=""),height=4,width=6.5) 

return(WGCNA.voom5.signedhybrid.target.customcat.GOseq.plot.InXcomparison)
}
# test 
p.test <- InX.comparison.plot5.2()
target.module <- "root.voom5.module.signedhybrid.darkred"
ggsave(path=file.path("RNA","output"),filename=paste("WGCNA.customcatv5.consol1.ORA.InXcomp.soilFC.",target.module,".pdf",sep=""),height=4,width=6.5) # legend needs to be improved. How?
```

# apply InX.comparison.plot5.2() to all modules 
```{r}
# traditional way using apply()
# data3 %>% distinct(NAME) %>% apply(.,1,function(x) InX.comparison.plot5.2(target.module=x))
# tidyverse way using map()
data3 %>% distinct(NAME) %>% as_vector() %>% map(function(x) InX.comparison.plot5.2(target.module=x))
```

# only Total (under construction, Jan 30, 2022)
```{r}
onlyTotal.soilFC <- InX.all.soilFC %>% filter(InX=="Total")
onlyTolal.data3 <- data3 %>% filter(InX=="Total")
# heatmap plot function
InX.onlyTotal.plot5.3 <- function(data=onlyTolal.data3,soilFC=onlyTotal.soilFC,target.module="root.voom5.module.signedhybrid.darkred",level.x = full.level.x, level.y="Total") {
  data <- data %>% mutate(NAME2=str_remove_all(NAME,"voom5.module.signedhybrid.")) #%>% separate(NAME2,into=c("tissue","module"))
  # unite
    data <- data %>% unite(col=tissue.module.category.InX, c(NAME2,tissue3,category3.up_down,InX),sep=".",remove=FALSE) # this tissue3 is for category, not for module.
  # modify soilFC
    soilFC <- soilFC %>% separate(category,into=c("category3","tissue3","FC","up_down")) %>% unite(category3.up_down,c(category3,up_down),sep="_") %>% unite(col=tissue.module.category.InX, c(tissue,module,tissue3,category3.up_down,InX),sep=".",remove=FALSE) #%>% View(). what is this tissue? tissue for modules.
    # combine data and soilFC
    data.combine <- data %>% inner_join(soilFC %>% dplyr::select(tissue.module.category.InX,soilFC),by="tissue.module.category.InX") #
    data.combine
    # modify over_represented_padjust according to soilFC
    data.combine <- data.combine %>% mutate(minuslog10.soilFC.over_represented_padjust=ifelse(soilFC>=0,-log10(over_represented_padjust),log10(over_represented_padjust))) #%>% View()
# target module 
    if(target.module=="all") { 
      print(paste("taget.module is ",target.module)) #all modules
      # add missing categories + tissue3, which is 0 in minuslog10.soilFC.over_represented_padjust and FDR, 0 in numDEInCat (taht is overlap gene numbers)
      temp <- level.x %>% mutate(minuslog10.soilFC.over_represented_padjust=0,over_represented_padjust=0,numDEInCat=0) 
      temp.1 <- temp %>% mutate(InX=level.y[1])
      # temp.1 has to be set for all modules or selected modules
      temp.1.list <-list()
      for(i in 1:length(unique(data$NAME2))) {
      temp.1.list[[i]] <- temp.1
      }
      names(temp.1.list) <- unique(data$NAME)
      # convert list into data.frame
      # temp.1.list %>% enframe() %>% deframe() # this is not what I want
      # temp.1.list %>%  as_tibble() # do not run
      temp.2 <- temp.1.list %>% bind_rows(.id="NAME") # list to data.frame (see my blog 2/9/2023)

      } else { 
         print(paste("taget.module is ",target.module)) # target modules
      data.combine <- data.combine %>% filter(NAME %in% target.module) # using filter for multiple target.module???? Does %in% work in filter() ?  (Feb 3, 2023)
      # add missing categories + tissue3, which is 0 in minuslog10.soilFC.over_represented_padjust and FDR, 0 in numDEInCat (taht is overlap gene numbers)
      temp <- level.x %>% mutate(minuslog10.soilFC.over_represented_padjust=0,over_represented_padjust=0,numDEInCat=0) 
      temp.1 <- temp %>% mutate(InX=level.y[1])
      # temp.1 has to be set for selected modules
 temp.1.list <-list()
      for(i in 1:length(target.module)) {
      temp.1.list[[i]] <- temp.1
      }
      names(temp.1.list) <- target.module
      # convert list into data.frame
      temp.2 <- temp.1.list %>% bind_rows(.id="NAME") # list to data.frame (see my blog 2/9/2023)
    }
data.combine %>% View()
# rename temp.2 and make NAME2 column
level.all.InX <- temp.2 %>% mutate(NAME2=str_remove_all(NAME,"voom5.module.signedhybrid.")) # temporary data.frame for all data is zero.

# using anti_join, unite category3.up_down,tissue3, and InX
data.combine.mod <- data.combine %>% unite(cat.tissue3.InX,category3.up_down,tissue3,InX,remove=FALSE)
# 
level.all.InX.mod <- level.all.InX %>% unite(cat.tissue3.InX,category3.up_down,tissue3,InX,remove=FALSE)
#
anti.temp <- level.all.InX.mod  %>% anti_join(data.combine.mod,by="cat.tissue3.InX") %>% dplyr::select(colnames(level.all.InX.mod))
# combine (also NAME2 for no target module version)
all.temp2 <- data.combine.mod %>% dplyr::select(colnames(level.all.InX.mod),NAME2) #

colnames(all.temp2) #
colnames(anti.temp) # should be the same as colnames(all.temp2), but...
all.temp3 <- all.temp2 %>% bind_rows(anti.temp) # anti.temp does not have "NAME2" why??? That is hwy all.temp3 has "NA" in NAME2
# change InX levels (no need)
#all.temp <- all.temp %>% mutate(InX=factor(InX,levels=level.y))
# check
all.temp3 %>% View() # "NA" in NAME2 (Feb 8, 2022)
str(all.temp3)
# remove 0 in numDEInCat
all.temp3 <- all.temp3 %>% mutate(numDEInCat=ifelse(numDEInCat==0,"",numDEInCat)) #%>% View()
# plot (leaf module + root tissue)
#### under construction ###
WGCNA.voom5.signedhybrid.target.customcat.GOseq.plot.Total <- ggplot(all.temp3, aes(x=category3.up_down,y=InX)) + geom_tile(aes(fill=minuslog10.soilFC.over_represented_padjust),colour="white") + scale_fill_gradient2(limit=c(-10,10), low="green",high=muted("magenta")) + facet_grid(NAME2~tissue3,scales="free_x",space="free")
# 
WGCNA.voom5.signedhybrid.target.customcat.GOseq.plot.Total <- WGCNA.voom5.signedhybrid.target.customcat.GOseq.plot.Total + theme(axis.text.x=element_text(size=8,angle=90,hjust=1,vjust=0.3),
                                      axis.text.y=element_text(size=8),
                                      axis.title=element_text(size=9),
                                      axis.ticks = element_blank(),
                                      strip.text.y = element_text(size=16,angle=0),
                                      strip.text.x = element_text(size=8,angle=90),
                                      panel.background = element_rect(fill = "white",colour="black"),
                                      plot.title=element_text(size=8),
                                      axis.line=element_blank()) + 
      labs(x="",y="",fill="-log10\n FDR") # 
# adding overlap numbers
WGCNA.voom5.signedhybrid.target.customcat.GOseq.plot.Total <- WGCNA.voom5.signedhybrid.target.customcat.GOseq.plot.Total + geom_text(aes(label=numDEInCat),size=2, angle=90)
# 
WGCNA.voom5.signedhybrid.target.customcat.GOseq.plot.Total
#ggsave(path=file.path("RNA","output"),filename=paste("WGCNA.customcatv5.consol1.ORA.onlyTotal.soilFC.",target.module,".pdf",sep=""),height=4,width=6.5) 
#ggsave(path=file.path("RNA","output"),filename=paste("onlyTotal",target.module,".png",sep=""),height=4,width=6.5) 

return(WGCNA.voom5.signedhybrid.target.customcat.GOseq.plot.Total)
}

InX.onlyTotal.plot5.3(data=data3,soilFC=onlyTotal.soilFC,target.module="root.voom5.module.signedhybrid.darkred",level.x = full.level.x, level.y="Total") # why "NA" in the plot?? (Feb 8, 2023)
#
# test 
p.test <- InX.onlyTotal.plot5.3(data=data3,soilFC=onlyTotal.soilFC,target.module="root.voom5.module.signedhybrid.darkred",level.x = full.level.x, level.y="Total")

ggsave(path=file.path("RNA","output"),filename=paste("WGCNA.customcatv5.consol1.ORA.onlyTotal.soilFC.pdf",sep=""),height=18,width=11) 
```

# make user friendly fig
# split plots with root or leaf modules
```{r}
InX.onlyTotal.plot5.3(data=data3,soilFC=onlyTotal.soilFC,target.module="all",level.x = full.level.x, level.y="Total") 
# split plots
# leaf modules
leaf.modules <- unique(data3$NAME) %>% str_subset("leaf")
leaf.plot <- InX.onlyTotal.plot5.3(data=data3,soilFC=onlyTotal.soilFC,target.module=leaf.modules,level.x = full.level.x, level.y="Total") 
```

# split plots with root/leaf modules and "tissue3" in custom categories
```{r}
# only leaf modules
leaf.modules <- unique(data3$NAME) %>% str_subset("leaf")
# only root modules
root.modules <- unique(data3$NAME) %>% str_subset("root")
# only root category
onlyTolal.data3.rootcategory <- onlyTolal.data3 %>% filter(str_detect("root",tissue3))
onlyTolal.data3.rootcategory %>% View()
# only non root category
onlyTolal.data3.nonrootcategory <- onlyTolal.data3 %>% filter(!str_detect("root",tissue3))
onlyTolal.data3.nonrootcategory %>% View()
#onlyTotal.soilFC.rootcategory <- onlyTotal.soilFC %>% filter(str_detect("root",tissue))
root.level.x <- full.level.x %>% filter(str_detect("root",tissue3))
nonroot.level.x <- full.level.x %>% filter(!str_detect("root",tissue3))

# plot
leaf.rootcategory.plot <- InX.onlyTotal.plot5.3(data=onlyTolal.data3.rootcategory,soilFC=onlyTotal.soilFC,target.module=leaf.modules,level.x = root.level.x, level.y="Total") # does not work well
# plot
leaf.nonrootcategory.plot <- InX.onlyTotal.plot5.3(data=onlyTolal.data3.nonrootcategory,soilFC=onlyTotal.soilFC,target.module=leaf.modules,level.x = nonroot.level.x, level.y="Total") # does not work well
# plot
root.rootcategory.plot <- InX.onlyTotal.plot5.3(data=onlyTolal.data3.rootcategory,soilFC=onlyTotal.soilFC,target.module=root.modules,level.x = root.level.x, level.y="Total") # does not work well
# plot
root.nonrootcategory.plot <- InX.onlyTotal.plot5.3(data=onlyTolal.data3.nonrootcategory,soilFC=onlyTotal.soilFC,target.module=root.modules,level.x = nonroot.level.x, level.y="Total") # does not work well
# combine plots
plot.all <- plot_grid(leaf.rootcategory.plot,leaf.nonrootcategory.plot,root.rootcategory.plot,root.nonrootcategory.plot, labels=c("A","B","C","D"),label_size=12,ncol=2)
# pdf
ggsave(plot.all,path=file.path("RNA","output"),filename="WGCNA.customcatv5.consol1.ORA.Total.soilFC.allmodules.pdf",height=22,width=16)
# png
ggsave(plot.all,path=file.path("RNA","output"),filename="WGCNA.customcatv5.consol1.ORA.Total.soilFC.allmodules.png",height=22,width=16)
```





```{r}
sessionInfo()
```


