---
title: "Br.microbiom.exp1.v3.0.annotationDEG.2018v1"
author: "Kazu"
date: "9/25/2018"
output: 
  html_document: 
    keep_md: yes
editor_options: 
  chunk_output_type: console
---
# purpose: Raw count files to DEGs
# History
* For reorganization, create this script from scratch. (092518)
* Use only FULLtoptag (all gene list) (021219)
* Adding interaction term (only live:genotype) to interaction model (021319)
* Try quasi-likelihood (QL) F-Test (see edgeRUsersGuide2018 section 2.10.3)
* Adding test method description (NB or QLF) to DEG csv files (070919)
```{r}
knitr::opts_chunk$set(echo = TRUE,error=TRUE)
```
```{r}
library(edgeR)
library(tidyverse)
library(readr)
library(readxl)
# home directry should be root of this repository (not location of this script)
```
# For Mapping reads with Kalisoga see "Map and normalize.Rmd" in "./v3.0annotation/20170617-samples/scripts" 
# load reads mapped to v3.0 Brassica genome
```{r}
getwd()
counts <- readr::read_csv(file.path("..","input","20170617_V3.0_raw_counts.csv.gz"))
counts # make sure this is v3.0 annotation (look target_id column)
```
## make sample description data frame (copied from "Map and normalized.Rmd")
```{r}
sample.description <- tibble(sample=colnames(counts)[-1]) %>%
  separate(sample,
           c("location","tissue","genotype","block","pot"),
           remove=FALSE,
           convert=TRUE) 
head(sample.description) 
## get additional metadata
sample.info<-readxl::read_excel(file.path("..","input","wy001_RNAseq.xlsx"),sheet = 1)
head(sample.info)
##combine
sample.description <- left_join(sample.description, sample.info)
sample.description <- sample.description %>% 
  mutate(group=paste(tissue,genotype,soil,autoclave,sep="_")) %>%
  unite(trt,c(soil,autoclave),sep="_") # add trt (092718)

head(sample.description)
# block has to be treated as descrite.
sample.description<-sample.description %>% mutate(block=str_c("b", as.integer(block))) ## get additional metadata. (Oct 14, 2019) Block effects have to be character, not integer!!!!

write_csv(sample.description,path="../output/Br.mbio.e1.sample.description.csv")
sample.description %>% summarize(n_distinct(group))
```
## summarize counts
```{r}
pl.orig <- counts[,-1] %>% colSums() %>% tibble(sample=names(.),count=.) %>%
  ggplot(aes(x=sample,y=count)) + 
  geom_col() +
  theme(axis.text.x  = element_text(angle=90, vjust=0.5,size = 7)) 
pl.orig
```
## load to edgeR, normalize
```{r}
#confirm that everthing is in the right order
all(colnames(counts)[-1]==sample.description$sample)
dge <- DGEList(counts[,-1],
               group=sample.description$group,
               samples=sample.description,
               genes=counts$target_id)
```
# 
```{r}
dge <- calcNormFactors(dge)
barplot(dge$samples$lib.size)
ggplot(dge$samples,aes(x=sample,y=norm.factors,fill=tissue)) + geom_col() + 
  theme(axis.text.x  = element_text(angle=90, vjust=0.5,size = 7)) 
ggplot(dge$samples,aes(x=sample,y=norm.factors,fill=genotype)) + geom_col() + 
  theme(axis.text.x  = element_text(angle=90, vjust=0.5,size = 7)) 
ggplot(dge$samples,aes(x=sample,y=norm.factors,fill=as.factor(block))) + geom_col() +
  theme(axis.text.x  = element_text(angle=90, vjust=0.5,size = 7)) 
```


Looks like we should normalize separately for root and leaf
# do separately for leaf and root values
```{r}
counts.leaf <- counts %>% select(target_id, contains("leaf"))
counts.root <- counts %>% select(target_id, contains("root"))
sample.description.leaf <- sample.description %>% filter(tissue=="leaf")
sample.description.root <- sample.description %>% filter(tissue=="root")
```

# Leaf
```{r}
#confirm that everthing is in the right order
all(colnames(counts.leaf)[-1]==sample.description.leaf$sample)
dge.leaf <- DGEList(counts.leaf[,-1],
                    group=sample.description.leaf$group,
                    samples=sample.description.leaf,
                    genes=counts.leaf$target_id)
# dge.leaf <- calcNormFactors(dge.leaf)
# # remove genes with low expression level
# dge.leaf<-dge.leaf[rowSums(cpm(dge.leaf)>1) >= 6,,keep.lib.sizes=FALSE]
# Kazu's new way:  keep > normalize (012319)
# Prevent repeated normalization
ifelse(all(dge.leaf$samples$norm.factors==1),"Normalization has not yet done","Normalization has been done already.") # 
dge.leaf$samples$norm.factors<-1 # No matter which case, non-normalize libraries.
# remove genes with low expression level
dge.leaf<-dge.leaf[rowSums(cpm(dge.leaf)>1) >= 6,,keep.lib.sizes=FALSE]
# normalize
dge.leaf <- calcNormFactors(dge.leaf)

```

# Root
```{r}
#confirm that everthing is in the right order
all(colnames(counts.root)[-1]==sample.description.root$sample)
dge.root <- DGEList(counts.root[,-1],
                    group=sample.description.root$group,
                    samples=sample.description.root,
                    genes=counts.root$target_id)
# dge.root <- calcNormFactors(dge.root)
# # remove genes with low expression level
# dge.root<-dge.root[rowSums(cpm(dge.root)>1) >= 6,,keep.lib.sizes=FALSE]
# Kazu's new way:  keep > normalize (012319)
# Prevent repeated normalization
ifelse(all(dge.root$samples$norm.factors==1),"Normalization has not yet done","Normalization has been done already.") # 
dge.root$samples$norm.factors<-1 # No matter which case, non-normalize libraries.
# remove genes with low expression level
dge.root<-dge.root[rowSums(cpm(dge.root)>1) >= 6,,keep.lib.sizes=FALSE]
# normalize
dge.root <- calcNormFactors(dge.root)
```
# save edgeR objects
```{r}
save(dge.leaf,dge.root,sample.description.leaf,sample.description.root,file=file.path("..","output","edgeR_dge_objects.e1.v3.0anno.RData")) # Question: dge has sample info. Do I need to ssave sample.descritipon.leaf or .root? (092718)
```
# leaf MDS plot bcv
```{r}
# leaf
mds.leaf <- plotMDS(dge.leaf,method = "bcv",labels=dge.leaf$samples$group,gene.selection = "pairwise",dim.plot = c(1,2),ndim=5)
mds.leaf.pl <- as_tibble(mds.leaf$cmdscale.out) %>% 
  bind_cols(data.frame(sample=row.names(mds.leaf$cmdscale.out)),.) %>%
  inner_join(dge.leaf$samples)
p<-mds.leaf.pl %>% ggplot(aes(x=V1,y=V2, shape=genotype, color=trt,label=genotype)) + geom_point(size=3) + ggtitle("Leaf Samples DIM 1 vs 2 bcv")+ facet_grid(.~block) + theme_gray()
ggsave(p,filename="../output/leaf.mds.bcv.1and2.png",width=6,height=8)
mds.leaf.pl %>% ggplot(aes(x=V1,y=V2, shape=genotype, color=trt,label=genotype)) + geom_point(size=3) + ggtitle("Leaf Samples DIM 1 vs 2 bcv")+ facet_grid(.~block)
mds.leaf.pl %>% ggplot(aes(x=V2,y=V3, shape=genotype, color=trt,label=genotype)) + geom_point(size=3) + ggtitle("Leaf Samples DIM 2 vs 3 bcv")+ facet_grid(.~block)
mds.leaf.pl %>% ggplot(aes(x=V1,y=V3, shape=genotype, color=trt,label=genotype)) + geom_point(size=3) + ggtitle("Leaf Samples DIM 1 vs 3 bcv")+ facet_grid(.~block)
mds.leaf.pl %>% ggplot(aes(x=V1,y=V4, shape=genotype, color=trt,label=genotype)) + geom_point(size=3) + ggtitle("Leaf Samples DIM 1 vs 4 bcv")+ facet_grid(.~block)
mds.leaf.pl %>% ggplot(aes(x=V1,y=V5, shape=genotype, color=trt,label=genotype)) + geom_point(size=3) + ggtitle("Leaf Samples DIM 1 vs 5")+ facet_grid(.~block)

```

# leaf MDS logFC
```{r}
mds.leaf.logFC <- plotMDS(dge.leaf,method = "logFC",labels=dge.leaf$samples$group,gene.selection = "pairwise",dim.plot = c(1,2),ndim=5) # changed ndim=3 into 5
mds.leaf.pl.logFC <- as_tibble(mds.leaf.logFC$cmdscale.out) %>% 
  bind_cols(data.frame(sample=row.names(mds.leaf.logFC$cmdscale.out)),.) %>%
  inner_join(dge.leaf$samples)
# adding block info
mds.leaf.pl.logFC %>% ggplot(aes(x=V1,y=V2, shape=genotype, color=trt,label=genotype)) + geom_point(size=3) + ggtitle("Leaf Samples DIM 1 vs 2, logFC") + facet_grid(.~block)
mds.leaf.pl.logFC %>% ggplot(aes(x=V1,y=V3, shape=genotype, color=trt,label=genotype)) + geom_point(size=3) + ggtitle("Leaf Samples DIM 1 vs 3, logFC")+ facet_grid(.~block)
mds.leaf.pl.logFC %>% ggplot(aes(x=V1,y=V4, shape=genotype, color=trt,label=genotype)) + geom_point(size=3) + ggtitle("Leaf Samples DIM 1 vs 4, logFC")+ facet_grid(.~block)
mds.leaf.pl.logFC %>% ggplot(aes(x=V1,y=V5, shape=genotype, color=trt,label=genotype)) + geom_point(size=3) + ggtitle("Leaf Samples DIM 1 vs 5, logFC")+ facet_grid(.~block)
```

# root MDS plot bcv
```{r}
# root
mds.root <- plotMDS(dge.root,method = "bcv",labels=dge.root$samples$group,gene.selection = "pairwise",dim.plot = c(1,2),ndim=5)
mds.root.pl <- as_tibble(mds.root$cmdscale.out) %>% 
  bind_cols(data.frame(sample=row.names(mds.root$cmdscale.out)),.) %>%
  inner_join(dge.root$samples)
mds.root.pl %>% ggplot(aes(x=V1,y=V2, shape=genotype, color=trt,label=genotype)) + geom_point(size=3) + ggtitle("root Samples DIM 1 vs 2 bcv")+ facet_grid(.~block)
mds.root.pl %>% ggplot(aes(x=V2,y=V3, shape=genotype, color=trt,label=genotype)) + geom_point(size=3) + ggtitle("root Samples DIM 2 vs 3 bcv")+ facet_grid(.~block)
mds.root.pl %>% ggplot(aes(x=V1,y=V3, shape=genotype, color=trt,label=genotype)) + geom_point(size=3) + ggtitle("root Samples DIM 1 vs 3 bcv")+ facet_grid(.~block)
mds.root.pl %>% ggplot(aes(x=V1,y=V4, shape=genotype, color=trt,label=genotype)) + geom_point(size=3) + ggtitle("root Samples DIM 1 vs 4 bcv")+ facet_grid(.~block)
mds.root.pl %>% ggplot(aes(x=V1,y=V5, shape=genotype, color=trt,label=genotype)) + geom_point(size=3) + ggtitle("root Samples DIM 1 vs 5")+ facet_grid(.~block)

```

# root MDS logFC
```{r}
mds.root.logFC <- plotMDS(dge.root,method = "logFC",labels=dge.root$samples$group,gene.selection = "pairwise",dim.plot = c(1,2),ndim=5) # changed ndim=3 into 5
mds.root.pl.logFC <- as_tibble(mds.root.logFC$cmdscale.out) %>% 
  bind_cols(data.frame(sample=row.names(mds.root.logFC$cmdscale.out)),.) %>%
  inner_join(dge.root$samples)
# adding block info
mds.root.pl.logFC %>% ggplot(aes(x=V1,y=V2, shape=genotype, color=trt,label=genotype)) + geom_point(size=3) + ggtitle("root Samples DIM 1 vs 2, logFC") + facet_grid(.~block)
mds.root.pl.logFC %>% ggplot(aes(x=V1,y=V3, shape=genotype, color=trt,label=genotype)) + geom_point(size=3) + ggtitle("root Samples DIM 1 vs 3, logFC")+ facet_grid(.~block)
mds.root.pl.logFC %>% ggplot(aes(x=V1,y=V4, shape=genotype, color=trt,label=genotype)) + geom_point(size=3) + ggtitle("root Samples DIM 1 vs 4, logFC")+ facet_grid(.~block)
mds.root.pl.logFC %>% ggplot(aes(x=V1,y=V5, shape=genotype, color=trt,label=genotype)) + geom_point(size=3) + ggtitle("root Samples DIM 1 vs 5, logFC")+ facet_grid(.~block)
```


# DEGs
```{r include=FALSE}
# clean working directory
rm(list=ls())
knitr::opts_chunk$set(echo = TRUE,error=TRUE)
library(edgeR)
library(tidyverse)
# annotation file for v3.0annotation
Br.v3.0.At.BLAST <- read_csv(file.path("..","..","..","Annotation","output","v3.0annotation","Brapa_v3.0_annotated.csv")) 
# This annotation is redundant with name (Br grene). Eg 
Br.v3.0.At.BLAST %>% filter(name=="BraA01g040570.3C")
# reduce the redundancy (112418)
Br.v3.0anno.At.BLAST.highscore <- Br.v3.0.At.BLAST %>% group_by(name) %>% arrange(desc(score)) %>% dplyr::slice(1)

# load exp1 v3.0 annotation dge files and sample info
load(file.path("..","output","edgeR_dge_objects.e1.v3.0anno.RData"))
```
# Additive model, rD.rFPsc, Negative biomial (NB) method (old method)
```{r}
# relevel trt and genotype to "5E_dead" and "FPsc"
dge.root$samples<-dge.root$samples %>% mutate(trt=fct_relevel(trt,"5E_dead")) #%>% str() 
dge.root$samples<-dge.root$samples %>% mutate(genotype=fct_relevel(genotype,"FPsc")) #%>% str() 
dge.leaf$samples<-dge.leaf$samples %>% mutate(trt=fct_relevel(trt,"5E_dead")) #%>% str() 
dge.leaf$samples<-dge.leaf$samples %>% mutate(genotype=fct_relevel(genotype,"FPsc")) #%>% str() 
## dge
root.design.add <- with(dge.root$samples, model.matrix(~ genotype + trt + block))
leaf.design.add <- with(dge.leaf$samples, model.matrix(~ genotype + trt + block))
# estimateDisp
root.dge.add <- estimateDisp(dge.root,design = root.design.add)
leaf.dge.add <- estimateDisp(dge.leaf,design = leaf.design.add)
## fit linear model
root.fit.add <- glmFit(root.dge.add,root.design.add)
leaf.fit.add <- glmFit(leaf.dge.add,leaf.design.add)
# get DEGs, trt effects (coef=c("trt5E_live"))
root.trt5E_live.lrt <- glmLRT(root.fit.add,coef = c("trt5E_live"))
leaf.trt5E_live.lrt <- glmLRT(leaf.fit.add,coef = c("trt5E_live"))
topTags(root.trt5E_live.lrt);topTags(leaf.trt5E_live.lrt)
# all
root.trt5E_live.DEGs.add.all <- topTags(root.trt5E_live.lrt,n = Inf,p.value = 1)$table
leaf.trt5E_live.DEGs.add.all <- topTags(leaf.trt5E_live.lrt,n = Inf,p.value = 1)$table 
# get DEGs, genotype effects (coef = c("genotypeR500"))
root.genotype.lrt <- glmLRT(root.fit.add,coef = c("genotypeR500"))
leaf.genotype.lrt <- glmLRT(leaf.fit.add,coef = c("genotypeR500"))
topTags(root.genotype.lrt);topTags(leaf.genotype.lrt)
# all
root.genotype.DEGs.add.all <- topTags(root.genotype.lrt,n = Inf,p.value = 1)$table
leaf.genotype.DEGs.add.all <- topTags(leaf.genotype.lrt,n = Inf,p.value = 1)$table
# add annotiona with full name objects (all)
root.genotype.DEGs.add.all.rD.rFPsc.v3.0anno <- left_join(root.genotype.DEGs.add.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>%  select(1:6, AGI, At_symbol, At_short_description, perc_ID)
leaf.genotype.DEGs.add.all.rD.rFPsc.v3.0anno <- left_join(leaf.genotype.DEGs.add.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>% select(1:6, AGI, At_symbol, At_short_description, perc_ID)
root.trt5E_live.DEGs.add.all.rD.rFPsc.v3.0anno <- left_join(root.trt5E_live.DEGs.add.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>%  select(1:6, AGI, At_symbol, At_short_description, perc_ID)
leaf.trt5E_live.DEGs.add.all.rD.rFPsc.v3.0anno <- left_join(leaf.trt5E_live.DEGs.add.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>% select(1:6, AGI, At_symbol, At_short_description, perc_ID)
# write csv (all)
## genotype
write.csv(root.genotype.DEGs.add.all.rD.rFPsc.v3.0anno,file=file.path("..","output","FULLtoptag","root.genotypeR500.DEGs.add.rD.rFPsc.NB.v3.0anno.csv"))
write.csv(leaf.genotype.DEGs.add.all.rD.rFPsc.v3.0anno,file=file.path("..","output","FULLtoptag","leaf.genotypeR500.DEGs.add.rD.rFPsc.NB.v3.0anno.csv"))
## trt
write.csv(root.trt5E_live.DEGs.add.all.rD.rFPsc.v3.0anno,file=file.path("..","output","FULLtoptag","root.trt5E_live.DEGs.add.rD.rFPsc.NB.v3.0anno.csv"))
write.csv(leaf.trt5E_live.DEGs.add.all.rD.rFPsc.v3.0anno,file=file.path("..","output","FULLtoptag","leaf.trt5E_live.DEGs.add.rD.rFPsc.NB.v3.0anno.csv"))

```
# cleanup working directory
```{r include=FALSE}
rm(list=ls())
knitr::opts_chunk$set(echo = TRUE,error=TRUE)
library(edgeR);library(tidyverse)
# annotation file for v3.0annotation
Br.v3.0.At.BLAST <- read_csv(file.path("..","..","..","Annotation","output","v3.0annotation","Brapa_v3.0_annotated.csv")) 
# This annotation is redundant with name (Br grene). Eg 
Br.v3.0.At.BLAST %>% filter(name=="BraA01g040570.3C")
# reduce the redundancy (112418)
Br.v3.0anno.At.BLAST.highscore <- Br.v3.0.At.BLAST %>% group_by(name) %>% arrange(desc(score)) %>% dplyr::slice(1)

# load exp1 v3.0 annotation dge files and sample info
load(file.path("..","output","edgeR_dge_objects.e1.v3.0anno.RData"))
```

# Additive model, rB.rFPsc
```{r}
# relevel trt and genotype to "BLANK_NA" and "FPsc"
dge.root$samples<-dge.root$samples %>% mutate(trt=fct_relevel(trt,"BLANK_NA")) #%>% str() 
dge.root$samples<-dge.root$samples %>% mutate(genotype=fct_relevel(genotype,"FPsc")) #%>% str() 
dge.leaf$samples<-dge.leaf$samples %>% mutate(trt=fct_relevel(trt,"BLANK_NA")) #%>% str() 
dge.leaf$samples<-dge.leaf$samples %>% mutate(genotype=fct_relevel(genotype,"FPsc")) #%>% str() 
## dge
root.design.add <- with(dge.root$samples, model.matrix(~ genotype + trt + block))
leaf.design.add <- with(dge.leaf$samples, model.matrix(~ genotype + trt + block))
# estimateDisp
root.dge.add <- estimateDisp(dge.root,design = root.design.add)
leaf.dge.add <- estimateDisp(dge.leaf,design = leaf.design.add)
## fit linear model
root.fit.add <- glmFit(root.dge.add,root.design.add)
leaf.fit.add <- glmFit(leaf.dge.add,leaf.design.add)
# get DEGs, trt effects (coef=c("trt5E_live"))
root.trt5E_live.lrt <- glmLRT(root.fit.add,coef = c("trt5E_live"))
leaf.trt5E_live.lrt <- glmLRT(leaf.fit.add,coef = c("trt5E_live"))
topTags(root.trt5E_live.lrt);topTags(leaf.trt5E_live.lrt)
# all
root.trt5E_live.DEGs.add.all <- topTags(root.trt5E_live.lrt,n = Inf,p.value = 1)$table
leaf.trt5E_live.DEGs.add.all <- topTags(leaf.trt5E_live.lrt,n = Inf,p.value = 1)$table

# get DEGs, trt effects (coef=c("trt5E_dead"))
root.trt5E_dead.lrt <- glmLRT(root.fit.add,coef = c("trt5E_dead"))
leaf.trt5E_dead.lrt <- glmLRT(leaf.fit.add,coef = c("trt5E_dead"))
topTags(root.trt5E_dead.lrt);topTags(leaf.trt5E_dead.lrt)
## all
root.trt5E_dead.DEGs.add.all <- topTags(root.trt5E_dead.lrt,n = Inf,p.value = 1)$table
leaf.trt5E_dead.DEGs.add.all <- topTags(leaf.trt5E_dead.lrt,n = Inf,p.value = 1)$table
# get DEGs, genotype effects (coef = c("genotypeR500"))
root.genotype.lrt <- glmLRT(root.fit.add,coef = c("genotypeR500"))
leaf.genotype.lrt <- glmLRT(leaf.fit.add,coef = c("genotypeR500"))
topTags(root.genotype.lrt);topTags(leaf.genotype.lrt)
## all
root.genotype.DEGs.add.all <- topTags(root.genotype.lrt,n = Inf,p.value = 1)$table
leaf.genotype.DEGs.add.all <- topTags(leaf.genotype.lrt,n = Inf,p.value = 1)$table

# add annotiona with full name objects (all)
root.genotype.DEGs.add.all.rB.rFPsc.v3.0anno <- left_join(root.genotype.DEGs.add.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>%  select(1:6, AGI, At_symbol, At_short_description, perc_ID)
leaf.genotype.DEGs.add.all.rB.rFPsc.v3.0anno <- left_join(leaf.genotype.DEGs.add.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>% select(1:6, AGI, At_symbol, At_short_description, perc_ID)
root.trt5E_live.DEGs.add.all.rB.rFPsc.v3.0anno <- left_join(root.trt5E_live.DEGs.add.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>%  select(1:6, AGI, At_symbol, At_short_description, perc_ID)
leaf.trt5E_live.DEGs.add.all.rB.rFPsc.v3.0anno <- left_join(leaf.trt5E_live.DEGs.add.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>% select(1:6, AGI, At_symbol, At_short_description, perc_ID)
root.trt5E_dead.DEGs.add.all.rB.rFPsc.v3.0anno <- left_join(root.trt5E_dead.DEGs.add.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>%  select(1:6, AGI, At_symbol, At_short_description, perc_ID)
leaf.trt5E_dead.DEGs.add.all.rB.rFPsc.v3.0anno <- left_join(leaf.trt5E_dead.DEGs.add.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>% select(1:6, AGI, At_symbol, At_short_description, perc_ID)
# write csv (all)
## genotype
write.csv(root.genotype.DEGs.add.all.rB.rFPsc.v3.0anno,file=file.path("..","output","FULLtoptag","root.genotypeR500.DEGs.add.rB.rFPsc.NB.v3.0anno.csv"))
write.csv(leaf.genotype.DEGs.add.all.rB.rFPsc.v3.0anno,file=file.path("..","output","FULLtoptag","leaf.genotypeR500.DEGs.add.rB.rFPsc.NB.v3.0anno.csv"))
## trt5E_live
write.csv(root.trt5E_live.DEGs.add.all.rB.rFPsc.v3.0anno,file=file.path("..","output","FULLtoptag","root.trt5E_live.DEGs.add.rB.rFPsc.NB.v3.0anno.csv"))
write.csv(leaf.trt5E_live.DEGs.add.all.rB.rFPsc.v3.0anno,file=file.path("..","output","FULLtoptag","leaf.trt5E_live.DEGs.add.rB.rFPsc.NB.v3.0anno.csv"))
## trt5E_dead
write.csv(root.trt5E_dead.DEGs.add.all.rB.rFPsc.v3.0anno,file=file.path("..","output","FULLtoptag","root.trt5E_dead.DEGs.add.rB.rFPsc.NB.v3.0anno.csv"))
write.csv(leaf.trt5E_dead.DEGs.add.all.rB.rFPsc.v3.0anno,file=file.path("..","output","FULLtoptag","leaf.trt5E_dead.DEGs.add.rB.rFPsc.NB.v3.0anno.csv"))

```
# clean working directory
```{r include=FALSE}
rm(list=ls())
knitr::opts_chunk$set(echo = TRUE,error=TRUE)
library(edgeR);library(tidyverse)
# annotation file for v3.0annotation
Br.v3.0.At.BLAST <- read_csv(file.path("..","..","..","Annotation","output","v3.0annotation","Brapa_v3.0_annotated.csv")) 
# This annotation is redundant with name (Br grene). Eg 
Br.v3.0.At.BLAST %>% filter(name=="BraA01g040570.3C")
# reduce the redundancy (112418)
Br.v3.0anno.At.BLAST.highscore <- Br.v3.0.At.BLAST %>% group_by(name) %>% arrange(desc(score)) %>% dplyr::slice(1)
# load exp1 v3.0 annotation dge files and sample info
load(file.path("..","output","edgeR_dge_objects.e1.v3.0anno.RData"))
```
# Additive model, rD.rR500
```{r}
# relevel trt and genotype to "5E_dead" and "R500"
dge.root$samples<-dge.root$samples %>% mutate(trt=fct_relevel(trt,"5E_dead")) #%>% str() 
dge.root$samples<-dge.root$samples %>% mutate(genotype=fct_relevel(genotype,"R500")) #%>% str() 
dge.leaf$samples<-dge.leaf$samples %>% mutate(trt=fct_relevel(trt,"5E_dead")) #%>% str() 
dge.leaf$samples<-dge.leaf$samples %>% mutate(genotype=fct_relevel(genotype,"R500")) #%>% str() 
## dge
root.design.add <- with(dge.root$samples, model.matrix(~ genotype + trt + block))
leaf.design.add <- with(dge.leaf$samples, model.matrix(~ genotype + trt + block))
# estimateDisp
root.dge.add <- estimateDisp(dge.root,design = root.design.add)
leaf.dge.add <- estimateDisp(dge.leaf,design = leaf.design.add)
## fit linear model
root.fit.add <- glmFit(root.dge.add,root.design.add)
leaf.fit.add <- glmFit(leaf.dge.add,leaf.design.add)
# get DEGs, trt effects (coef=c("trt5E_live"))
root.trt5E_live.lrt <- glmLRT(root.fit.add,coef = c("trt5E_live"))
leaf.trt5E_live.lrt <- glmLRT(leaf.fit.add,coef = c("trt5E_live"))
topTags(root.trt5E_live.lrt);topTags(leaf.trt5E_live.lrt)
## all
root.trt5E_live.DEGs.add.all <- topTags(root.trt5E_live.lrt,n = Inf,p.value = 1)$table
leaf.trt5E_live.DEGs.add.all <- topTags(leaf.trt5E_live.lrt,n = Inf,p.value = 1)$table
# get DEGs, genotype effects (coef = c("genotypeFPsc"))
root.genotype.lrt <- glmLRT(root.fit.add,coef = c("genotypeFPsc"))
leaf.genotype.lrt <- glmLRT(leaf.fit.add,coef = c("genotypeFPsc"))
topTags(root.genotype.lrt);topTags(leaf.genotype.lrt)
## all
root.genotype.DEGs.add.all <- topTags(root.genotype.lrt,n = Inf,p.value = 1)$table
leaf.genotype.DEGs.add.all <- topTags(leaf.genotype.lrt,n = Inf,p.value = 1)$table
# add annotiona with full name objects (all)
root.genotype.DEGs.add.all.rD.rR500.v3.0anno <- left_join(root.genotype.DEGs.add.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>%  select(1:6, AGI, At_symbol, At_short_description, perc_ID)
leaf.genotype.DEGs.add.all.rD.rR500.v3.0anno <- left_join(leaf.genotype.DEGs.add.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>% select(1:6, AGI, At_symbol, At_short_description, perc_ID)
root.trt5E_live.DEGs.add.all.rD.rR500.v3.0anno <- left_join(root.trt5E_live.DEGs.add.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>%  select(1:6, AGI, At_symbol, At_short_description, perc_ID)
leaf.trt5E_live.DEGs.add.all.rD.rR500.v3.0anno <- left_join(leaf.trt5E_live.DEGs.add.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>% select(1:6, AGI, At_symbol, At_short_description, perc_ID)

# write csv (all)
## genotype
write.csv(root.genotype.DEGs.add.all.rD.rR500.v3.0anno,file=file.path("..","output","FULLtoptag","root.genotypeFPsc.DEGs.add.rD.rR500.NB.v3.0anno.csv"))
write.csv(leaf.genotype.DEGs.add.all.rD.rR500.v3.0anno,file=file.path("..","output","FULLtoptag","leaf.genotypeFPsc.DEGs.add.rD.rR500.NB.v3.0anno.csv"))
## trt
write.csv(root.trt5E_live.DEGs.add.all.rD.rR500.v3.0anno,file=file.path("..","output","FULLtoptag","root.trt5E_live.DEGs.add.rD.rR500.NB.v3.0anno.csv"))
write.csv(leaf.trt5E_live.DEGs.add.all.rD.rR500.v3.0anno,file=file.path("..","output","FULLtoptag","leaf.trt5E_live.DEGs.add.rD.rR500.NB.v3.0anno.csv"))

```
# clean working directory
```{r include=FALSE}
rm(list=ls())
knitr::opts_chunk$set(echo = TRUE,error=TRUE)
library(edgeR);library(tidyverse)
# annotation file for v3.0annotation
Br.v3.0.At.BLAST <- read_csv(file.path("..","..","..","Annotation","output","v3.0annotation","Brapa_v3.0_annotated.csv")) 
# This annotation is redundant with name (Br grene). Eg 
Br.v3.0.At.BLAST %>% filter(name=="BraA01g040570.3C")
# reduce the redundancy (112418)
Br.v3.0anno.At.BLAST.highscore <- Br.v3.0.At.BLAST %>% group_by(name) %>% arrange(desc(score)) %>% dplyr::slice(1)
# load exp1 v3.0 annotation dge files and sample info
load(file.path("..","output","edgeR_dge_objects.e1.v3.0anno.RData"))
```
# Additive model, rB.rR500
```{r}
# relevel trt and genotype to "BLANK_NA" and "R500"
dge.root$samples<-dge.root$samples %>% mutate(trt=fct_relevel(trt,"BLANK_NA")) #%>% str() 
dge.root$samples<-dge.root$samples %>% mutate(genotype=fct_relevel(genotype,"R500")) #%>% str() 
dge.leaf$samples<-dge.leaf$samples %>% mutate(trt=fct_relevel(trt,"BLANK_NA")) #%>% str() 
dge.leaf$samples<-dge.leaf$samples %>% mutate(genotype=fct_relevel(genotype,"R500")) #%>% str() 
## dge
root.design.add <- with(dge.root$samples, model.matrix(~ genotype + trt + block))
leaf.design.add <- with(dge.leaf$samples, model.matrix(~ genotype + trt + block))
# estimateDisp
root.dge.add <- estimateDisp(dge.root,design = root.design.add)
leaf.dge.add <- estimateDisp(dge.leaf,design = leaf.design.add)
## fit linear model
root.fit.add <- glmFit(root.dge.add,root.design.add)
leaf.fit.add <- glmFit(leaf.dge.add,leaf.design.add)
# get DEGs, trt effects (coef=c("trt5E_live"))
root.trt5E_live.lrt <- glmLRT(root.fit.add,coef = c("trt5E_live"))
leaf.trt5E_live.lrt <- glmLRT(leaf.fit.add,coef = c("trt5E_live"))
topTags(root.trt5E_live.lrt);topTags(leaf.trt5E_live.lrt)
## all
root.trt5E_live.DEGs.add.all <- topTags(root.trt5E_live.lrt,n = Inf,p.value = 1)$table
leaf.trt5E_live.DEGs.add.all <- topTags(leaf.trt5E_live.lrt,n = Inf,p.value = 1)$table

# get DEGs, trt effects (coef=c("trt5E_dead"))
root.trt5E_dead.lrt <- glmLRT(root.fit.add,coef = c("trt5E_dead"))
leaf.trt5E_dead.lrt <- glmLRT(leaf.fit.add,coef = c("trt5E_dead"))
topTags(root.trt5E_live.lrt);topTags(leaf.trt5E_live.lrt)
## all
root.trt5E_dead.DEGs.add.all <- topTags(root.trt5E_dead.lrt,n = Inf,p.value = 1)$table
leaf.trt5E_dead.DEGs.add.all <- topTags(leaf.trt5E_dead.lrt,n = Inf,p.value = 1)$table
# get DEGs, genotype effects (coef = c("genotypeFPsc"))
root.genotype.lrt <- glmLRT(root.fit.add,coef = c("genotypeFPsc"))
leaf.genotype.lrt <- glmLRT(leaf.fit.add,coef = c("genotypeFPsc"))
topTags(root.genotype.lrt);topTags(leaf.genotype.lrt)
## all
root.genotype.DEGs.add.all <- topTags(root.genotype.lrt,n = Inf,p.value = 1)$table
leaf.genotype.DEGs.add.all <- topTags(leaf.genotype.lrt,n = Inf,p.value = 1)$table

# add annotiona with full name objects (all)
root.genotype.DEGs.add.all.rB.rR500.v3.0anno <- left_join(root.genotype.DEGs.add.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>%  select(1:6, AGI, At_symbol, At_short_description, perc_ID)
leaf.genotype.DEGs.add.all.rB.rR500.v3.0anno <- left_join(leaf.genotype.DEGs.add.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>% select(1:6, AGI, At_symbol, At_short_description, perc_ID)
##
root.trt5E_live.DEGs.add.all.rB.rR500.v3.0anno <- left_join(root.trt5E_live.DEGs.add.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>%  select(1:6, AGI, At_symbol, At_short_description, perc_ID)
leaf.trt5E_live.DEGs.add.all.rB.rR500.v3.0anno <- left_join(leaf.trt5E_live.DEGs.add.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>% select(1:6, AGI, At_symbol, At_short_description, perc_ID)
##
root.trt5E_dead.DEGs.add.all.rB.rR500.v3.0anno <- left_join(root.trt5E_dead.DEGs.add.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>%  select(1:6, AGI, At_symbol, At_short_description, perc_ID)
leaf.trt5E_dead.DEGs.add.all.rB.rR500.v3.0anno <- left_join(leaf.trt5E_dead.DEGs.add.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>% select(1:6, AGI, At_symbol, At_short_description, perc_ID)

# write csv (all)
## genotype
write.csv(root.genotype.DEGs.add.all.rB.rR500.v3.0anno,file=file.path("..","output","FULLtoptag","root.genotypeFPsc.DEGs.add.rB.rR500.NB.v3.0anno.csv"))
write.csv(leaf.genotype.DEGs.add.all.rB.rR500.v3.0anno,file=file.path("..","output","FULLtoptag","leaf.genotypeFPsc.DEGs.add.rB.rR500.NB.v3.0anno.csv"))
## trt5E_live
write.csv(root.trt5E_live.DEGs.add.all.rB.rR500.v3.0anno,file=file.path("..","output","FULLtoptag","root.trt5E_live.DEGs.add.rB.rR500.NB.v3.0anno.csv"))
write.csv(leaf.trt5E_live.DEGs.add.all.rB.rR500.v3.0anno,file=file.path("..","output","FULLtoptag","leaf.trt5E_live.DEGs.add.rB.rR500.NB.v3.0anno.csv"))
## trt5E_dead
write.csv(root.trt5E_dead.DEGs.add.all.rB.rR500.v3.0anno,file=file.path("..","output","FULLtoptag","root.trt5E_dead.DEGs.add.rB.rR500.NB.v3.0anno.csv"))
write.csv(leaf.trt5E_dead.DEGs.add.all.rB.rR500.v3.0anno,file=file.path("..","output","FULLtoptag","leaf.trt5E_dead.DEGs.add.rB.rR500.NB.v3.0anno.csv"))
```
# Interaction model
# clean working directory
```{r include=FALSE}
rm(list=ls())
knitr::opts_chunk$set(echo = TRUE,error=TRUE)
library(edgeR);library(tidyverse)
# annotation file for v3.0annotation
Br.v3.0.At.BLAST <- read_csv(file.path("..","..","..","Annotation","output","v3.0annotation","Brapa_v3.0_annotated.csv")) 
# This annotation is redundant with name (Br grene). Eg 
Br.v3.0.At.BLAST %>% filter(name=="BraA01g040570.3C")
# reduce the redundancy (112418)
Br.v3.0anno.At.BLAST.highscore <- Br.v3.0.At.BLAST %>% group_by(name) %>% arrange(desc(score)) %>% dplyr::slice(1)

# load exp1 v3.0 annotation dge files and sample info
load(file.path("..","output","edgeR_dge_objects.e1.v3.0anno.RData"))
```
# Interaction model, rD.rFPsc
```{r}
# relevel trt and genotype to "5E_dead" and "FPsc"
dge.root$samples<-dge.root$samples %>% mutate(trt=fct_relevel(trt,"5E_dead")) #%>% str() 
dge.root$samples<-dge.root$samples %>% mutate(genotype=fct_relevel(genotype,"FPsc")) #%>% str() 
dge.leaf$samples<-dge.leaf$samples %>% mutate(trt=fct_relevel(trt,"5E_dead")) #%>% str() 
dge.leaf$samples<-dge.leaf$samples %>% mutate(genotype=fct_relevel(genotype,"FPsc")) #%>% str() 
## dge
root.design.int <- with(dge.root$samples, model.matrix(~ genotype*trt + block))
leaf.design.int <- with(dge.leaf$samples, model.matrix(~ genotype*trt + block))
# estimateDisp
root.dge.int <- estimateDisp(dge.root,design = root.design.int)
leaf.dge.int <- estimateDisp(dge.leaf,design = leaf.design.int)
## fit linear model
root.fit.int <- glmFit(root.dge.int,root.design.int)
leaf.fit.int <- glmFit(leaf.dge.int,leaf.design.int)
# get DEGs, trt effects (coef=c("trt5E_live","genotypeR500:trt5E_live"))
root.trt5E_live.lrt <- glmLRT(root.fit.int,coef = c("trt5E_live","genotypeR500:trt5E_live")) # all coeff containing "trt"
leaf.trt5E_live.lrt <- glmLRT(leaf.fit.int,coef = c("trt5E_live","genotypeR500:trt5E_live")) # all coeff containing "trt"
topTags(root.trt5E_live.lrt);topTags(leaf.trt5E_live.lrt)
## all
root.trt5E_live.DEGs.int.all <- topTags(root.trt5E_live.lrt,n = Inf,p.value = 1)$table
leaf.trt5E_live.DEGs.int.all <- topTags(leaf.trt5E_live.lrt,n = Inf,p.value = 1)$table
# get DEGs, genotype effects (coef = c("genotypeR500","genotypeR500:trtBLANK_NA","genotypeR500:trt5E_live"))
root.genotype.lrt <- glmLRT(root.fit.int,coef = c("genotypeR500","genotypeR500:trtBLANK_NA","genotypeR500:trt5E_live"))
leaf.genotype.lrt <- glmLRT(leaf.fit.int,coef = c("genotypeR500","genotypeR500:trtBLANK_NA","genotypeR500:trt5E_live"))
topTags(root.genotype.lrt);topTags(leaf.genotype.lrt)
## all
root.genotype.DEGs.int.all <- topTags(root.genotype.lrt,n = Inf,p.value = 1)$table 
leaf.genotype.DEGs.int.all <- topTags(leaf.genotype.lrt,n = Inf,p.value = 1)$table
# get DEGs, genotype:treatment (live) effects
root.trt5E_live.genotypeR500.lrt <- glmLRT(root.fit.int,coef = "genotypeR500:trt5E_live")
leaf.trt5E_live.genotypeR500.lrt <- glmLRT(leaf.fit.int,coef = "genotypeR500:trt5E_live")
topTags(root.trt5E_live.genotypeR500.lrt);topTags(leaf.trt5E_live.genotypeR500.lrt)
## all
root.trt5E_live.genotypeR500.DEGs.int.all <- topTags(root.trt5E_live.genotypeR500.lrt,n = Inf,p.value = 1)$table 
leaf.trt5E_live.genotypeR500.DEGs.int.all <- topTags(leaf.trt5E_live.genotypeR500.lrt,n = Inf,p.value = 1)$table
# add annotiona with full name objects (all)
## genotype
root.genotypeR500_trt5E_live.genotypeR500_trtBLANK_NA.genotypeR500.DEGs.int.all.rD.rFPsc.v3.0anno <- left_join(root.genotype.DEGs.int.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>%  select(1:8, AGI, At_symbol, At_short_description, perc_ID)
leaf.genotypeR500_trt5E_live.genotypeR500_trtBLANK_NA.genotypeR500.DEGs.int.all.rD.rFPsc.v3.0anno <- left_join(leaf.genotype.DEGs.int.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>% select(1:8, AGI, At_symbol, At_short_description, perc_ID)
# trt5E_live
root.trt5E_live.genotypeR500_trt5E_live.DEGs.int.all.rD.rFPsc.v3.0anno <- left_join(root.trt5E_live.DEGs.int.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>%  select(1:7, AGI, At_symbol, At_short_description, perc_ID)
leaf.trt5E_live.genotypeR500_trt5E_live.DEGs.int.all.rD.rFPsc.v3.0anno <- left_join(leaf.trt5E_live.DEGs.int.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>% select(1:7, AGI, At_symbol, At_short_description, perc_ID)
# genotypeR500:trt5E_live
root.trt5E_live.genotypeR500.DEGs.int.all.rD.rFPsc.v3.0anno <- left_join(root.trt5E_live.genotypeR500.DEGs.int.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>%  select(1:6, AGI, At_symbol, At_short_description, perc_ID)
leaf.trt5E_live.genotypeR500.DEGs.int.all.rD.rFPsc.v3.0anno <- left_join(leaf.trt5E_live.genotypeR500.DEGs.int.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>% select(1:6, AGI, At_symbol, At_short_description, perc_ID)
# write csv (naming: "tissue"."coef1"_"coef2"_"coef3"."DEGs"."model"."reference for soil (D,dead;L,live)"."reference for genotype"."annotation version") if there are triple coefs (all)
## genotype
write.csv(root.genotypeR500_trt5E_live.genotypeR500_trtBLANK_NA.genotypeR500.DEGs.int.all.rD.rFPsc.v3.0anno,file=file.path("..","output","FULLtoptag","root.genotypeR500_trt5E_live.genotypeR500_trtBLANK_NA.genotypeR500.DEGs.int.rD.rFPsc.NB.v3.0anno.csv"))
write.csv(leaf.genotypeR500_trt5E_live.genotypeR500_trtBLANK_NA.genotypeR500.DEGs.int.all.rD.rFPsc.v3.0anno,file=file.path("..","output","FULLtoptag","leaf.genotypeR500_trt5E_live.genotypeR500_trtBLANK_NA.genotypeR500.DEGs.int.rD.rFPsc.NB.v3.0anno.csv"))
## trt5E_live
write.csv(root.trt5E_live.genotypeR500_trt5E_live.DEGs.int.all.rD.rFPsc.v3.0anno,file=file.path("..","output","FULLtoptag","root.trt5E_live.genotypeR500_trt5E_live.DEGs.int.rD.rFPsc.NB.v3.0anno.csv"))
write.csv(leaf.trt5E_live.genotypeR500_trt5E_live.DEGs.int.all.rD.rFPsc.v3.0anno,file=file.path("..","output","FULLtoptag","leaf.trt5E_live.genotypeR500_trt5E_live.DEGs.int.rD.rFPsc.NB.v3.0anno.csv"))
```
# clean working directory
```{r include=FALSE}
rm(list=ls())
knitr::opts_chunk$set(echo = TRUE,error=TRUE)
library(edgeR);library(tidyverse)
# annotation file for v3.0annotation
Br.v3.0.At.BLAST <- read_csv(file.path("..","..","..","Annotation","output","v3.0annotation","Brapa_v3.0_annotated.csv")) 
# This annotation is redundant with name (Br grene). Eg 
Br.v3.0.At.BLAST %>% filter(name=="BraA01g040570.3C")
# reduce the redundancy (112418)
Br.v3.0anno.At.BLAST.highscore <- Br.v3.0.At.BLAST %>% group_by(name) %>% arrange(desc(score)) %>% dplyr::slice(1)

# load exp1 v3.0 annotation dge files and sample info
load(file.path("..","output","edgeR_dge_objects.e1.v3.0anno.RData"))
```
# Interaction model, rB.rFPsc
```{r}
# relevel trt and genotype to "BLANK_NA" and "FPsc"
dge.root$samples<-dge.root$samples %>% mutate(trt=fct_relevel(trt,"BLANK_NA")) #%>% str() 
dge.root$samples<-dge.root$samples %>% mutate(genotype=fct_relevel(genotype,"FPsc")) #%>% str() 
dge.leaf$samples<-dge.leaf$samples %>% mutate(trt=fct_relevel(trt,"BLANK_NA")) #%>% str() 
dge.leaf$samples<-dge.leaf$samples %>% mutate(genotype=fct_relevel(genotype,"FPsc")) #%>% str() 
## dge
root.design.int <- with(dge.root$samples, model.matrix(~ genotype*trt + block))
leaf.design.int <- with(dge.leaf$samples, model.matrix(~ genotype*trt + block))
# estimateDisp
root.dge.int <- estimateDisp(dge.root,design = root.design.int)
leaf.dge.int <- estimateDisp(dge.leaf,design = leaf.design.int)
## fit linear model
root.fit.int <- glmFit(root.dge.int,root.design.int)
leaf.fit.int <- glmFit(leaf.dge.int,leaf.design.int)
# get DEGs, trt effects (coef=c("trt5E_live","genotypeR500:trt5E_live"))
root.trt5E_live.lrt <- glmLRT(root.fit.int,coef = c("trt5E_live","genotypeR500:trt5E_live")) # all coeff containing "trt5E_live"
leaf.trt5E_live.lrt <- glmLRT(leaf.fit.int,coef = c("trt5E_live","genotypeR500:trt5E_live")) # all coeff containing "trt5E_live"
topTags(root.trt5E_live.lrt);topTags(leaf.trt5E_live.lrt)
## all
root.trt5E_live.DEGs.int.all <- topTags(root.trt5E_live.lrt,n = Inf,p.value = 1)$table
leaf.trt5E_live.DEGs.int.all <- topTags(leaf.trt5E_live.lrt,n = Inf,p.value = 1)$table
# get DEGs, trt effects (coef=c("trt5E_dead","genotypeR500:trt5E_dead"))
root.trt5E_dead.lrt <- glmLRT(root.fit.int,coef = c("trt5E_dead","genotypeR500:trt5E_dead")) # all coeff containing "trt5E_dead"
leaf.trt5E_dead.lrt <- glmLRT(leaf.fit.int,coef = c("trt5E_dead","genotypeR500:trt5E_dead")) # all coeff containing "trt5E_dead"
topTags(root.trt5E_dead.lrt);topTags(leaf.trt5E_dead.lrt)
## all
root.trt5E_dead.DEGs.int.all <- topTags(root.trt5E_dead.lrt,n = Inf,p.value = 1)$table
leaf.trt5E_dead.DEGs.int.all <- topTags(leaf.trt5E_dead.lrt,n = Inf,p.value = 1)$table

# get DEGs, genotype effects (coef = c("genotypeR500","genotypeR500:trt5E_dead","genotypeR500:trt5E_live"))?
root.genotype.lrt <- glmLRT(root.fit.int,coef = c("genotypeR500","genotypeR500:trt5E_dead","genotypeR500:trt5E_live"))
leaf.genotype.lrt <- glmLRT(leaf.fit.int,coef = c("genotypeR500","genotypeR500:trt5E_dead","genotypeR500:trt5E_live"))
topTags(root.genotype.lrt);topTags(leaf.genotype.lrt)
## all
root.genotype.DEGs.int.all <- topTags(root.genotype.lrt,n = Inf,p.value = 1)$table
leaf.genotype.DEGs.int.all <- topTags(leaf.genotype.lrt,n = Inf,p.value = 1)$table
# get DEGs, genotype:treatment (live) effects
root.trt5E_live.genotypeR500.lrt <- glmLRT(root.fit.int,coef = "genotypeR500:trt5E_live")
leaf.trt5E_live.genotypeR500.lrt <- glmLRT(leaf.fit.int,coef = "genotypeR500:trt5E_live")
topTags(root.trt5E_live.genotypeR500.lrt);topTags(leaf.trt5E_live.genotypeR500.lrt)
## all
root.trt5E_live.genotypeR500.DEGs.int.all <- topTags(root.trt5E_live.genotypeR500.lrt,n = Inf,p.value = 1)$table 
leaf.trt5E_live.genotypeR500.DEGs.int.all <- topTags(leaf.trt5E_live.genotypeR500.lrt,n = Inf,p.value = 1)$table


# add annotiona with full name objects (all)
## genotype
root.genotypeR500_trt5E_live.genotypeR500_trt5E_dead.genotypeR500.DEGs.int.all.rB.rFPsc.v3.0anno <- left_join(root.genotype.DEGs.int.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>%  select(1:8, AGI, At_symbol, At_short_description, perc_ID)
leaf.genotypeR500_trt5E_live.genotypeR500_trt5E_dead.genotypeR500.DEGs.int.all.rB.rFPsc.v3.0anno <- left_join(leaf.genotype.DEGs.int.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>% select(1:8, AGI, At_symbol, At_short_description, perc_ID)
# trt5E_live
root.trt5E_live.genotypeR500_trt5E_live.DEGs.int.all.rB.rFPsc.v3.0anno <- left_join(root.trt5E_live.DEGs.int.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>%  select(1:7, AGI, At_symbol, At_short_description, perc_ID)
leaf.trt5E_live.genotypeR500_trt5E_live.DEGs.int.all.rB.rFPsc.v3.0anno <- left_join(leaf.trt5E_live.DEGs.int.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>% select(1:7, AGI, At_symbol, At_short_description, perc_ID)
# trt5E_dead
root.trt5E_dead.genotypeR500_trt5E_dead.DEGs.int.all.rB.rFPsc.v3.0anno <- left_join(root.trt5E_dead.DEGs.int.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>%  select(1:7, AGI, At_symbol, At_short_description, perc_ID)
leaf.trt5E_dead.genotypeR500_trt5E_dead.DEGs.int.all.rB.rFPsc.v3.0anno <- left_join(leaf.trt5E_dead.DEGs.int.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>% select(1:7, AGI, At_symbol, At_short_description, perc_ID)
# genotypeR500:trt5E_live
root.trt5E_live.genotypeR500.DEGs.int.all.rB.rFPsc.v3.0anno <- left_join(root.trt5E_live.genotypeR500.DEGs.int.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>%  select(1:6, AGI, At_symbol, At_short_description, perc_ID)
leaf.trt5E_live.genotypeR500.DEGs.int.all.rB.rFPsc.v3.0anno <- left_join(leaf.trt5E_live.genotypeR500.DEGs.int.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>% select(1:6, AGI, At_symbol, At_short_description, perc_ID)

# write csv (naming: "tissue"."coef1"_"coef2"_"coef3","DEGs","model","reference for soil (B, blank;D,dead;L,live)","reference for genotype","annotation version") (all)
## genotype
write.csv(root.genotypeR500_trt5E_live.genotypeR500_trt5E_dead.genotypeR500.DEGs.int.all.rB.rFPsc.v3.0anno,file=file.path("..","output","FULLtoptag","root.genotypeR500_trt5E_live.genotypeR500_trt5E_dead.genotypeR500.DEGs.int.rB.rFPsc.NB.v3.0anno.csv"))
write.csv(leaf.genotypeR500_trt5E_live.genotypeR500_trt5E_dead.genotypeR500.DEGs.int.all.rB.rFPsc.v3.0anno,file=file.path("..","output","FULLtoptag","leaf.genotypeR500_trt5E_live.genotypeR500_trt5E_dead.genotypeR500.DEGs.int.rB.rFPsc.NB.v3.0anno.csv"))
## trt (5E_live)
write.csv(root.trt5E_live.genotypeR500_trt5E_live.DEGs.int.all.rB.rFPsc.v3.0anno,file=file.path("..","output","FULLtoptag","root.trt5E_live.genotypeR500_trt5E_live.DEGs.int.rB.rFPsc.NB.v3.0anno.csv"))
write.csv(leaf.trt5E_live.genotypeR500_trt5E_live.DEGs.int.all.rB.rFPsc.v3.0anno,file=file.path("..","output","FULLtoptag","leaf.trt5E_live.genotypeR500_trt5E_live.DEGs.int.rB.rFPsc.NB.v3.0anno.csv"))
## trt (5E_dead)
write.csv(root.trt5E_dead.genotypeR500_trt5E_dead.DEGs.int.all.rB.rFPsc.v3.0anno,file=file.path("..","output","FULLtoptag","root.trt5E_dead.genotypeR500_trt5E_dead.DEGs.int.rB.rFPsc.NB.v3.0anno.csv"))
write.csv(leaf.trt5E_dead.genotypeR500_trt5E_dead.DEGs.int.all.rB.rFPsc.v3.0anno,file=file.path("..","output","FULLtoptag","leaf.trt5E_dead.genotypeR500_trt5E_dead.DEGs.int.rB.rFPsc.NB.v3.0anno.csv"))
## genotypeR500:trt5E_live
write.csv(root.trt5E_live.genotypeR500.DEGs.int.all.rB.rFPsc.v3.0anno,file=file.path("..","output","FULLtoptag","root.trt5E_live.genotypeR500.DEGs.int.rB.rFPsc.NB.v3.0anno.csv"))
write.csv(leaf.trt5E_live.genotypeR500.DEGs.int.all.rB.rFPsc.v3.0anno,file=file.path("..","output","FULLtoptag","leaf.trt5E_live.genotypeR500.DEGs.int.rB.rFPsc.NB.v3.0anno.csv"))

```

# clean working directory
```{r include=FALSE}
rm(list=ls())
knitr::opts_chunk$set(echo = TRUE,error=TRUE)
library(edgeR);library(tidyverse)
# annotation file for v3.0annotation
Br.v3.0.At.BLAST <- read_csv(file.path("..","..","..","Annotation","output","v3.0annotation","Brapa_v3.0_annotated.csv")) 
# This annotation is redundant with name (Br grene). Eg 
Br.v3.0.At.BLAST %>% filter(name=="BraA01g040570.3C")
# reduce the redundancy (112418)
Br.v3.0anno.At.BLAST.highscore <- Br.v3.0.At.BLAST %>% group_by(name) %>% arrange(desc(score)) %>% dplyr::slice(1)
# load exp1 v3.0 annotation dge files and sample info
load(file.path("..","output","edgeR_dge_objects.e1.v3.0anno.RData"))
```

# Interaction model, rD.rR500
```{r}
# relevel trt and genotype to "5E_dead" and "R500"
dge.root$samples<-dge.root$samples %>% mutate(trt=fct_relevel(trt,"5E_dead")) #%>% str() 
dge.root$samples<-dge.root$samples %>% mutate(genotype=fct_relevel(genotype,"R500")) #%>% str() 
dge.leaf$samples<-dge.leaf$samples %>% mutate(trt=fct_relevel(trt,"5E_dead")) #%>% str() 
dge.leaf$samples<-dge.leaf$samples %>% mutate(genotype=fct_relevel(genotype,"R500")) #%>% str() 
## dge
root.design.int <- with(dge.root$samples, model.matrix(~ genotype*trt + block))
leaf.design.int <- with(dge.leaf$samples, model.matrix(~ genotype*trt + block))
# estimateDisp
root.dge.int <- estimateDisp(dge.root,design = root.design.int)
leaf.dge.int <- estimateDisp(dge.leaf,design = leaf.design.int)
## fit linear model
root.fit.int <- glmFit(root.dge.int,root.design.int)
leaf.fit.int <- glmFit(leaf.dge.int,leaf.design.int)
# get DEGs, trt effects (coef=c("trt5E_live","genotypeFPsc:trt5E_live"))
root.trt5E_live.lrt <- glmLRT(root.fit.int,coef = c("trt5E_live","genotypeFPsc:trt5E_live")) # all coeff containing "trt"
leaf.trt5E_live.lrt <- glmLRT(leaf.fit.int,coef = c("trt5E_live","genotypeFPsc:trt5E_live")) # all coeff containing "trt"
topTags(root.trt5E_live.lrt);topTags(leaf.trt5E_live.lrt)
## all
root.trt5E_live.DEGs.int.all <- topTags(root.trt5E_live.lrt,n = Inf,p.value = 1)$table
leaf.trt5E_live.DEGs.int.all <- topTags(leaf.trt5E_live.lrt,n = Inf,p.value = 1)$table
# get DEGs, genotype effects (coef = c("genotypeR500","genotypeR500:trtBLANK_NA","genotypeR500:trt5E_live"))
root.genotype.lrt <- glmLRT(root.fit.int,coef = c("genotypeFPsc","genotypeFPsc:trtBLANK_NA","genotypeFPsc:trt5E_live"))
leaf.genotype.lrt <- glmLRT(leaf.fit.int,coef = c("genotypeFPsc","genotypeFPsc:trtBLANK_NA","genotypeFPsc:trt5E_live"))
topTags(root.genotype.lrt);topTags(leaf.genotype.lrt)
## all
root.genotype.DEGs.int.all <- topTags(root.genotype.lrt,n = Inf,p.value = 1)$table
leaf.genotype.DEGs.int.all <- topTags(leaf.genotype.lrt,n = Inf,p.value = 1)$table 
# get DEGs, genotype:treatment (live) effects
root.trt5E_live.genotypeFPsc.lrt <- glmLRT(root.fit.int,coef = "genotypeFPsc:trt5E_live")
leaf.trt5E_live.genotypeFPsc.lrt <- glmLRT(leaf.fit.int,coef = "genotypeFPsc:trt5E_live")
topTags(root.trt5E_live.genotypeFPsc.lrt);topTags(leaf.trt5E_live.genotypeFPsc.lrt)
## all
root.trt5E_live.genotypeFPsc.DEGs.int.all <- topTags(root.trt5E_live.genotypeFPsc.lrt,n = Inf,p.value = 1)$table 
leaf.trt5E_live.genotypeFPsc.DEGs.int.all <- topTags(leaf.trt5E_live.genotypeFPsc.lrt,n = Inf,p.value = 1)$table

# add annotiona with full name objects (all)
## genotype
root.genotypeFPsc_trt5E_live.genotypeFPsc_trtBLANK_NA.genotypeFPsc.DEGs.int.all.rD.rR500.v3.0anno <- left_join(root.genotype.DEGs.int.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>% dplyr::select(1:8, AGI, At_symbol, At_short_description, perc_ID)
leaf.genotypeFPsc_trt5E_live.genotypeFPsc_trtBLANK_NA.genotypeFPsc.DEGs.int.all.rD.rR500.v3.0anno <- left_join(leaf.genotype.DEGs.int.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>% dplyr::select(1:8, AGI, At_symbol, At_short_description, perc_ID)
# trt5E_live
root.trt5E_live.genotypeFPsc_trt5E_live.DEGs.int.all.rD.rR500.v3.0anno <- left_join(root.trt5E_live.DEGs.int.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>% dplyr::select(1:7, AGI, At_symbol, At_short_description, perc_ID)
leaf.trt5E_live.genotypeFPsc_trt5E_live.DEGs.int.all.rD.rR500.v3.0anno <- left_join(leaf.trt5E_live.DEGs.int.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>% dplyr::select(1:7, AGI, At_symbol, At_short_description, perc_ID)

# genotypeFPsc:trt5E_live
root.trt5E_live.genotypeFPsc.DEGs.int.all.rD.rFPsc.v3.0anno <- left_join(root.trt5E_live.genotypeFPsc.DEGs.int.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>%  select(1:6, AGI, At_symbol, At_short_description, perc_ID)
leaf.trt5E_live.genotypeFPsc.DEGs.int.all.rD.rFPsc.v3.0anno <- left_join(leaf.trt5E_live.genotypeFPsc.DEGs.int.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>% select(1:6, AGI, At_symbol, At_short_description, perc_ID)

# write csv (naming: "tissue"."coef1"_"coef2"_"coef3"."DEGs"."model"."reference for soil (D,dead;L,live)"."reference for genotype"."annotation version") if there are triple coefs (all)
## genotype
write.csv(root.genotypeFPsc_trt5E_live.genotypeFPsc_trtBLANK_NA.genotypeFPsc.DEGs.int.all.rD.rR500.v3.0anno,file=file.path("..","output","FULLtoptag","root.genotypeFPsc_trt5E_live.genotypeFPsc_trtBLANK_NA.genotypeFPsc.DEGs.int.rD.rR500.NB.v3.0anno.csv"))
write.csv(leaf.genotypeFPsc_trt5E_live.genotypeFPsc_trtBLANK_NA.genotypeFPsc.DEGs.int.all.rD.rR500.v3.0anno,file=file.path("..","output","FULLtoptag","leaf.genotypeFPsc_trt5E_live.genotypeFPsc_trtBLANK_NA.genotypeFPsc.DEGs.int.rD.rR500.NB.v3.0anno.csv"))
## trt5E_live
write.csv(root.trt5E_live.genotypeFPsc_trt5E_live.DEGs.int.all.rD.rR500.v3.0anno,file=file.path("..","output","FULLtoptag","root.trt5E_live.genotypeFPsc_trt5E_live.DEGs.int.rD.rR500.NB.v3.0anno.csv"))
write.csv(leaf.trt5E_live.genotypeFPsc_trt5E_live.DEGs.int.all.rD.rR500.v3.0anno,file=file.path("..","output","FULLtoptag","leaf.trt5E_live.genotypeFPsc_trt5E_live.DEGs.int.rD.rR500.NB.v3.0anno.csv"))
## genotypeFPsc:trt5E_live
write.csv(root.trt5E_live.genotypeFPsc.DEGs.int.all.rD.rFPsc.v3.0anno,file=file.path("..","output","FULLtoptag","root.trt5E_live.genotypeFPsc.DEGs.int.rD.rR500.NB.v3.0anno.csv"))
write.csv(leaf.trt5E_live.genotypeFPsc.DEGs.int.all.rD.rFPsc.v3.0anno,file=file.path("..","output","FULLtoptag","leaf.trt5E_live.genotypeFPsc.DEGs.int.rD.rR500.NB.v3.0anno.csv"))

```

# clean working directory
```{r include=FALSE}
rm(list=ls())
knitr::opts_chunk$set(echo = TRUE,error=TRUE)
library(edgeR);library(tidyverse)
# annotation file for v3.0annotation
Br.v3.0.At.BLAST <- read_csv(file.path("..","..","..","Annotation","output","v3.0annotation","Brapa_v3.0_annotated.csv")) 
# This annotation is redundant with name (Br grene). Eg 
Br.v3.0.At.BLAST %>% filter(name=="BraA01g040570.3C")
# reduce the redundancy (112418)
Br.v3.0anno.At.BLAST.highscore <- Br.v3.0.At.BLAST %>% group_by(name) %>% arrange(desc(score)) %>% dplyr::slice(1)

# load exp1 v3.0 annotation dge files and sample info
load(file.path("..","output","edgeR_dge_objects.e1.v3.0anno.RData"))
```

# Interaction model, rB.rR500
```{r}
# relevel trt and genotype to "BLANK_NA" and "R500"
dge.root$samples<-dge.root$samples %>% mutate(trt=fct_relevel(trt,"BLANK_NA")) #%>% str() 
dge.root$samples<-dge.root$samples %>% mutate(genotype=fct_relevel(genotype,"R500")) #%>% str() 
dge.leaf$samples<-dge.leaf$samples %>% mutate(trt=fct_relevel(trt,"BLANK_NA")) #%>% str() 
dge.leaf$samples<-dge.leaf$samples %>% mutate(genotype=fct_relevel(genotype,"R500")) #%>% str() 
## dge
root.design.int <- with(dge.root$samples, model.matrix(~ genotype*trt + block))
leaf.design.int <- with(dge.leaf$samples, model.matrix(~ genotype*trt + block))
# estimateDisp
root.dge.int <- estimateDisp(dge.root,design = root.design.int)
leaf.dge.int <- estimateDisp(dge.leaf,design = leaf.design.int)
## fit linear model
root.fit.int <- glmFit(root.dge.int,root.design.int)
leaf.fit.int <- glmFit(leaf.dge.int,leaf.design.int)
# get DEGs, trt effects (coef=c("trt5E_live","genotypeFPsc:trt5E_live"))
root.trt5E_live.lrt <- glmLRT(root.fit.int,coef = c("trt5E_live","genotypeFPsc:trt5E_live")) # all coeff containing "trt5E_live"
leaf.trt5E_live.lrt <- glmLRT(leaf.fit.int,coef = c("trt5E_live","genotypeFPsc:trt5E_live")) # all coeff containing "trt5E_live"
topTags(root.trt5E_live.lrt);topTags(leaf.trt5E_live.lrt)
## all
root.trt5E_live.DEGs.int.all <- topTags(root.trt5E_live.lrt,n = Inf,p.value = 1)$table
leaf.trt5E_live.DEGs.int.all <- topTags(leaf.trt5E_live.lrt,n = Inf,p.value = 1)$table

# get DEGs, trt effects (coef=c("trt5E_dead","genotypeFPsc:trt5E_dead"))
root.trt5E_dead.lrt <- glmLRT(root.fit.int,coef = c("trt5E_dead","genotypeFPsc:trt5E_dead")) # all coeff containing "trt5E_dead"
leaf.trt5E_dead.lrt <- glmLRT(leaf.fit.int,coef = c("trt5E_dead","genotypeFPsc:trt5E_dead")) # all coeff containing "trt5E_dead"
topTags(root.trt5E_dead.lrt);topTags(leaf.trt5E_dead.lrt)
## all
root.trt5E_dead.DEGs.int.all <- topTags(root.trt5E_dead.lrt,n = Inf,p.value = 1)$table
leaf.trt5E_dead.DEGs.int.all <- topTags(leaf.trt5E_dead.lrt,n = Inf,p.value = 1)$table

# get DEGs, genotype effects (coef = c("genotypeFPsc","genotypeFPsc:trt5E_dead","genotypeFPsc:trt5E_live"))?
root.genotype.lrt <- glmLRT(root.fit.int,coef = c("genotypeFPsc","genotypeFPsc:trt5E_dead","genotypeFPsc:trt5E_live"))
leaf.genotype.lrt <- glmLRT(leaf.fit.int,coef = c("genotypeFPsc","genotypeFPsc:trt5E_dead","genotypeFPsc:trt5E_live"))
topTags(root.genotype.lrt);topTags(leaf.genotype.lrt)
## all
root.genotype.DEGs.int.all <- topTags(root.genotype.lrt,n = Inf,p.value = 1)$table
leaf.genotype.DEGs.int.all <- topTags(leaf.genotype.lrt,n = Inf,p.value = 1)$table
# get DEGs, genotype:treatment (live) effects
root.trt5E_live.genotypeFPsc.lrt <- glmLRT(root.fit.int,coef = "genotypeFPsc:trt5E_live")
leaf.trt5E_live.genotypeFPsc.lrt <- glmLRT(leaf.fit.int,coef = "genotypeFPsc:trt5E_live")
topTags(root.trt5E_live.genotypeFPsc.lrt);topTags(leaf.trt5E_live.genotypeFPsc.lrt)
## all
root.trt5E_live.genotypeFPsc.DEGs.int.all <- topTags(root.trt5E_live.genotypeFPsc.lrt,n = Inf,p.value = 1)$table 
leaf.trt5E_live.genotypeFPsc.DEGs.int.all <- topTags(leaf.trt5E_live.genotypeFPsc.lrt,n = Inf,p.value = 1)$table

# add annotiona with full name objects (all)
## genotype
root.genotypeFPsc_trt5E_live.genotypeFPsc_trt5E_dead.genotypeFPsc.DEGs.int.all.rB.rR500.v3.0anno <- left_join(root.genotype.DEGs.int.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>% dplyr::select(1:8, AGI, At_symbol, At_short_description, perc_ID)
leaf.genotypeFPsc_trt5E_live.genotypeFPsc_trt5E_dead.genotypeFPsc.DEGs.int.all.rB.rR500.v3.0anno <- left_join(leaf.genotype.DEGs.int.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>% dplyr::select(1:8, AGI, At_symbol, At_short_description, perc_ID)
# trt5E_live
root.trt5E_live.genotypeFPsc_trt5E_live.DEGs.int.all.rB.rR500.v3.0anno <- left_join(root.trt5E_live.DEGs.int.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>% dplyr::select(1:7, AGI, At_symbol, At_short_description, perc_ID)
leaf.trt5E_live.genotypeFPsc_trt5E_live.DEGs.int.all.rB.rR500.v3.0anno <- left_join(leaf.trt5E_live.DEGs.int.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>% dplyr::select(1:7, AGI, At_symbol, At_short_description, perc_ID)
# trt5E_dead
root.trt5E_dead.genotypeFPsc_trt5E_dead.DEGs.int.all.rB.rR500.v3.0anno <- left_join(root.trt5E_dead.DEGs.int.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>% dplyr::select(1:7, AGI, At_symbol, At_short_description, perc_ID)
leaf.trt5E_dead.genotypeFPsc_trt5E_dead.DEGs.int.all.rB.rR500.v3.0anno <- left_join(leaf.trt5E_dead.DEGs.int.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>% dplyr::select(1:7, AGI, At_symbol, At_short_description, perc_ID)
# genotypeFPsc:trt5E_live
root.trt5E_live.genotypeFPsc.DEGs.int.all.rB.rR500.v3.0anno <- left_join(root.trt5E_live.genotypeFPsc.DEGs.int.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>%  select(1:6, AGI, At_symbol, At_short_description, perc_ID)
leaf.trt5E_live.genotypeFPsc.DEGs.int.all.rB.rR500.v3.0anno <- left_join(leaf.trt5E_live.genotypeFPsc.DEGs.int.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>% select(1:6, AGI, At_symbol, At_short_description, perc_ID)

# write csv (naming: "tissue"."coef1"_"coef2"_"coef3","DEGs","model","reference for soil (B, blank;D,dead;L,live)","reference for genotype","annotation version") (all)
## genotype
write.csv(root.genotypeFPsc_trt5E_live.genotypeFPsc_trt5E_dead.genotypeFPsc.DEGs.int.all.rB.rR500.v3.0anno,file=file.path("..","output","FULLtoptag","root.genotypeFPsc_trt5E_live.genotypeFPsc_trt5E_dead.genotypeFPsc.DEGs.int.rB.rR500.NB.v3.0anno.csv"))
write.csv(leaf.genotypeFPsc_trt5E_live.genotypeFPsc_trt5E_dead.genotypeFPsc.DEGs.int.all.rB.rR500.v3.0anno,file=file.path("..","output","FULLtoptag","leaf.genotypeFPsc_trt5E_live.genotypeFPsc_trt5E_dead.genotypeFPsc.DEGs.int.rB.rR500.NB.v3.0anno.csv"))
## trt (5E_live)
write.csv(root.trt5E_live.genotypeFPsc_trt5E_live.DEGs.int.all.rB.rR500.v3.0anno,file=file.path("..","output","FULLtoptag","root.trt5E_live.genotypeFPsc_trt5E_live.DEGs.int.rB.rR500.NB.v3.0anno.csv"))
write.csv(leaf.trt5E_live.genotypeFPsc_trt5E_live.DEGs.int.all.rB.rR500.v3.0anno,file=file.path("..","output","FULLtoptag","leaf.trt5E_live.genotypeFPsc_trt5E_live.DEGs.int.rB.rR500.NB.v3.0anno.csv"))
## trt (5E_dead)
write.csv(root.trt5E_dead.genotypeFPsc_trt5E_dead.DEGs.int.all.rB.rR500.v3.0anno,file=file.path("..","output","FULLtoptag","root.trt5E_dead.genotypeFPsc_trt5E_dead.DEGs.int.rB.rR500.NB.v3.0anno.csv"))
write.csv(leaf.trt5E_dead.genotypeFPsc_trt5E_dead.DEGs.int.all.rB.rR500.v3.0anno,file=file.path("..","output","FULLtoptag","leaf.trt5E_dead.genotypeFPsc_trt5E_dead.DEGs.int.rB.rR500.NB.v3.0anno.csv"))
## genotypeFPsc:trt5E_live
write.csv(root.trt5E_live.genotypeFPsc.DEGs.int.all.rB.rR500.v3.0anno,file=file.path("..","output","FULLtoptag","root.trt5E_live.genotypeFPsc.DEGs.int.rB.rR500.NB.v3.0anno.csv"))
write.csv(leaf.trt5E_live.genotypeFPsc.DEGs.int.all.rB.rR500.v3.0anno,file=file.path("..","output","FULLtoptag","leaf.trt5E_live.genotypeFPsc.DEGs.int.rB.rR500.NB.v3.0anno.csv"))
```

# clean working directory
```{r include=FALSE}
rm(list=ls())
knitr::opts_chunk$set(echo = TRUE,error=TRUE)
library(edgeR);library(tidyverse)
# annotation file for v3.0annotation
Br.v3.0.At.BLAST <- read_csv(file.path("..","..","..","Annotation","output","v3.0annotation","Brapa_v3.0_annotated.csv")) 
# This annotation is redundant with name (Br grene). Eg 
Br.v3.0.At.BLAST %>% filter(name=="BraA01g040570.3C")
# reduce the redundancy (112418)
Br.v3.0anno.At.BLAST.highscore <- Br.v3.0.At.BLAST %>% group_by(name) %>% arrange(desc(score)) %>% dplyr::slice(1)

# load exp1 v3.0 annotation dge files and sample info
load(file.path("..","output","edgeR_dge_objects.e1.v3.0anno.RData"))
```

# combine blank and dead as a one group (051319)
```{r}
dge.leaf$samples <- dge.leaf$samples %>% 
  mutate(group2=str_replace_all(group,"BLANK_NA|5E_dead","blankdead"),trt=str_replace_all(trt,"BLANK_NA|5E_dead","blankdead")) 
dge.root$samples <- dge.root$samples %>% 
  mutate(group2=str_replace_all(group,"BLANK_NA|5E_dead","blankdead"),trt=str_replace_all(trt,"BLANK_NA|5E_dead","blankdead")) 

```
# 
# Interaction model, rBD.rR500
```{r}
# relevel trt and genotype to "blankdead" and "R500"
dge.root$samples<-dge.root$samples %>% mutate(trt=fct_relevel(trt,"blankdead")) #%>% str() 
dge.root$samples<-dge.root$samples %>% mutate(genotype=fct_relevel(genotype,"R500")) #%>% str() 
dge.leaf$samples<-dge.leaf$samples %>% mutate(trt=fct_relevel(trt,"blankdead")) #%>% str() 
dge.leaf$samples<-dge.leaf$samples %>% mutate(genotype=fct_relevel(genotype,"R500")) #%>% str() 
## dge
root.design.int <- with(dge.root$samples, model.matrix(~ genotype*trt + block))
leaf.design.int <- with(dge.leaf$samples, model.matrix(~ genotype*trt + block))
# estimateDisp
root.dge.int <- estimateDisp(dge.root,design = root.design.int)
leaf.dge.int <- estimateDisp(dge.leaf,design = leaf.design.int)
## fit linear model
root.fit.int <- glmFit(root.dge.int,root.design.int)
leaf.fit.int <- glmFit(leaf.dge.int,leaf.design.int)
# get DEGs, trt effects (coef=c("trt5E_live","genotypeFPsc:trt5E_live"))
root.trt5E_live.lrt <- glmLRT(root.fit.int,coef = c("trt5E_live","genotypeFPsc:trt5E_live")) # all coeff containing "trt5E_live"
leaf.trt5E_live.lrt <- glmLRT(leaf.fit.int,coef = c("trt5E_live","genotypeFPsc:trt5E_live")) # all coeff containing "trt5E_live"
topTags(root.trt5E_live.lrt);topTags(leaf.trt5E_live.lrt)
## all
root.trt5E_live.DEGs.int.all <- topTags(root.trt5E_live.lrt,n = Inf,p.value = 1)$table
leaf.trt5E_live.DEGs.int.all <- topTags(leaf.trt5E_live.lrt,n = Inf,p.value = 1)$table

# get DEGs, genotype effects (coef = c("genotypeFPsc","genotypeFPsc:trt5E_dead","genotypeFPsc:trt5E_live"))?
root.genotype.lrt <- glmLRT(root.fit.int,coef = c("genotypeFPsc","genotypeFPsc:trt5E_live"))
leaf.genotype.lrt <- glmLRT(leaf.fit.int,coef = c("genotypeFPsc","genotypeFPsc:trt5E_live"))
topTags(root.genotype.lrt);topTags(leaf.genotype.lrt)
## all
root.genotype.DEGs.int.all <- topTags(root.genotype.lrt,n = Inf,p.value = 1)$table
leaf.genotype.DEGs.int.all <- topTags(leaf.genotype.lrt,n = Inf,p.value = 1)$table
# get DEGs, genotype:treatment (live) effects
root.trt5E_live.genotypeFPsc.lrt <- glmLRT(root.fit.int,coef = "genotypeFPsc:trt5E_live")
leaf.trt5E_live.genotypeFPsc.lrt <- glmLRT(leaf.fit.int,coef = "genotypeFPsc:trt5E_live")
topTags(root.trt5E_live.genotypeFPsc.lrt);topTags(leaf.trt5E_live.genotypeFPsc.lrt)
## all
root.trt5E_live.genotypeFPsc.DEGs.int.all <- topTags(root.trt5E_live.genotypeFPsc.lrt,n = Inf,p.value = 1)$table 
leaf.trt5E_live.genotypeFPsc.DEGs.int.all <- topTags(leaf.trt5E_live.genotypeFPsc.lrt,n = Inf,p.value = 1)$table

# add annotiona with full name objects (all)
## genotype
root.genotypeFPsc_trt5E_live.genotypeFPsc.DEGs.int.all.rBD.rR500.v3.0anno <- left_join(root.genotype.DEGs.int.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>% dplyr::select(1:8, AGI, At_symbol, At_short_description, perc_ID)
leaf.genotypeFPsc_trt5E_live.genotypeFPsc.DEGs.int.all.rBD.rR500.v3.0anno <- left_join(leaf.genotype.DEGs.int.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>% dplyr::select(1:8, AGI, At_symbol, At_short_description, perc_ID)
# trt5E_live
root.trt5E_live.genotypeFPsc_trt5E_live.DEGs.int.all.rBD.rR500.v3.0anno <- left_join(root.trt5E_live.DEGs.int.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>% dplyr::select(1:7, AGI, At_symbol, At_short_description, perc_ID)
leaf.trt5E_live.genotypeFPsc_trt5E_live.DEGs.int.all.rBD.rR500.v3.0anno <- left_join(leaf.trt5E_live.DEGs.int.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>% dplyr::select(1:7, AGI, At_symbol, At_short_description, perc_ID)
# genotypeFPsc:trt5E_live
root.trt5E_live.genotypeFPsc.DEGs.int.all.rBD.rR500.v3.0anno <- left_join(root.trt5E_live.genotypeFPsc.DEGs.int.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>%  select(1:6, AGI, At_symbol, At_short_description, perc_ID)
leaf.trt5E_live.genotypeFPsc.DEGs.int.all.rBD.rR500.v3.0anno <- left_join(leaf.trt5E_live.genotypeFPsc.DEGs.int.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>% select(1:6, AGI, At_symbol, At_short_description, perc_ID)

# write csv (naming: "tissue"."coef1"_"coef2"_"coef3","DEGs","model","reference for soil (B, blank;D,dead;L,live)","reference for genotype","annotation version") (all)
## genotype
write.csv(root.genotypeFPsc_trt5E_live.genotypeFPsc.DEGs.int.all.rBD.rR500.v3.0anno,file=file.path("..","output","FULLtoptag","root.genotypeFPsc_trt5E_live.genotypeFPsc.DEGs.int.rBD.rR500.NB.v3.0anno.csv"))
write.csv(leaf.genotypeFPsc_trt5E_live.genotypeFPsc.DEGs.int.all.rBD.rR500.v3.0anno,file=file.path("..","output","FULLtoptag","leaf.genotypeFPsc_trt5E_live.genotypeFPsc.DEGs.int.rBD.rR500.NB.v3.0anno.csv"))
## trt (5E_live)
write.csv(root.trt5E_live.genotypeFPsc_trt5E_live.DEGs.int.all.rBD.rR500.v3.0anno,file=file.path("..","output","FULLtoptag","root.trt5E_live.genotypeFPsc_trt5E_live.DEGs.int.rBD.rR500.NB.v3.0anno.csv"))
write.csv(leaf.trt5E_live.genotypeFPsc_trt5E_live.DEGs.int.all.rBD.rR500.v3.0anno,file=file.path("..","output","FULLtoptag","leaf.trt5E_live.genotypeFPsc_trt5E_live.DEGs.int.rBD.rR500.NB.v3.0anno.csv"))
## genotypeFPsc:trt5E_live
write.csv(root.trt5E_live.genotypeFPsc.DEGs.int.all.rBD.rR500.v3.0anno,file=file.path("..","output","FULLtoptag","root.trt5E_live.genotypeFPsc.DEGs.int.rBD.rR500.NB.v3.0anno.csv"))
write.csv(leaf.trt5E_live.genotypeFPsc.DEGs.int.all.rBD.rR500.v3.0anno,file=file.path("..","output","FULLtoptag","leaf.trt5E_live.genotypeFPsc.DEGs.int.rBD.rR500.NB.v3.0anno.csv"))
```

# clean working directory
```{r include=FALSE}
rm(list=ls())
knitr::opts_chunk$set(echo = TRUE,error=TRUE)
library(edgeR);library(tidyverse)
# annotation file for v3.0annotation
Br.v3.0.At.BLAST <- read_csv(file.path("..","..","..","Annotation","output","v3.0annotation","Brapa_v3.0_annotated.csv")) 
# This annotation is redundant with name (Br grene). Eg 
Br.v3.0.At.BLAST %>% filter(name=="BraA01g040570.3C")
# reduce the redundancy (112418)
Br.v3.0anno.At.BLAST.highscore <- Br.v3.0.At.BLAST %>% group_by(name) %>% arrange(desc(score)) %>% dplyr::slice(1)

# load exp1 v3.0 annotation dge files and sample info
load(file.path("..","output","edgeR_dge_objects.e1.v3.0anno.RData"))
```

# combine blank and dead as a one group (051319)
```{r}
dge.leaf$samples <- dge.leaf$samples %>% 
  mutate(group2=str_replace_all(group,"BLANK_NA|5E_dead","blankdead"),trt=str_replace_all(trt,"BLANK_NA|5E_dead","blankdead")) 
dge.root$samples <- dge.root$samples %>% 
  mutate(group2=str_replace_all(group,"BLANK_NA|5E_dead","blankdead"),trt=str_replace_all(trt,"BLANK_NA|5E_dead","blankdead")) 

```
# 
# Interaction model, rBD.rFPsc
```{r}
# relevel trt and genotype to "blankdead" and "FPsc"
dge.root$samples<-dge.root$samples %>% mutate(trt=fct_relevel(trt,"blankdead")) #%>% str() 
dge.root$samples<-dge.root$samples %>% mutate(genotype=fct_relevel(genotype,"FPsc")) #%>% str() 
dge.leaf$samples<-dge.leaf$samples %>% mutate(trt=fct_relevel(trt,"blankdead")) #%>% str() 
dge.leaf$samples<-dge.leaf$samples %>% mutate(genotype=fct_relevel(genotype,"FPsc")) #%>% str() 
## dge
root.design.int <- with(dge.root$samples, model.matrix(~ genotype*trt + block))
leaf.design.int <- with(dge.leaf$samples, model.matrix(~ genotype*trt + block))
# estimateDisp
root.dge.int <- estimateDisp(dge.root,design = root.design.int)
leaf.dge.int <- estimateDisp(dge.leaf,design = leaf.design.int)
## fit linear model
root.fit.int <- glmFit(root.dge.int,root.design.int)
leaf.fit.int <- glmFit(leaf.dge.int,leaf.design.int)
# get DEGs, trt effects (coef=c("trt5E_live","genotypeR500:trt5E_live"))
root.trt5E_live.lrt <- glmLRT(root.fit.int,coef = c("trt5E_live","genotypeR500:trt5E_live")) # all coeff containing "trt5E_live"
leaf.trt5E_live.lrt <- glmLRT(leaf.fit.int,coef = c("trt5E_live","genotypeR500:trt5E_live")) # all coeff containing "trt5E_live"
topTags(root.trt5E_live.lrt);topTags(leaf.trt5E_live.lrt)
## all
root.trt5E_live.DEGs.int.all <- topTags(root.trt5E_live.lrt,n = Inf,p.value = 1)$table
leaf.trt5E_live.DEGs.int.all <- topTags(leaf.trt5E_live.lrt,n = Inf,p.value = 1)$table

# get DEGs, genotype effects 
root.genotype.lrt <- glmLRT(root.fit.int,coef = c("genotypeR500","genotypeR500:trt5E_live"))
leaf.genotype.lrt <- glmLRT(leaf.fit.int,coef = c("genotypeR500","genotypeR500:trt5E_live"))
topTags(root.genotype.lrt);topTags(leaf.genotype.lrt)
## all
root.genotype.DEGs.int.all <- topTags(root.genotype.lrt,n = Inf,p.value = 1)$table
leaf.genotype.DEGs.int.all <- topTags(leaf.genotype.lrt,n = Inf,p.value = 1)$table
# get DEGs, genotype:treatment (live) effects
root.trt5E_live.genotypeFPsc.lrt <- glmLRT(root.fit.int,coef = "genotypeR500:trt5E_live")
leaf.trt5E_live.genotypeFPsc.lrt <- glmLRT(leaf.fit.int,coef = "genotypeR500:trt5E_live")
topTags(root.trt5E_live.genotypeFPsc.lrt);topTags(leaf.trt5E_live.genotypeFPsc.lrt)
## all
root.trt5E_live.genotypeFPsc.DEGs.int.all <- topTags(root.trt5E_live.genotypeFPsc.lrt,n = Inf,p.value = 1)$table 
leaf.trt5E_live.genotypeFPsc.DEGs.int.all <- topTags(leaf.trt5E_live.genotypeFPsc.lrt,n = Inf,p.value = 1)$table

# add annotiona with full name objects (all)
## genotype
root.genotypeFPsc_trt5E_live.genotypeFPsc.DEGs.int.all.rBD.rFPsc.v3.0anno <- left_join(root.genotype.DEGs.int.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>% dplyr::select(1:8, AGI, At_symbol, At_short_description, perc_ID)
leaf.genotypeFPsc_trt5E_live.genotypeFPsc.DEGs.int.all.rBD.rFPsc.v3.0anno <- left_join(leaf.genotype.DEGs.int.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>% dplyr::select(1:8, AGI, At_symbol, At_short_description, perc_ID)
# trt5E_live
root.trt5E_live.genotypeFPsc_trt5E_live.DEGs.int.all.rBD.rFPsc.v3.0anno <- left_join(root.trt5E_live.DEGs.int.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>% dplyr::select(1:7, AGI, At_symbol, At_short_description, perc_ID)
leaf.trt5E_live.genotypeFPsc_trt5E_live.DEGs.int.all.rBD.rFPsc.v3.0anno <- left_join(leaf.trt5E_live.DEGs.int.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>% dplyr::select(1:7, AGI, At_symbol, At_short_description, perc_ID)
# genotypeFPsc:trt5E_live
root.trt5E_live.genotypeFPsc.DEGs.int.all.rBD.rFPsc.v3.0anno <- left_join(root.trt5E_live.genotypeFPsc.DEGs.int.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>%  select(1:6, AGI, At_symbol, At_short_description, perc_ID)
leaf.trt5E_live.genotypeFPsc.DEGs.int.all.rBD.rFPsc.v3.0anno <- left_join(leaf.trt5E_live.genotypeFPsc.DEGs.int.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>% select(1:6, AGI, At_symbol, At_short_description, perc_ID)

# write csv (naming: "tissue"."coef1"_"coef2"_"coef3","DEGs","model","reference for soil (B, blank;D,dead;L,live)","reference for genotype","annotation version") (all)
## genotype
write.csv(root.genotypeFPsc_trt5E_live.genotypeFPsc.DEGs.int.all.rBD.rFPsc.v3.0anno,file=file.path("..","output","FULLtoptag","root.genotypeFPsc_trt5E_live.genotypeFPsc.DEGs.int.rBD.rFPsc.NB.v3.0anno.csv"))
write.csv(leaf.genotypeFPsc_trt5E_live.genotypeFPsc.DEGs.int.all.rBD.rFPsc.v3.0anno,file=file.path("..","output","FULLtoptag","leaf.genotypeFPsc_trt5E_live.genotypeFPsc.DEGs.int.rBD.rFPsc.NB.v3.0anno.csv"))
## trt (5E_live)
write.csv(root.trt5E_live.genotypeFPsc_trt5E_live.DEGs.int.all.rBD.rFPsc.v3.0anno,file=file.path("..","output","FULLtoptag","root.trt5E_live.genotypeFPsc_trt5E_live.DEGs.int.rBD.rFPsc.NB.v3.0anno.csv"))
write.csv(leaf.trt5E_live.genotypeFPsc_trt5E_live.DEGs.int.all.rBD.rFPsc.v3.0anno,file=file.path("..","output","FULLtoptag","leaf.trt5E_live.genotypeFPsc_trt5E_live.DEGs.int.rBD.rFPsc.NB.v3.0anno.csv"))
## genotypeFPsc:trt5E_live
write.csv(root.trt5E_live.genotypeFPsc.DEGs.int.all.rBD.rFPsc.v3.0anno,file=file.path("..","output","FULLtoptag","root.trt5E_live.genotypeFPsc.DEGs.int.rBD.rFPsc.NB.v3.0anno.csv"))
write.csv(leaf.trt5E_live.genotypeFPsc.DEGs.int.all.rBD.rFPsc.v3.0anno,file=file.path("..","output","FULLtoptag","leaf.trt5E_live.genotypeFPsc.DEGs.int.rBD.rFPsc.NB.v3.0anno.csv"))
```

# Interaction model, rBD.rFPsc
```{r}
# relevel trt and genotype to "blankdead" and "FPsc"
dge.root$samples<-dge.root$samples %>% mutate(trt=fct_relevel(trt,"blankdead")) #%>% str() 
dge.root$samples<-dge.root$samples %>% mutate(genotype=fct_relevel(genotype,"FPsc")) #%>% str() 
dge.leaf$samples<-dge.leaf$samples %>% mutate(trt=fct_relevel(trt,"blankdead")) #%>% str() 
dge.leaf$samples<-dge.leaf$samples %>% mutate(genotype=fct_relevel(genotype,"FPsc")) #%>% str() 
## dge
root.design.add <- with(dge.root$samples, model.matrix(~ genotype+trt + block))
leaf.design.add <- with(dge.leaf$samples, model.matrix(~ genotype+trt + block))
# estimateDisp
root.dge.add <- estimateDisp(dge.root,design = root.design.add)
leaf.dge.add <- estimateDisp(dge.leaf,design = leaf.design.add)
## fit linear model
root.fit.add <- glmFit(root.dge.add,root.design.add)
leaf.fit.add <- glmFit(leaf.dge.add,leaf.design.add)
# get DEGs, trt effects (coef="trt5E_live")
root.trt5E_live.lrt <- glmLRT(root.fit.add,coef = "trt5E_live") 
leaf.trt5E_live.lrt <- glmLRT(leaf.fit.add,coef = "trt5E_live") 
topTags(root.trt5E_live.lrt);topTags(leaf.trt5E_live.lrt)
## all
root.trt5E_live.DEGs.add.all <- topTags(root.trt5E_live.lrt,n = Inf,p.value = 1)$table
leaf.trt5E_live.DEGs.add.all <- topTags(leaf.trt5E_live.lrt,n = Inf,p.value = 1)$table

# get DEGs, genotype effects 
root.genotype.lrt <- glmLRT(root.fit.add,coef = "genotypeR500")
leaf.genotype.lrt <- glmLRT(leaf.fit.add,coef = "genotypeR500")
topTags(root.genotype.lrt);topTags(leaf.genotype.lrt)
## all
root.genotype.DEGs.add.all <- topTags(root.genotype.lrt,n = Inf,p.value = 1)$table
leaf.genotype.DEGs.add.all <- topTags(leaf.genotype.lrt,n = Inf,p.value = 1)$table

# add annotiona with full name objects (all)
## genotype
root.genotypeFPsc.DEGs.add.all.rBD.rFPsc.v3.0anno <- left_join(root.genotype.DEGs.add.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>% dplyr::select(1:8, AGI, At_symbol, At_short_description, perc_ID)
leaf.genotypeFPsc.DEGs.add.all.rBD.rFPsc.v3.0anno <- left_join(leaf.genotype.DEGs.add.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>% dplyr::select(1:8, AGI, At_symbol, At_short_description, perc_ID)
# trt5E_live
root.trt5E_live.DEGs.add.all.rBD.rFPsc.v3.0anno <- left_join(root.trt5E_live.DEGs.add.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>% dplyr::select(1:7, AGI, At_symbol, At_short_description, perc_ID)
leaf.trt5E_live.DEGs.add.all.rBD.rFPsc.v3.0anno <- left_join(leaf.trt5E_live.DEGs.add.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>% dplyr::select(1:7, AGI, At_symbol, At_short_description, perc_ID)

# write csv (naming: "tissue"."coef1"_"coef2"_"coef3","DEGs","model","reference for soil (B, blank;D,dead;L,live)","reference for genotype","annotation version") (all)
## genotype
write.csv(root.genotypeFPsc.DEGs.add.all.rBD.rFPsc.v3.0anno,file=file.path("..","output","FULLtoptag","root.genotypeFPsc.DEGs.add.rBD.rFPsc.NB.v3.0anno.csv"))
write.csv(leaf.genotypeFPsc.DEGs.add.all.rBD.rFPsc.v3.0anno,file=file.path("..","output","FULLtoptag","leaf.genotypeFPsc.DEGs.add.rBD.rFPsc.NB.v3.0anno.csv"))
## trt (5E_live)
write.csv(root.trt5E_live.DEGs.add.all.rBD.rFPsc.v3.0anno,file=file.path("..","output","FULLtoptag","root.trt5E_live.DEGs.add.rBD.rFPsc.NB.v3.0anno.csv"))
write.csv(leaf.trt5E_live.DEGs.add.all.rBD.rFPsc.v3.0anno,file=file.path("..","output","FULLtoptag","leaf.trt5E_live.DEGs.add.rBD.rFPsc.NB.v3.0anno.csv"))
```


# Subsetting data {data-navmenu="subset data by genotype"}
```{r}
# clean working directory
rm(list=ls())
knitr::opts_chunk$set(echo = TRUE,error=TRUE)
library(edgeR);library(tidyverse)
# annotation file for v3.0annotation
Br.v3.0.At.BLAST <- read_csv(file.path("..","..","..","Annotation","output","v3.0annotation","Brapa_v3.0_annotated.csv")) 
# This annotation is redundant with name (Br grene). Eg 
Br.v3.0.At.BLAST %>% filter(name=="BraA01g040570.3C")
# reduce the redundancy (112418)
Br.v3.0anno.At.BLAST.highscore <- Br.v3.0.At.BLAST %>% group_by(name) %>% dplyr::arrange(desc(score)) %>% dplyr::slice(1)

# load exp1 v3.0 annotation dge files and sample info
load(file.path("..","output","edgeR_dge_objects.e1.v3.0anno.RData")) # for exp1
dge.root.R500<-dge.root[,dge.root$samples$genotype=="R500"]
dge.root.FPsc<-dge.root[,dge.root$samples$genotype=="FPsc"]
dge.leaf.R500<-dge.leaf[,dge.leaf$samples$genotype=="R500"]
dge.leaf.FPsc<-dge.leaf[,dge.leaf$samples$genotype=="FPsc"]

# reset norm.factors
dge.root.R500$samples$norm.factors<-1 # No matter which case, non-normalize libraries.
dge.root.FPsc$samples$norm.factors<-1 # No matter which case, non-normalize libraries.
dge.leaf.R500$samples$norm.factors<-1 # No matter which case, non-normalize libraries.
dge.leaf.FPsc$samples$norm.factors<-1 # No matter which case, non-normalize libraries.
# remove genes with low expression level
dge.root.R500<-dge.root.R500[rowSums(cpm(dge.root.R500)>1) >= 3,,keep.lib.sizes=FALSE]
dge.root.FPsc<-dge.root.FPsc[rowSums(cpm(dge.root.FPsc)>1) >= 3,,keep.lib.sizes=FALSE]
dge.leaf.R500<-dge.leaf.R500[rowSums(cpm(dge.leaf.R500)>1) >= 3,,keep.lib.sizes=FALSE]
dge.leaf.FPsc<-dge.leaf.FPsc[rowSums(cpm(dge.leaf.FPsc)>1) >= 3,,keep.lib.sizes=FALSE]
# normalize
dge.root.R500 <- calcNormFactors(dge.root.R500)
dge.root.FPsc <- calcNormFactors(dge.root.FPsc)
dge.leaf.R500 <- calcNormFactors(dge.leaf.R500)
dge.leaf.FPsc <- calcNormFactors(dge.leaf.FPsc)

```

#relevel trt to "dead" ("rD")  {data-navmenu="subset data by genotype"}
```{r}
# relevel trt to "dead" 
dge.root.R500$samples<-dge.root.R500$samples %>% mutate(trt=fct_relevel(trt,"5E_dead")) # %>% str() 
dge.root.FPsc$samples<-dge.root.FPsc$samples %>% mutate(trt=fct_relevel(trt,"5E_dead")) # %>% str() 
dge.leaf.R500$samples<-dge.leaf.R500$samples %>% mutate(trt=fct_relevel(trt,"5E_dead")) # %>% str() 
dge.leaf.FPsc$samples<-dge.leaf.FPsc$samples %>% mutate(trt=fct_relevel(trt,"5E_dead")) # %>% str() 
## model
root.R500.design.add <- with(dge.root.R500$samples, model.matrix(~ trt + block))
root.FPsc.design.add <- with(dge.root.FPsc$samples, model.matrix(~ trt + block))
leaf.R500.design.add <- with(dge.leaf.R500$samples, model.matrix(~ trt + block))
leaf.FPsc.design.add <- with(dge.leaf.FPsc$samples, model.matrix(~ trt + block))

# estimateDisp
root.R500.dge.add <- estimateDisp(dge.root.R500,design = root.R500.design.add)
root.FPsc.dge.add <- estimateDisp(dge.root.FPsc,design = root.FPsc.design.add)
leaf.R500.dge.add <- estimateDisp(dge.leaf.R500,design = leaf.R500.design.add)
leaf.FPsc.dge.add <- estimateDisp(dge.leaf.FPsc,design = leaf.FPsc.design.add)

## fit linear model
root.R500.fit.add <- glmFit(root.R500.dge.add,root.R500.design.add)
root.FPsc.fit.add <- glmFit(root.FPsc.dge.add,root.FPsc.design.add)
leaf.R500.fit.add <- glmFit(leaf.R500.dge.add,leaf.R500.design.add)
leaf.FPsc.fit.add <- glmFit(leaf.FPsc.dge.add,leaf.FPsc.design.add)

# get DEGs, trt effects (coef=c("trtlive"))
root.R500.trtlive.lrt <- glmLRT(root.R500.fit.add,coef = c("trt5E_live"))
root.FPsc.trtlive.lrt <- glmLRT(root.FPsc.fit.add,coef = c("trt5E_live"))
leaf.R500.trtlive.lrt <- glmLRT(leaf.R500.fit.add,coef = c("trt5E_live"))
leaf.FPsc.trtlive.lrt <- glmLRT(leaf.FPsc.fit.add,coef = c("trt5E_live"))
topTags(root.R500.trtlive.lrt);topTags(root.FPsc.trtlive.lrt);topTags(leaf.R500.trtlive.lrt);topTags(leaf.FPsc.trtlive.lrt)
## all
root.R500.trtlive.DEGs.all <- topTags(root.R500.trtlive.lrt,n = Inf,p.value = 1)$table
root.FPsc.trtlive.DEGs.all <- topTags(root.FPsc.trtlive.lrt,n = Inf,p.value = 1)$table
leaf.R500.trtlive.DEGs.all <- topTags(leaf.R500.trtlive.lrt,n = Inf,p.value = 1)$table
leaf.FPsc.trtlive.DEGs.all <- topTags(leaf.FPsc.trtlive.lrt,n = Inf,p.value = 1)$table

## pvalue < 0.05
root.R500.trtlive.DEGs <- topTags(root.R500.trtlive.lrt,n = Inf,p.value = 0.05)$table;nrow(root.R500.trtlive.DEGs) #
root.FPsc.trtlive.DEGs <- topTags(root.FPsc.trtlive.lrt,n = Inf,p.value = 0.05)$table;nrow(root.FPsc.trtlive.DEGs)  # 
leaf.R500.trtlive.DEGs <- topTags(leaf.R500.trtlive.lrt,n = Inf,p.value = 0.05)$table;nrow(leaf.R500.trtlive.DEGs) #
leaf.FPsc.trtlive.DEGs <- topTags(leaf.FPsc.trtlive.lrt,n = Inf,p.value = 0.05)$table;nrow(leaf.FPsc.trtlive.DEGs)  # 
```

# Adding annotation {data-navmenu="subset data by genotype"}
```{R}
# add annotiona with full name objects (all)
root.R500.trtlive.DEGs.all.rD.v3.0anno <- left_join(root.R500.trtlive.DEGs.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>%  dplyr::select(1:6, AGI, At_symbol, At_short_description, perc_ID)
root.FPsc.trtlive.DEGs.all.rD.v3.0anno <- left_join(root.FPsc.trtlive.DEGs.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>%  dplyr::select(1:6, AGI, At_symbol, At_short_description, perc_ID)
leaf.R500.trtlive.DEGs.all.rD.v3.0anno <- left_join(leaf.R500.trtlive.DEGs.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>%  dplyr::select(1:6, AGI, At_symbol, At_short_description, perc_ID)
leaf.FPsc.trtlive.DEGs.all.rD.v3.0anno <- left_join(leaf.FPsc.trtlive.DEGs.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>%  dplyr::select(1:6, AGI, At_symbol, At_short_description, perc_ID) # NULL

# write csv (all)
## trt
write.csv(root.R500.trtlive.DEGs.all.rD.v3.0anno,file=file.path("..","output","FULLtoptag","root.R500.trtlive.DEGs.all.rD.NB.v3.0anno.csv"))
write.csv(root.FPsc.trtlive.DEGs.all.rD.v3.0anno,file=file.path("..","output","FULLtoptag","root.FPsc.trtlive.DEGs.all.rD.NB.v3.0anno.csv"))
write.csv(leaf.R500.trtlive.DEGs.all.rD.v3.0anno,file=file.path("..","output","FULLtoptag","leaf.R500.trtlive.DEGs.all.rD.NB.v3.0anno.csv"))
write.csv(leaf.FPsc.trtlive.DEGs.all.rD.v3.0anno,file=file.path("..","output","FULLtoptag","leaf.FPsc.trtlive.DEGs.all.rD.NB.v3.0anno.csv"))

```

# Subsetting data {data-navmenu="subset data by genotype with rBD"}
```{r}
# clean working directory
rm(list=ls())
knitr::opts_chunk$set(echo = TRUE,error=TRUE)
library(edgeR);library(tidyverse)
# annotation file for v3.0annotation
Br.v3.0.At.BLAST <- read_csv(file.path("..","..","..","Annotation","output","v3.0annotation","Brapa_v3.0_annotated.csv")) 
# This annotation is redundant with name (Br grene). Eg 
Br.v3.0.At.BLAST %>% filter(name=="BraA01g040570.3C")
# reduce the redundancy (112418)
Br.v3.0anno.At.BLAST.highscore <- Br.v3.0.At.BLAST %>% group_by(name) %>% dplyr::arrange(desc(score)) %>% dplyr::slice(1)

# load exp1 v3.0 annotation dge files and sample info
load(file.path("..","output","edgeR_dge_objects.e1.v3.0anno.RData")) # for exp1
```

# combine blank and dead as a one group {data-navmenu="subset data by genotype with rBD"}
```{r}
dge.leaf$samples <- dge.leaf$samples %>% 
  mutate(group2=str_replace_all(group,"BLANK_NA|5E_dead","blankdead"),trt=str_replace_all(trt,"BLANK_NA|5E_dead","blankdead")) 
dge.root$samples <- dge.root$samples %>% 
  mutate(group2=str_replace_all(group,"BLANK_NA|5E_dead","blankdead"),trt=str_replace_all(trt,"BLANK_NA|5E_dead","blankdead")) 
```

# subsetting by gentoype {data-navmenu="subset data by genotype with rBD"}
```{r}
dge.root.R500<-dge.root[,dge.root$samples$genotype=="R500"]
dge.root.FPsc<-dge.root[,dge.root$samples$genotype=="FPsc"]
dge.leaf.R500<-dge.leaf[,dge.leaf$samples$genotype=="R500"]
dge.leaf.FPsc<-dge.leaf[,dge.leaf$samples$genotype=="FPsc"]

# reset norm.factors
dge.root.R500$samples$norm.factors<-1 # No matter which case, non-normalize libraries.
dge.root.FPsc$samples$norm.factors<-1 # No matter which case, non-normalize libraries.
dge.leaf.R500$samples$norm.factors<-1 # No matter which case, non-normalize libraries.
dge.leaf.FPsc$samples$norm.factors<-1 # No matter which case, non-normalize libraries.
# remove genes with low expression level
dge.root.R500<-dge.root.R500[rowSums(cpm(dge.root.R500)>1) >= 3,,keep.lib.sizes=FALSE]
dge.root.FPsc<-dge.root.FPsc[rowSums(cpm(dge.root.FPsc)>1) >= 3,,keep.lib.sizes=FALSE]
dge.leaf.R500<-dge.leaf.R500[rowSums(cpm(dge.leaf.R500)>1) >= 3,,keep.lib.sizes=FALSE]
dge.leaf.FPsc<-dge.leaf.FPsc[rowSums(cpm(dge.leaf.FPsc)>1) >= 3,,keep.lib.sizes=FALSE]
# normalize
dge.root.R500 <- calcNormFactors(dge.root.R500)
dge.root.FPsc <- calcNormFactors(dge.root.FPsc)
dge.leaf.R500 <- calcNormFactors(dge.leaf.R500)
dge.leaf.FPsc <- calcNormFactors(dge.leaf.FPsc)

```

# relevel trt to "blank and dead" ("rBD")  {data-navmenu="subset data by genotype with rBD"}
```{r}

# relevel trt to "blankd and dead" (under construction)
dge.root.R500$samples<-dge.root.R500$samples %>% mutate(trt=fct_relevel(trt,"blankdead")) # %>% str() 
dge.root.FPsc$samples<-dge.root.FPsc$samples %>% mutate(trt=fct_relevel(trt,"blankdead")) # %>% str() 
dge.leaf.R500$samples<-dge.leaf.R500$samples %>% mutate(trt=fct_relevel(trt,"blankdead")) # %>% str() 
dge.leaf.FPsc$samples<-dge.leaf.FPsc$samples %>% mutate(trt=fct_relevel(trt,"blankdead")) # %>% str() 
## model
root.R500.design.add <- with(dge.root.R500$samples, model.matrix(~ trt + block))
root.FPsc.design.add <- with(dge.root.FPsc$samples, model.matrix(~ trt + block))
leaf.R500.design.add <- with(dge.leaf.R500$samples, model.matrix(~ trt + block))
leaf.FPsc.design.add <- with(dge.leaf.FPsc$samples, model.matrix(~ trt + block))

# estimateDisp
root.R500.dge.add <- estimateDisp(dge.root.R500,design = root.R500.design.add)
root.FPsc.dge.add <- estimateDisp(dge.root.FPsc,design = root.FPsc.design.add)
leaf.R500.dge.add <- estimateDisp(dge.leaf.R500,design = leaf.R500.design.add)
leaf.FPsc.dge.add <- estimateDisp(dge.leaf.FPsc,design = leaf.FPsc.design.add)

## fit linear model
root.R500.fit.add <- glmFit(root.R500.dge.add,root.R500.design.add)
root.FPsc.fit.add <- glmFit(root.FPsc.dge.add,root.FPsc.design.add)
leaf.R500.fit.add <- glmFit(leaf.R500.dge.add,leaf.R500.design.add)
leaf.FPsc.fit.add <- glmFit(leaf.FPsc.dge.add,leaf.FPsc.design.add)

# get DEGs, trt effects (coef=c("trtlive"))
root.R500.trtlive.lrt <- glmLRT(root.R500.fit.add,coef = c("trt5E_live"))
root.FPsc.trtlive.lrt <- glmLRT(root.FPsc.fit.add,coef = c("trt5E_live"))
leaf.R500.trtlive.lrt <- glmLRT(leaf.R500.fit.add,coef = c("trt5E_live"))
leaf.FPsc.trtlive.lrt <- glmLRT(leaf.FPsc.fit.add,coef = c("trt5E_live"))
topTags(root.R500.trtlive.lrt);topTags(root.FPsc.trtlive.lrt);topTags(leaf.R500.trtlive.lrt);topTags(leaf.FPsc.trtlive.lrt)
## all
root.R500.trtlive.DEGs.all <- topTags(root.R500.trtlive.lrt,n = Inf,p.value = 1)$table
root.FPsc.trtlive.DEGs.all <- topTags(root.FPsc.trtlive.lrt,n = Inf,p.value = 1)$table
leaf.R500.trtlive.DEGs.all <- topTags(leaf.R500.trtlive.lrt,n = Inf,p.value = 1)$table
leaf.FPsc.trtlive.DEGs.all <- topTags(leaf.FPsc.trtlive.lrt,n = Inf,p.value = 1)$table

## pvalue < 0.05
root.R500.trtlive.DEGs <- topTags(root.R500.trtlive.lrt,n = Inf,p.value = 0.05)$table;nrow(root.R500.trtlive.DEGs) #
root.FPsc.trtlive.DEGs <- topTags(root.FPsc.trtlive.lrt,n = Inf,p.value = 0.05)$table;nrow(root.FPsc.trtlive.DEGs)  # 
leaf.R500.trtlive.DEGs <- topTags(leaf.R500.trtlive.lrt,n = Inf,p.value = 0.05)$table;nrow(leaf.R500.trtlive.DEGs) #
leaf.FPsc.trtlive.DEGs <- topTags(leaf.FPsc.trtlive.lrt,n = Inf,p.value = 0.05)$table;nrow(leaf.FPsc.trtlive.DEGs)  # 
```

# Adding annotation {data-navmenu="subset data by genotype with rBD"}
```{r}
# add annotiona with full name objects (all)
root.R500.trtlive.DEGs.all.rBD.v3.0anno <- left_join(root.R500.trtlive.DEGs.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>%  dplyr::select(1:6, AGI, At_symbol, At_short_description, perc_ID)
root.FPsc.trtlive.DEGs.all.rBD.v3.0anno <- left_join(root.FPsc.trtlive.DEGs.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>%  dplyr::select(1:6, AGI, At_symbol, At_short_description, perc_ID)
leaf.R500.trtlive.DEGs.all.rBD.v3.0anno <- left_join(leaf.R500.trtlive.DEGs.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>%  dplyr::select(1:6, AGI, At_symbol, At_short_description, perc_ID)
leaf.FPsc.trtlive.DEGs.all.rBD.v3.0anno <- left_join(leaf.FPsc.trtlive.DEGs.all,Br.v3.0anno.At.BLAST.highscore,by=c(genes="name")) %>%  dplyr::select(1:6, AGI, At_symbol, At_short_description, perc_ID) # NULL

# write csv (all)
## trt
write.csv(root.R500.trtlive.DEGs.all.rBD.v3.0anno,file=file.path("..","output","FULLtoptag","root.R500.trtlive.DEGs.all.rBD.v3.0anno.csv"))
write.csv(root.FPsc.trtlive.DEGs.all.rBD.v3.0anno,file=file.path("..","output","FULLtoptag","root.FPsc.trtlive.DEGs.all.rBD.v3.0anno.csv"))
write.csv(leaf.R500.trtlive.DEGs.all.rBD.v3.0anno,file=file.path("..","output","FULLtoptag","leaf.R500.trtlive.DEGs.all.rBD.v3.0anno.csv"))
write.csv(leaf.FPsc.trtlive.DEGs.all.rBD.v3.0anno,file=file.path("..","output","FULLtoptag","leaf.FPsc.trtlive.DEGs.all.rBD.v3.0anno.csv"))

```



# Try quasi-likelihood (QL) F-Test (see edgeRUsersGuide2018 section 2.10.3)
##  Additive model, rD.rFPsc
```{r}
# load exp1 v3.0 annotation dge files and sample info
load(file.path("..","output","edgeR_dge_objects.e1.v3.0anno.RData"))
# relevel trt and genotype to "5E_dead" and "FPsc"
dge.root$samples<-dge.root$samples %>% mutate(trt=fct_relevel(trt,"5E_dead")) #%>% str() 
dge.root$samples<-dge.root$samples %>% mutate(genotype=fct_relevel(genotype,"FPsc")) #%>% str() 
dge.leaf$samples<-dge.leaf$samples %>% mutate(trt=fct_relevel(trt,"5E_dead")) #%>% str() 
dge.leaf$samples<-dge.leaf$samples %>% mutate(genotype=fct_relevel(genotype,"FPsc")) #%>% str() 
## dge
root.design.add <- with(dge.root$samples, model.matrix(~ genotype + trt + block))
leaf.design.add <- with(dge.leaf$samples, model.matrix(~ genotype + trt + block))
# estimateDisp
root.dge.add <- estimateDisp(dge.root,design = root.design.add)
leaf.dge.add <- estimateDisp(dge.leaf,design = leaf.design.add)
# QLF
root.fit <- glmQLFit(root.dge.add, root.design.add)
leaf.fit <- glmQLFit(leaf.dge.add, leaf.design.add)
# to compare "trt5E_live" to "5E_dead"
root.trt5E_live.add.qlf <- glmQLFTest(root.fit,coef="trt5E_live")
root.trt5E_live.DEGs.add.rD.qlf.all <- topTags(root.trt5E_live.add.qlf,n = Inf,p.value = 1)$table
leaf.trt5E_live.add.qlf <- glmQLFTest(leaf.fit,coef="trt5E_live")
leaf.trt5E_live.DEGs.add.rD.qlf.all <- topTags(leaf.trt5E_live.add.qlf,n = Inf,p.value = 1)$table
## writing output table
write.csv(root.trt5E_live.DEGs.add.rD.qlf.all,file=file.path("..","output","FULLtoptag","root.trt5E_live.DEGs.add.rD.rFPsc.QLF.v3.0anno.csv"))
write.csv(leaf.trt5E_live.DEGs.add.rD.qlf.all,file=file.path("..","output","FULLtoptag","leaf.trt5E_live.DEGs.add.rD.rFPsc.QLF.v3.0anno.csv"))

# to compare "trt5E_live" vs "trtBLANK_NA"
# relevel trt and genotype to "trtBLANK_NA" 
dge.root$samples<-dge.root$samples %>% mutate(trt=fct_relevel(trt,"BLANK_NA"))

dge.leaf$samples<-dge.leaf$samples %>% mutate(trt=fct_relevel(trt,"BLANK_NA")) 
## dge
root.design.add <- with(dge.root$samples, model.matrix(~ genotype + trt + block))
leaf.design.add <- with(dge.leaf$samples, model.matrix(~ genotype + trt + block))
# estimateDisp
root.dge.add <- estimateDisp(dge.root,design = root.design.add)
leaf.dge.add <- estimateDisp(dge.leaf,design = leaf.design.add)
# QL
root.fit <- glmQLFit(root.dge.add, root.design.add)
leaf.fit <- glmQLFit(leaf.dge.add, leaf.design.add)
# to compare "trt5E_live" to "5E_dead"
root.trt5E_live.add.qlf <- glmQLFTest(root.fit,coef="trt5E_live")
root.trt5E_live.DEGs.add.rB.qlf.all <- topTags(root.trt5E_live.add.qlf,n = Inf,p.value = 1)$table
leaf.trt5E_live.add.qlf <- glmQLFTest(leaf.fit,coef="trt5E_live")
leaf.trt5E_live.DEGs.add.rB.qlf.all <- topTags(leaf.trt5E_live.add.qlf,n = Inf,p.value = 1)$table
## writing output table
write.csv(root.trt5E_live.DEGs.add.rB.qlf.all,file=file.path("..","output","FULLtoptag","root.trt5E_live.DEGs.add.rB.rFPsc.QLF.v3.0anno.csv"))
write.csv(leaf.trt5E_live.DEGs.add.rB.qlf.all,file=file.path("..","output","FULLtoptag","leaf.trt5E_live.DEGs.add.rB.rFPsc.QLF.v3.0anno.csv"))
```

# rB and rD combined. 
Combine blank and dead as a one group (70919)
```{r}
dge.leaf$samples <- dge.leaf$samples %>% 
  mutate(group2=str_replace_all(group,"BLANK_NA|5E_dead","blankdead"),trt=str_replace_all(trt,"BLANK_NA|5E_dead","blankdead")) 
dge.root$samples <- dge.root$samples %>% 
  mutate(group2=str_replace_all(group,"BLANK_NA|5E_dead","blankdead"),trt=str_replace_all(trt,"BLANK_NA|5E_dead","blankdead")) 
# to compare "trt5E_live" vs "blankdead"
dge.root$samples<-dge.root$samples %>% mutate(trt=fct_relevel(trt,"blankdead"))

dge.leaf$samples<-dge.leaf$samples %>% mutate(trt=fct_relevel(trt,"blankdead")) 
## dge
root.design.add <- with(dge.root$samples, model.matrix(~ genotype + trt + block))
leaf.design.add <- with(dge.leaf$samples, model.matrix(~ genotype + trt + block))
# estimateDisp
root.dge.add <- estimateDisp(dge.root,design = root.design.add)
leaf.dge.add <- estimateDisp(dge.leaf,design = leaf.design.add)
# QLF
root.fit <- glmQLFit(root.dge.add, root.design.add)
leaf.fit <- glmQLFit(leaf.dge.add, leaf.design.add)
root.trt5E_live.add.qlf <- glmQLFTest(root.fit,coef="trt5E_live")
root.trt5E_live.DEGs.add.rBD.qlf.all <- topTags(root.trt5E_live.add.qlf,n = Inf,p.value = 1)$table
leaf.trt5E_live.add.qlf <- glmQLFTest(leaf.fit,coef="trt5E_live")
leaf.trt5E_live.DEGs.add.rBD.qlf.all <- topTags(leaf.trt5E_live.add.qlf,n = Inf,p.value = 1)$table
## writing output table
write.csv(root.trt5E_live.DEGs.add.rBD.qlf.all,file=file.path("..","output","FULLtoptag","root.trt5E_live.DEGs.add.rBD.rFPsc.QLF.v3.0anno.csv"))
write.csv(leaf.trt5E_live.DEGs.add.rBD.qlf.all,file=file.path("..","output","FULLtoptag","leaf.trt5E_live.DEGs.add.rBD.rFPsc.QLF.v3.0anno.csv"))
```


